// @pulumi/azure-native/types/enums/storagesync/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.LocalCacheMode = exports.InitialUploadPolicy = exports.InitialDownloadPolicy = exports.IncomingTrafficPolicy = exports.FeatureStatus = void 0;
exports.FeatureStatus = {
    On: "on",
    Off: "off",
};
exports.IncomingTrafficPolicy = {
    AllowAllTraffic: "AllowAllTraffic",
    AllowVirtualNetworksOnly: "AllowVirtualNetworksOnly",
};
exports.InitialDownloadPolicy = {
    NamespaceOnly: "NamespaceOnly",
    NamespaceThenModifiedFiles: "NamespaceThenModifiedFiles",
    AvoidTieredFiles: "AvoidTieredFiles",
};
exports.InitialUploadPolicy = {
    ServerAuthoritative: "ServerAuthoritative",
    Merge: "Merge",
};
exports.LocalCacheMode = {
    DownloadNewAndModifiedFiles: "DownloadNewAndModifiedFiles",
    UpdateLocallyCachedFiles: "UpdateLocallyCachedFiles",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
