// @pulumi/azure-native/datalakeanalytics/account.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Data Lake Analytics account object, containing all information associated with the named Data Lake Analytics account.
 *
 * Uses Azure REST API version 2019-11-01-preview. In version 2.x of the Azure Native provider, it used API version 2019-11-01-preview.
 */
class Account extends pulumi.CustomResource {
    /**
     * Get an existing Account resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Account(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Account.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Account.__pulumiType;
    }
    /**
     * Create a Account resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e, _f, _g;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataLakeStoreAccounts === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataLakeStoreAccounts'");
            }
            if ((!args || args.defaultDataLakeStoreAccount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultDataLakeStoreAccount'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["computePolicies"] = args ? args.computePolicies : undefined;
            resourceInputs["dataLakeStoreAccounts"] = args ? args.dataLakeStoreAccounts : undefined;
            resourceInputs["defaultDataLakeStoreAccount"] = args ? args.defaultDataLakeStoreAccount : undefined;
            resourceInputs["firewallAllowAzureIps"] = (_a = (args ? args.firewallAllowAzureIps : undefined)) !== null && _a !== void 0 ? _a : "Disabled";
            resourceInputs["firewallRules"] = args ? args.firewallRules : undefined;
            resourceInputs["firewallState"] = (_b = (args ? args.firewallState : undefined)) !== null && _b !== void 0 ? _b : "Disabled";
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["maxDegreeOfParallelism"] = (_c = (args ? args.maxDegreeOfParallelism : undefined)) !== null && _c !== void 0 ? _c : 30;
            resourceInputs["maxDegreeOfParallelismPerJob"] = (_d = (args ? args.maxDegreeOfParallelismPerJob : undefined)) !== null && _d !== void 0 ? _d : 32;
            resourceInputs["maxJobCount"] = (_e = (args ? args.maxJobCount : undefined)) !== null && _e !== void 0 ? _e : 3;
            resourceInputs["minPriorityPerJob"] = args ? args.minPriorityPerJob : undefined;
            resourceInputs["newTier"] = (_f = (args ? args.newTier : undefined)) !== null && _f !== void 0 ? _f : "Consumption";
            resourceInputs["queryStoreRetention"] = (_g = (args ? args.queryStoreRetention : undefined)) !== null && _g !== void 0 ? _g : 30;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storageAccounts"] = args ? args.storageAccounts : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["accountId"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["currentTier"] = undefined /*out*/;
            resourceInputs["debugDataAccessLevel"] = undefined /*out*/;
            resourceInputs["defaultDataLakeStoreAccountType"] = undefined /*out*/;
            resourceInputs["endpoint"] = undefined /*out*/;
            resourceInputs["hiveMetastores"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["maxActiveJobCountPerUser"] = undefined /*out*/;
            resourceInputs["maxJobRunningTimeInMin"] = undefined /*out*/;
            resourceInputs["maxQueuedJobCountPerUser"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicDataLakeStoreAccounts"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["systemMaxDegreeOfParallelism"] = undefined /*out*/;
            resourceInputs["systemMaxJobCount"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualNetworkRules"] = undefined /*out*/;
        }
        else {
            resourceInputs["accountId"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["computePolicies"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["currentTier"] = undefined /*out*/;
            resourceInputs["dataLakeStoreAccounts"] = undefined /*out*/;
            resourceInputs["debugDataAccessLevel"] = undefined /*out*/;
            resourceInputs["defaultDataLakeStoreAccount"] = undefined /*out*/;
            resourceInputs["defaultDataLakeStoreAccountType"] = undefined /*out*/;
            resourceInputs["endpoint"] = undefined /*out*/;
            resourceInputs["firewallAllowAzureIps"] = undefined /*out*/;
            resourceInputs["firewallRules"] = undefined /*out*/;
            resourceInputs["firewallState"] = undefined /*out*/;
            resourceInputs["hiveMetastores"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maxActiveJobCountPerUser"] = undefined /*out*/;
            resourceInputs["maxDegreeOfParallelism"] = undefined /*out*/;
            resourceInputs["maxDegreeOfParallelismPerJob"] = undefined /*out*/;
            resourceInputs["maxJobCount"] = undefined /*out*/;
            resourceInputs["maxJobRunningTimeInMin"] = undefined /*out*/;
            resourceInputs["maxQueuedJobCountPerUser"] = undefined /*out*/;
            resourceInputs["minPriorityPerJob"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["newTier"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicDataLakeStoreAccounts"] = undefined /*out*/;
            resourceInputs["queryStoreRetention"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["storageAccounts"] = undefined /*out*/;
            resourceInputs["systemMaxDegreeOfParallelism"] = undefined /*out*/;
            resourceInputs["systemMaxJobCount"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualNetworkRules"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:datalakeanalytics/v20151001preview:Account" }, { type: "azure-native:datalakeanalytics/v20161101:Account" }, { type: "azure-native:datalakeanalytics/v20191101preview:Account" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Account.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Account = Account;
/** @internal */
Account.__pulumiType = 'azure-native:datalakeanalytics:Account';
