// @pulumi/azure-native/storagesync/serverEndpoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Server Endpoint object.
 *
 * Uses Azure REST API version 2022-09-01. In version 2.x of the Azure Native provider, it used API version 2022-06-01.
 *
 * Other available API versions: 2022-06-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native storagesync [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ServerEndpoint extends pulumi.CustomResource {
    /**
     * Get an existing ServerEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServerEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServerEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerEndpoint.__pulumiType;
    }
    /**
     * Create a ServerEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.storageSyncServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageSyncServiceName'");
            }
            if ((!args || args.syncGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'syncGroupName'");
            }
            resourceInputs["cloudTiering"] = args ? args.cloudTiering : undefined;
            resourceInputs["friendlyName"] = args ? args.friendlyName : undefined;
            resourceInputs["initialDownloadPolicy"] = args ? args.initialDownloadPolicy : undefined;
            resourceInputs["initialUploadPolicy"] = args ? args.initialUploadPolicy : undefined;
            resourceInputs["localCacheMode"] = args ? args.localCacheMode : undefined;
            resourceInputs["offlineDataTransfer"] = args ? args.offlineDataTransfer : undefined;
            resourceInputs["offlineDataTransferShareName"] = args ? args.offlineDataTransferShareName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverEndpointName"] = args ? args.serverEndpointName : undefined;
            resourceInputs["serverLocalPath"] = args ? args.serverLocalPath : undefined;
            resourceInputs["serverResourceId"] = args ? args.serverResourceId : undefined;
            resourceInputs["storageSyncServiceName"] = args ? args.storageSyncServiceName : undefined;
            resourceInputs["syncGroupName"] = args ? args.syncGroupName : undefined;
            resourceInputs["tierFilesOlderThanDays"] = (_a = (args ? args.tierFilesOlderThanDays : undefined)) !== null && _a !== void 0 ? _a : 0;
            resourceInputs["volumeFreeSpacePercent"] = (_b = (args ? args.volumeFreeSpacePercent : undefined)) !== null && _b !== void 0 ? _b : 20;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cloudTieringStatus"] = undefined /*out*/;
            resourceInputs["lastOperationName"] = undefined /*out*/;
            resourceInputs["lastWorkflowId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["offlineDataTransferStorageAccountResourceId"] = undefined /*out*/;
            resourceInputs["offlineDataTransferStorageAccountTenantId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["recallStatus"] = undefined /*out*/;
            resourceInputs["serverEndpointProvisioningStatus"] = undefined /*out*/;
            resourceInputs["serverName"] = undefined /*out*/;
            resourceInputs["syncStatus"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cloudTiering"] = undefined /*out*/;
            resourceInputs["cloudTieringStatus"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["initialDownloadPolicy"] = undefined /*out*/;
            resourceInputs["initialUploadPolicy"] = undefined /*out*/;
            resourceInputs["lastOperationName"] = undefined /*out*/;
            resourceInputs["lastWorkflowId"] = undefined /*out*/;
            resourceInputs["localCacheMode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["offlineDataTransfer"] = undefined /*out*/;
            resourceInputs["offlineDataTransferShareName"] = undefined /*out*/;
            resourceInputs["offlineDataTransferStorageAccountResourceId"] = undefined /*out*/;
            resourceInputs["offlineDataTransferStorageAccountTenantId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["recallStatus"] = undefined /*out*/;
            resourceInputs["serverEndpointProvisioningStatus"] = undefined /*out*/;
            resourceInputs["serverLocalPath"] = undefined /*out*/;
            resourceInputs["serverName"] = undefined /*out*/;
            resourceInputs["serverResourceId"] = undefined /*out*/;
            resourceInputs["syncStatus"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tierFilesOlderThanDays"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["volumeFreeSpacePercent"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storagesync/v20170605preview:ServerEndpoint" }, { type: "azure-native:storagesync/v20180402:ServerEndpoint" }, { type: "azure-native:storagesync/v20180701:ServerEndpoint" }, { type: "azure-native:storagesync/v20181001:ServerEndpoint" }, { type: "azure-native:storagesync/v20190201:ServerEndpoint" }, { type: "azure-native:storagesync/v20190301:ServerEndpoint" }, { type: "azure-native:storagesync/v20190601:ServerEndpoint" }, { type: "azure-native:storagesync/v20191001:ServerEndpoint" }, { type: "azure-native:storagesync/v20200301:ServerEndpoint" }, { type: "azure-native:storagesync/v20200901:ServerEndpoint" }, { type: "azure-native:storagesync/v20220601:ServerEndpoint" }, { type: "azure-native:storagesync/v20220901:ServerEndpoint" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerEndpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServerEndpoint = ServerEndpoint;
/** @internal */
ServerEndpoint.__pulumiType = 'azure-native:storagesync:ServerEndpoint';
