// @pulumi/azure-native/hybridnetwork/vendorSkus.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VendorSkus = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Sku sub resource.
 *
 * Uses Azure REST API version 2022-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2022-01-01-preview.
 */
class VendorSkus extends pulumi.CustomResource {
    /**
     * Get an existing VendorSkus resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VendorSkus(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VendorSkus.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VendorSkus.__pulumiType;
    }
    /**
     * Create a VendorSkus resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.vendorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vendorName'");
            }
            resourceInputs["deploymentMode"] = args ? args.deploymentMode : undefined;
            resourceInputs["managedApplicationParameters"] = args ? args.managedApplicationParameters : undefined;
            resourceInputs["managedApplicationTemplate"] = args ? args.managedApplicationTemplate : undefined;
            resourceInputs["networkFunctionTemplate"] = args ? args.networkFunctionTemplate : undefined;
            resourceInputs["networkFunctionType"] = args ? args.networkFunctionType : undefined;
            resourceInputs["preview"] = args ? args.preview : undefined;
            resourceInputs["skuName"] = args ? args.skuName : undefined;
            resourceInputs["skuType"] = args ? args.skuType : undefined;
            resourceInputs["vendorName"] = args ? args.vendorName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deploymentMode"] = undefined /*out*/;
            resourceInputs["managedApplicationParameters"] = undefined /*out*/;
            resourceInputs["managedApplicationTemplate"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkFunctionTemplate"] = undefined /*out*/;
            resourceInputs["networkFunctionType"] = undefined /*out*/;
            resourceInputs["preview"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["skuType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:hybridnetwork/v20200101preview:VendorSkus" }, { type: "azure-native:hybridnetwork/v20210501:VendorSkus" }, { type: "azure-native:hybridnetwork/v20220101preview:VendorSkus" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VendorSkus.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VendorSkus = VendorSkus;
/** @internal */
VendorSkus.__pulumiType = 'azure-native:hybridnetwork:VendorSkus';
