// @pulumi/azure-native/hybridnetwork/vendorSkuPreview.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VendorSkuPreview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Customer subscription which can use a sku.
 *
 * Uses Azure REST API version 2022-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2022-01-01-preview.
 */
class VendorSkuPreview extends pulumi.CustomResource {
    /**
     * Get an existing VendorSkuPreview resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VendorSkuPreview(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VendorSkuPreview.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VendorSkuPreview.__pulumiType;
    }
    /**
     * Create a VendorSkuPreview resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.skuName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'skuName'");
            }
            if ((!args || args.vendorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vendorName'");
            }
            resourceInputs["previewSubscription"] = args ? args.previewSubscription : undefined;
            resourceInputs["skuName"] = args ? args.skuName : undefined;
            resourceInputs["vendorName"] = args ? args.vendorName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:hybridnetwork/v20200101preview:VendorSkuPreview" }, { type: "azure-native:hybridnetwork/v20210501:VendorSkuPreview" }, { type: "azure-native:hybridnetwork/v20220101preview:VendorSkuPreview" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VendorSkuPreview.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VendorSkuPreview = VendorSkuPreview;
/** @internal */
VendorSkuPreview.__pulumiType = 'azure-native:hybridnetwork:VendorSkuPreview';
