// @pulumi/azure-native/types/enums/datafactory/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoDbAuthenticationType = exports.MappingType = exports.ManagedVirtualNetworkReferenceType = exports.IntegrationRuntimeType = exports.IntegrationRuntimeSsisCatalogPricingTier = exports.IntegrationRuntimeLicenseType = exports.IntegrationRuntimeEntityReferenceType = exports.IntegrationRuntimeEdition = exports.ImpalaThriftTransportProtocol = exports.ImpalaAuthenticationType = exports.HttpAuthenticationType = exports.HiveThriftTransportProtocol = exports.HiveServerType = exports.HiveAuthenticationType = exports.HDInsightActivityDebugInfoOption = exports.HBaseAuthenticationType = exports.GreenplumAuthenticationType = exports.GoogleBigQueryV2AuthenticationType = exports.GoogleBigQueryAuthenticationType = exports.GoogleAdWordsAuthenticationType = exports.GlobalParameterType = exports.FtpAuthenticationType = exports.FrequencyType = exports.FactoryIdentityType = exports.ExpressionV2Type = exports.DynamicsSinkWriteBehavior = exports.DependencyCondition = exports.Db2AuthenticationType = exports.DaysOfWeek = exports.DayOfWeek = exports.DataFlowReferenceType = exports.DataFlowComputeType = exports.CredentialReferenceType = exports.CosmosDbConnectionMode = exports.ConnectionType = exports.ConfigurationType = exports.CassandraSourceReadConsistencyLevels = exports.BlobEventTypes = exports.BigDataPoolReferenceType = exports.AzureStorageAuthenticationType = exports.AzureSqlMIAuthenticationType = exports.AzureSqlDatabaseAuthenticationType = exports.AzureSqlDWAuthenticationType = exports.AzureSearchIndexWriteBehaviorType = exports.AzurePostgreSqlWriteMethodEnum = exports.AzureFunctionActivityMethod = exports.AmazonRdsForSqlAuthenticationType = exports.AmazonRdsForOracleAuthenticationType = exports.ActivityState = exports.ActivityOnInactiveMarkAs = void 0;
exports.ZendeskAuthenticationType = exports.WebHookActivityMethod = exports.WebAuthenticationType = exports.WebActivityMethod = exports.VariableType = exports.ValueType = exports.Type = exports.TumblingWindowFrequency = exports.TriggerReferenceType = exports.TeradataAuthenticationType = exports.TeamDeskAuthenticationType = exports.SybaseAuthenticationType = exports.SsisPackageLocationType = exports.SsisLogLocationType = exports.SqlServerAuthenticationType = exports.SqlAlwaysEncryptedAkvAuthType = exports.SparkThriftTransportProtocol = exports.SparkServerType = exports.SparkJobReferenceType = exports.SparkConfigurationReferenceType = exports.SparkAuthenticationType = exports.SnowflakeAuthenticationType = exports.SftpAuthenticationType = exports.ServiceNowV2AuthenticationType = exports.ServiceNowAuthenticationType = exports.ScriptActivityParameterType = exports.ScriptActivityParameterDirection = exports.ScriptActivityLogDestination = exports.SapHanaAuthenticationType = exports.SapCloudForCustomerSinkWriteBehavior = exports.SalesforceV2SinkWriteBehavior = exports.SalesforceSinkWriteBehavior = exports.RestServiceAuthenticationType = exports.RecurrenceFrequency = exports.PublicNetworkAccess = exports.PrestoAuthenticationType = exports.PolybaseSettingsRejectType = exports.PhoenixAuthenticationType = exports.ParameterType = exports.OracleAuthenticationType = exports.ODataAuthenticationType = exports.ODataAadServicePrincipalCredentialType = exports.NotebookReferenceType = exports.NotebookParameterType = void 0;
exports.ActivityOnInactiveMarkAs = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Skipped: "Skipped",
};
exports.ActivityState = {
    Active: "Active",
    Inactive: "Inactive",
};
exports.AmazonRdsForOracleAuthenticationType = {
    Basic: "Basic",
};
exports.AmazonRdsForSqlAuthenticationType = {
    SQL: "SQL",
    Windows: "Windows",
};
exports.AzureFunctionActivityMethod = {
    GET: "GET",
    POST: "POST",
    PUT: "PUT",
    DELETE: "DELETE",
    OPTIONS: "OPTIONS",
    HEAD: "HEAD",
    TRACE: "TRACE",
};
exports.AzurePostgreSqlWriteMethodEnum = {
    BulkInsert: "BulkInsert",
    CopyCommand: "CopyCommand",
    Upsert: "Upsert",
};
exports.AzureSearchIndexWriteBehaviorType = {
    Merge: "Merge",
    Upload: "Upload",
};
exports.AzureSqlDWAuthenticationType = {
    SQL: "SQL",
    ServicePrincipal: "ServicePrincipal",
    SystemAssignedManagedIdentity: "SystemAssignedManagedIdentity",
    UserAssignedManagedIdentity: "UserAssignedManagedIdentity",
};
exports.AzureSqlDatabaseAuthenticationType = {
    SQL: "SQL",
    ServicePrincipal: "ServicePrincipal",
    SystemAssignedManagedIdentity: "SystemAssignedManagedIdentity",
    UserAssignedManagedIdentity: "UserAssignedManagedIdentity",
};
exports.AzureSqlMIAuthenticationType = {
    SQL: "SQL",
    ServicePrincipal: "ServicePrincipal",
    SystemAssignedManagedIdentity: "SystemAssignedManagedIdentity",
    UserAssignedManagedIdentity: "UserAssignedManagedIdentity",
};
exports.AzureStorageAuthenticationType = {
    Anonymous: "Anonymous",
    AccountKey: "AccountKey",
    SasUri: "SasUri",
    ServicePrincipal: "ServicePrincipal",
    Msi: "Msi",
};
exports.BigDataPoolReferenceType = {
    BigDataPoolReference: "BigDataPoolReference",
};
exports.BlobEventTypes = {
    Microsoft_Storage_BlobCreated: "Microsoft.Storage.BlobCreated",
    Microsoft_Storage_BlobDeleted: "Microsoft.Storage.BlobDeleted",
};
exports.CassandraSourceReadConsistencyLevels = {
    ALL: "ALL",
    EACH_QUORUM: "EACH_QUORUM",
    QUORUM: "QUORUM",
    LOCAL_QUORUM: "LOCAL_QUORUM",
    ONE: "ONE",
    TWO: "TWO",
    THREE: "THREE",
    LOCAL_ONE: "LOCAL_ONE",
    SERIAL: "SERIAL",
    LOCAL_SERIAL: "LOCAL_SERIAL",
};
exports.ConfigurationType = {
    Default: "Default",
    Customized: "Customized",
    Artifact: "Artifact",
};
exports.ConnectionType = {
    Linkedservicetype: "linkedservicetype",
};
exports.CosmosDbConnectionMode = {
    Gateway: "Gateway",
    Direct: "Direct",
};
exports.CredentialReferenceType = {
    CredentialReference: "CredentialReference",
};
exports.DataFlowComputeType = {
    General: "General",
    MemoryOptimized: "MemoryOptimized",
    ComputeOptimized: "ComputeOptimized",
};
exports.DataFlowReferenceType = {
    DataFlowReference: "DataFlowReference",
};
exports.DayOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.DaysOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.Db2AuthenticationType = {
    Basic: "Basic",
};
exports.DependencyCondition = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Skipped: "Skipped",
    Completed: "Completed",
};
exports.DynamicsSinkWriteBehavior = {
    Upsert: "Upsert",
};
exports.ExpressionV2Type = {
    Constant: "Constant",
    Field: "Field",
    Unary: "Unary",
    Binary: "Binary",
    NAry: "NAry",
};
exports.FactoryIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.FrequencyType = {
    Hour: "Hour",
    Minute: "Minute",
    Second: "Second",
};
exports.FtpAuthenticationType = {
    Basic: "Basic",
    Anonymous: "Anonymous",
};
exports.GlobalParameterType = {
    Object: "Object",
    String: "String",
    Int: "Int",
    Float: "Float",
    Bool: "Bool",
    Array: "Array",
};
exports.GoogleAdWordsAuthenticationType = {
    ServiceAuthentication: "ServiceAuthentication",
    UserAuthentication: "UserAuthentication",
};
exports.GoogleBigQueryAuthenticationType = {
    ServiceAuthentication: "ServiceAuthentication",
    UserAuthentication: "UserAuthentication",
};
exports.GoogleBigQueryV2AuthenticationType = {
    ServiceAuthentication: "ServiceAuthentication",
    UserAuthentication: "UserAuthentication",
};
exports.GreenplumAuthenticationType = {
    Basic: "Basic",
};
exports.HBaseAuthenticationType = {
    Anonymous: "Anonymous",
    Basic: "Basic",
};
exports.HDInsightActivityDebugInfoOption = {
    None: "None",
    Always: "Always",
    Failure: "Failure",
};
exports.HiveAuthenticationType = {
    Anonymous: "Anonymous",
    Username: "Username",
    UsernameAndPassword: "UsernameAndPassword",
    WindowsAzureHDInsightService: "WindowsAzureHDInsightService",
};
exports.HiveServerType = {
    HiveServer1: "HiveServer1",
    HiveServer2: "HiveServer2",
    HiveThriftServer: "HiveThriftServer",
};
exports.HiveThriftTransportProtocol = {
    Binary: "Binary",
    SASL: "SASL",
    HTTP_: "HTTP ",
};
exports.HttpAuthenticationType = {
    Basic: "Basic",
    Anonymous: "Anonymous",
    Digest: "Digest",
    Windows: "Windows",
    ClientCertificate: "ClientCertificate",
};
exports.ImpalaAuthenticationType = {
    Anonymous: "Anonymous",
    SASLUsername: "SASLUsername",
    UsernameAndPassword: "UsernameAndPassword",
};
exports.ImpalaThriftTransportProtocol = {
    Binary: "Binary",
    HTTP: "HTTP",
};
exports.IntegrationRuntimeEdition = {
    Standard: "Standard",
    Enterprise: "Enterprise",
};
exports.IntegrationRuntimeEntityReferenceType = {
    IntegrationRuntimeReference: "IntegrationRuntimeReference",
    LinkedServiceReference: "LinkedServiceReference",
};
exports.IntegrationRuntimeLicenseType = {
    BasePrice: "BasePrice",
    LicenseIncluded: "LicenseIncluded",
};
exports.IntegrationRuntimeSsisCatalogPricingTier = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
    PremiumRS: "PremiumRS",
};
exports.IntegrationRuntimeType = {
    Managed: "Managed",
    SelfHosted: "SelfHosted",
};
exports.ManagedVirtualNetworkReferenceType = {
    ManagedVirtualNetworkReference: "ManagedVirtualNetworkReference",
};
exports.MappingType = {
    Direct: "Direct",
    Derived: "Derived",
    Aggregate: "Aggregate",
};
exports.MongoDbAuthenticationType = {
    Basic: "Basic",
    Anonymous: "Anonymous",
};
exports.NotebookParameterType = {
    String: "string",
    Int: "int",
    Float: "float",
    Bool: "bool",
};
exports.NotebookReferenceType = {
    NotebookReference: "NotebookReference",
};
exports.ODataAadServicePrincipalCredentialType = {
    ServicePrincipalKey: "ServicePrincipalKey",
    ServicePrincipalCert: "ServicePrincipalCert",
};
exports.ODataAuthenticationType = {
    Basic: "Basic",
    Anonymous: "Anonymous",
    Windows: "Windows",
    AadServicePrincipal: "AadServicePrincipal",
    ManagedServiceIdentity: "ManagedServiceIdentity",
};
exports.OracleAuthenticationType = {
    Basic: "Basic",
};
exports.ParameterType = {
    Object: "Object",
    String: "String",
    Int: "Int",
    Float: "Float",
    Bool: "Bool",
    Array: "Array",
    SecureString: "SecureString",
};
exports.PhoenixAuthenticationType = {
    Anonymous: "Anonymous",
    UsernameAndPassword: "UsernameAndPassword",
    WindowsAzureHDInsightService: "WindowsAzureHDInsightService",
};
exports.PolybaseSettingsRejectType = {
    Value: "value",
    Percentage: "percentage",
};
exports.PrestoAuthenticationType = {
    Anonymous: "Anonymous",
    LDAP: "LDAP",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RecurrenceFrequency = {
    NotSpecified: "NotSpecified",
    Minute: "Minute",
    Hour: "Hour",
    Day: "Day",
    Week: "Week",
    Month: "Month",
    Year: "Year",
};
exports.RestServiceAuthenticationType = {
    Anonymous: "Anonymous",
    Basic: "Basic",
    AadServicePrincipal: "AadServicePrincipal",
    ManagedServiceIdentity: "ManagedServiceIdentity",
    OAuth2ClientCredential: "OAuth2ClientCredential",
};
exports.SalesforceSinkWriteBehavior = {
    Insert: "Insert",
    Upsert: "Upsert",
};
exports.SalesforceV2SinkWriteBehavior = {
    Insert: "Insert",
    Upsert: "Upsert",
};
exports.SapCloudForCustomerSinkWriteBehavior = {
    Insert: "Insert",
    Update: "Update",
};
exports.SapHanaAuthenticationType = {
    Basic: "Basic",
    Windows: "Windows",
};
exports.ScriptActivityLogDestination = {
    ActivityOutput: "ActivityOutput",
    ExternalStore: "ExternalStore",
};
exports.ScriptActivityParameterDirection = {
    ValueInput: "Input",
    ValueOutput: "Output",
    ValueInputOutput: "InputOutput",
};
exports.ScriptActivityParameterType = {
    Boolean: "Boolean",
    DateTime: "DateTime",
    DateTimeOffset: "DateTimeOffset",
    Decimal: "Decimal",
    Double: "Double",
    Guid: "Guid",
    Int16: "Int16",
    Int32: "Int32",
    Int64: "Int64",
    Single: "Single",
    String: "String",
    Timespan: "Timespan",
};
exports.ServiceNowAuthenticationType = {
    Basic: "Basic",
    OAuth2: "OAuth2",
};
exports.ServiceNowV2AuthenticationType = {
    Basic: "Basic",
    OAuth2: "OAuth2",
};
exports.SftpAuthenticationType = {
    Basic: "Basic",
    SshPublicKey: "SshPublicKey",
    MultiFactor: "MultiFactor",
};
exports.SnowflakeAuthenticationType = {
    Basic: "Basic",
    KeyPair: "KeyPair",
    AADServicePrincipal: "AADServicePrincipal",
};
exports.SparkAuthenticationType = {
    Anonymous: "Anonymous",
    Username: "Username",
    UsernameAndPassword: "UsernameAndPassword",
    WindowsAzureHDInsightService: "WindowsAzureHDInsightService",
};
exports.SparkConfigurationReferenceType = {
    SparkConfigurationReference: "SparkConfigurationReference",
};
exports.SparkJobReferenceType = {
    SparkJobDefinitionReference: "SparkJobDefinitionReference",
};
exports.SparkServerType = {
    SharkServer: "SharkServer",
    SharkServer2: "SharkServer2",
    SparkThriftServer: "SparkThriftServer",
};
exports.SparkThriftTransportProtocol = {
    Binary: "Binary",
    SASL: "SASL",
    HTTP_: "HTTP ",
};
exports.SqlAlwaysEncryptedAkvAuthType = {
    ServicePrincipal: "ServicePrincipal",
    ManagedIdentity: "ManagedIdentity",
    UserAssignedManagedIdentity: "UserAssignedManagedIdentity",
};
exports.SqlServerAuthenticationType = {
    SQL: "SQL",
    Windows: "Windows",
    UserAssignedManagedIdentity: "UserAssignedManagedIdentity",
};
exports.SsisLogLocationType = {
    File: "File",
};
exports.SsisPackageLocationType = {
    SSISDB: "SSISDB",
    File: "File",
    InlinePackage: "InlinePackage",
    PackageStore: "PackageStore",
};
exports.SybaseAuthenticationType = {
    Basic: "Basic",
    Windows: "Windows",
};
exports.TeamDeskAuthenticationType = {
    Basic: "Basic",
    Token: "Token",
};
exports.TeradataAuthenticationType = {
    Basic: "Basic",
    Windows: "Windows",
};
exports.TriggerReferenceType = {
    TriggerReference: "TriggerReference",
};
exports.TumblingWindowFrequency = {
    Minute: "Minute",
    Hour: "Hour",
    Month: "Month",
};
exports.Type = {
    LinkedServiceReference: "LinkedServiceReference",
};
exports.ValueType = {
    Actual: "actual",
    Display: "display",
};
exports.VariableType = {
    String: "String",
    Bool: "Bool",
    Array: "Array",
};
exports.WebActivityMethod = {
    GET: "GET",
    POST: "POST",
    PUT: "PUT",
    DELETE: "DELETE",
};
exports.WebAuthenticationType = {
    Basic: "Basic",
    Anonymous: "Anonymous",
    ClientCertificate: "ClientCertificate",
};
exports.WebHookActivityMethod = {
    POST: "POST",
};
exports.ZendeskAuthenticationType = {
    Basic: "Basic",
    Token: "Token",
};
