// @pulumi/azure/dynatrace/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagRules = exports.Monitor = exports.getMonitorOutput = exports.getMonitor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getMonitor = null;
exports.getMonitorOutput = null;
utilities.lazyLoad(exports, ["getMonitor", "getMonitorOutput"], () => require("./getMonitor"));
exports.Monitor = null;
utilities.lazyLoad(exports, ["Monitor"], () => require("./monitor"));
exports.TagRules = null;
utilities.lazyLoad(exports, ["TagRules"], () => require("./tagRules"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure:dynatrace/monitor:Monitor":
                return new exports.Monitor(name, undefined, { urn });
            case "azure:dynatrace/tagRules:TagRules":
                return new exports.TagRules(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure", "dynatrace/monitor", _module);
pulumi.runtime.registerResourceModule("azure", "dynatrace/tagRules", _module);
