// @pulumi/azure-native/monitor/actionGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An action group resource.
 *
 * Uses Azure REST API version 2024-10-01-preview.
 *
 * Other available API versions: 2018-03-01, 2022-06-01, 2023-09-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native monitor [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ActionGroup extends pulumi.CustomResource {
    /**
     * Get an existing ActionGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ActionGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ActionGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ActionGroup.__pulumiType;
    }
    /**
     * Create a ActionGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.enabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if ((!args || args.groupShortName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupShortName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["actionGroupName"] = args ? args.actionGroupName : undefined;
            resourceInputs["armRoleReceivers"] = args ? args.armRoleReceivers : undefined;
            resourceInputs["automationRunbookReceivers"] = args ? args.automationRunbookReceivers : undefined;
            resourceInputs["azureAppPushReceivers"] = args ? args.azureAppPushReceivers : undefined;
            resourceInputs["azureFunctionReceivers"] = args ? args.azureFunctionReceivers : undefined;
            resourceInputs["emailReceivers"] = args ? args.emailReceivers : undefined;
            resourceInputs["enabled"] = (_a = (args ? args.enabled : undefined)) !== null && _a !== void 0 ? _a : true;
            resourceInputs["eventHubReceivers"] = args ? args.eventHubReceivers : undefined;
            resourceInputs["groupShortName"] = args ? args.groupShortName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["incidentReceivers"] = args ? args.incidentReceivers : undefined;
            resourceInputs["itsmReceivers"] = args ? args.itsmReceivers : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["logicAppReceivers"] = args ? args.logicAppReceivers : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["smsReceivers"] = args ? args.smsReceivers : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["voiceReceivers"] = args ? args.voiceReceivers : undefined;
            resourceInputs["webhookReceivers"] = args ? args.webhookReceivers : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["armRoleReceivers"] = undefined /*out*/;
            resourceInputs["automationRunbookReceivers"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["azureAppPushReceivers"] = undefined /*out*/;
            resourceInputs["azureFunctionReceivers"] = undefined /*out*/;
            resourceInputs["emailReceivers"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["eventHubReceivers"] = undefined /*out*/;
            resourceInputs["groupShortName"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["incidentReceivers"] = undefined /*out*/;
            resourceInputs["itsmReceivers"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["logicAppReceivers"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["smsReceivers"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["voiceReceivers"] = undefined /*out*/;
            resourceInputs["webhookReceivers"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:insights/v20230101:ActionGroup" }, { type: "azure-native:insights/v20230901preview:ActionGroup" }, { type: "azure-native:insights/v20241001preview:ActionGroup" }, { type: "azure-native:insights:ActionGroup" }, { type: "azure-native:monitor/v20170401:ActionGroup" }, { type: "azure-native:monitor/v20180301:ActionGroup" }, { type: "azure-native:monitor/v20180901:ActionGroup" }, { type: "azure-native:monitor/v20190301:ActionGroup" }, { type: "azure-native:monitor/v20190601:ActionGroup" }, { type: "azure-native:monitor/v20210901:ActionGroup" }, { type: "azure-native:monitor/v20220401:ActionGroup" }, { type: "azure-native:monitor/v20220601:ActionGroup" }, { type: "azure-native:monitor/v20230101:ActionGroup" }, { type: "azure-native:monitor/v20230901preview:ActionGroup" }, { type: "azure-native:monitor/v20241001preview:ActionGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ActionGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ActionGroup = ActionGroup;
/** @internal */
ActionGroup.__pulumiType = 'azure-native:monitor:ActionGroup';
