// @pulumi/azure-native/monitor/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkScopedResource = exports.PrivateLinkScope = exports.PrivateEndpointConnection = exports.PipelineGroup = exports.MetricAlert = exports.ManagementGroupDiagnosticSetting = exports.listDiagnosticSettingsCategoryOutput = exports.listDiagnosticSettingsCategory = exports.getTenantActionGroupOutput = exports.getTenantActionGroup = exports.getSubscriptionDiagnosticSettingOutput = exports.getSubscriptionDiagnosticSetting = exports.getScheduledQueryRuleOutput = exports.getScheduledQueryRule = exports.getPrivateLinkScopedResourceOutput = exports.getPrivateLinkScopedResource = exports.getPrivateLinkScopeOutput = exports.getPrivateLinkScope = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getPipelineGroupOutput = exports.getPipelineGroup = exports.getMetricAlertOutput = exports.getMetricAlert = exports.getManagementGroupDiagnosticSettingOutput = exports.getManagementGroupDiagnosticSetting = exports.getDiagnosticSettingOutput = exports.getDiagnosticSetting = exports.getDataCollectionRuleAssociationOutput = exports.getDataCollectionRuleAssociation = exports.getDataCollectionRuleOutput = exports.getDataCollectionRule = exports.getDataCollectionEndpointOutput = exports.getDataCollectionEndpoint = exports.getAzureMonitorWorkspaceOutput = exports.getAzureMonitorWorkspace = exports.getAutoscaleSettingOutput = exports.getAutoscaleSetting = exports.getActivityLogAlertOutput = exports.getActivityLogAlert = exports.getActionGroupOutput = exports.getActionGroup = exports.DiagnosticSetting = exports.DataCollectionRuleAssociation = exports.DataCollectionRule = exports.DataCollectionEndpoint = exports.AzureMonitorWorkspace = exports.AutoscaleSetting = exports.ActivityLogAlert = exports.ActionGroup = void 0;
exports.TenantActionGroup = exports.SubscriptionDiagnosticSetting = exports.ScheduledQueryRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ActionGroup = null;
utilities.lazyLoad(exports, ["ActionGroup"], () => require("./actionGroup"));
exports.ActivityLogAlert = null;
utilities.lazyLoad(exports, ["ActivityLogAlert"], () => require("./activityLogAlert"));
exports.AutoscaleSetting = null;
utilities.lazyLoad(exports, ["AutoscaleSetting"], () => require("./autoscaleSetting"));
exports.AzureMonitorWorkspace = null;
utilities.lazyLoad(exports, ["AzureMonitorWorkspace"], () => require("./azureMonitorWorkspace"));
exports.DataCollectionEndpoint = null;
utilities.lazyLoad(exports, ["DataCollectionEndpoint"], () => require("./dataCollectionEndpoint"));
exports.DataCollectionRule = null;
utilities.lazyLoad(exports, ["DataCollectionRule"], () => require("./dataCollectionRule"));
exports.DataCollectionRuleAssociation = null;
utilities.lazyLoad(exports, ["DataCollectionRuleAssociation"], () => require("./dataCollectionRuleAssociation"));
exports.DiagnosticSetting = null;
utilities.lazyLoad(exports, ["DiagnosticSetting"], () => require("./diagnosticSetting"));
exports.getActionGroup = null;
exports.getActionGroupOutput = null;
utilities.lazyLoad(exports, ["getActionGroup", "getActionGroupOutput"], () => require("./getActionGroup"));
exports.getActivityLogAlert = null;
exports.getActivityLogAlertOutput = null;
utilities.lazyLoad(exports, ["getActivityLogAlert", "getActivityLogAlertOutput"], () => require("./getActivityLogAlert"));
exports.getAutoscaleSetting = null;
exports.getAutoscaleSettingOutput = null;
utilities.lazyLoad(exports, ["getAutoscaleSetting", "getAutoscaleSettingOutput"], () => require("./getAutoscaleSetting"));
exports.getAzureMonitorWorkspace = null;
exports.getAzureMonitorWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getAzureMonitorWorkspace", "getAzureMonitorWorkspaceOutput"], () => require("./getAzureMonitorWorkspace"));
exports.getDataCollectionEndpoint = null;
exports.getDataCollectionEndpointOutput = null;
utilities.lazyLoad(exports, ["getDataCollectionEndpoint", "getDataCollectionEndpointOutput"], () => require("./getDataCollectionEndpoint"));
exports.getDataCollectionRule = null;
exports.getDataCollectionRuleOutput = null;
utilities.lazyLoad(exports, ["getDataCollectionRule", "getDataCollectionRuleOutput"], () => require("./getDataCollectionRule"));
exports.getDataCollectionRuleAssociation = null;
exports.getDataCollectionRuleAssociationOutput = null;
utilities.lazyLoad(exports, ["getDataCollectionRuleAssociation", "getDataCollectionRuleAssociationOutput"], () => require("./getDataCollectionRuleAssociation"));
exports.getDiagnosticSetting = null;
exports.getDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getDiagnosticSetting", "getDiagnosticSettingOutput"], () => require("./getDiagnosticSetting"));
exports.getManagementGroupDiagnosticSetting = null;
exports.getManagementGroupDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getManagementGroupDiagnosticSetting", "getManagementGroupDiagnosticSettingOutput"], () => require("./getManagementGroupDiagnosticSetting"));
exports.getMetricAlert = null;
exports.getMetricAlertOutput = null;
utilities.lazyLoad(exports, ["getMetricAlert", "getMetricAlertOutput"], () => require("./getMetricAlert"));
exports.getPipelineGroup = null;
exports.getPipelineGroupOutput = null;
utilities.lazyLoad(exports, ["getPipelineGroup", "getPipelineGroupOutput"], () => require("./getPipelineGroup"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getPrivateLinkScope = null;
exports.getPrivateLinkScopeOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkScope", "getPrivateLinkScopeOutput"], () => require("./getPrivateLinkScope"));
exports.getPrivateLinkScopedResource = null;
exports.getPrivateLinkScopedResourceOutput = null;
utilities.lazyLoad(exports, ["getPrivateLinkScopedResource", "getPrivateLinkScopedResourceOutput"], () => require("./getPrivateLinkScopedResource"));
exports.getScheduledQueryRule = null;
exports.getScheduledQueryRuleOutput = null;
utilities.lazyLoad(exports, ["getScheduledQueryRule", "getScheduledQueryRuleOutput"], () => require("./getScheduledQueryRule"));
exports.getSubscriptionDiagnosticSetting = null;
exports.getSubscriptionDiagnosticSettingOutput = null;
utilities.lazyLoad(exports, ["getSubscriptionDiagnosticSetting", "getSubscriptionDiagnosticSettingOutput"], () => require("./getSubscriptionDiagnosticSetting"));
exports.getTenantActionGroup = null;
exports.getTenantActionGroupOutput = null;
utilities.lazyLoad(exports, ["getTenantActionGroup", "getTenantActionGroupOutput"], () => require("./getTenantActionGroup"));
exports.listDiagnosticSettingsCategory = null;
exports.listDiagnosticSettingsCategoryOutput = null;
utilities.lazyLoad(exports, ["listDiagnosticSettingsCategory", "listDiagnosticSettingsCategoryOutput"], () => require("./listDiagnosticSettingsCategory"));
exports.ManagementGroupDiagnosticSetting = null;
utilities.lazyLoad(exports, ["ManagementGroupDiagnosticSetting"], () => require("./managementGroupDiagnosticSetting"));
exports.MetricAlert = null;
utilities.lazyLoad(exports, ["MetricAlert"], () => require("./metricAlert"));
exports.PipelineGroup = null;
utilities.lazyLoad(exports, ["PipelineGroup"], () => require("./pipelineGroup"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.PrivateLinkScope = null;
utilities.lazyLoad(exports, ["PrivateLinkScope"], () => require("./privateLinkScope"));
exports.PrivateLinkScopedResource = null;
utilities.lazyLoad(exports, ["PrivateLinkScopedResource"], () => require("./privateLinkScopedResource"));
exports.ScheduledQueryRule = null;
utilities.lazyLoad(exports, ["ScheduledQueryRule"], () => require("./scheduledQueryRule"));
exports.SubscriptionDiagnosticSetting = null;
utilities.lazyLoad(exports, ["SubscriptionDiagnosticSetting"], () => require("./subscriptionDiagnosticSetting"));
exports.TenantActionGroup = null;
utilities.lazyLoad(exports, ["TenantActionGroup"], () => require("./tenantActionGroup"));
// Export enums:
__exportStar(require("../types/enums/monitor"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:monitor:ActionGroup":
                return new exports.ActionGroup(name, undefined, { urn });
            case "azure-native:monitor:ActivityLogAlert":
                return new exports.ActivityLogAlert(name, undefined, { urn });
            case "azure-native:monitor:AutoscaleSetting":
                return new exports.AutoscaleSetting(name, undefined, { urn });
            case "azure-native:monitor:AzureMonitorWorkspace":
                return new exports.AzureMonitorWorkspace(name, undefined, { urn });
            case "azure-native:monitor:DataCollectionEndpoint":
                return new exports.DataCollectionEndpoint(name, undefined, { urn });
            case "azure-native:monitor:DataCollectionRule":
                return new exports.DataCollectionRule(name, undefined, { urn });
            case "azure-native:monitor:DataCollectionRuleAssociation":
                return new exports.DataCollectionRuleAssociation(name, undefined, { urn });
            case "azure-native:monitor:DiagnosticSetting":
                return new exports.DiagnosticSetting(name, undefined, { urn });
            case "azure-native:monitor:ManagementGroupDiagnosticSetting":
                return new exports.ManagementGroupDiagnosticSetting(name, undefined, { urn });
            case "azure-native:monitor:MetricAlert":
                return new exports.MetricAlert(name, undefined, { urn });
            case "azure-native:monitor:PipelineGroup":
                return new exports.PipelineGroup(name, undefined, { urn });
            case "azure-native:monitor:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:monitor:PrivateLinkScope":
                return new exports.PrivateLinkScope(name, undefined, { urn });
            case "azure-native:monitor:PrivateLinkScopedResource":
                return new exports.PrivateLinkScopedResource(name, undefined, { urn });
            case "azure-native:monitor:ScheduledQueryRule":
                return new exports.ScheduledQueryRule(name, undefined, { urn });
            case "azure-native:monitor:SubscriptionDiagnosticSetting":
                return new exports.SubscriptionDiagnosticSetting(name, undefined, { urn });
            case "azure-native:monitor:TenantActionGroup":
                return new exports.TenantActionGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "monitor", _module);
