// @pulumi/azure-native/iothub/certificate.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Certificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The X509 Certificate.
 *
 * Uses Azure REST API version 2023-06-30.
 *
 * Other available API versions: 2017-07-01, 2018-01-22, 2018-04-01, 2018-12-01-preview, 2019-03-22, 2019-03-22-preview, 2019-07-01-preview, 2019-11-04, 2020-03-01, 2020-04-01, 2020-06-15, 2020-07-10-preview, 2020-08-01, 2020-08-31, 2020-08-31-preview, 2021-02-01-preview, 2021-03-03-preview, 2021-03-31, 2021-07-01, 2021-07-01-preview, 2021-07-02, 2021-07-02-preview, 2022-04-30-preview, 2022-11-15-preview, 2023-06-30-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native iothub [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Certificate extends pulumi.CustomResource {
    /**
     * Get an existing Certificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Certificate(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Certificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Certificate.__pulumiType;
    }
    /**
     * Create a Certificate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["certificateName"] = args ? args.certificateName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devices/v20200401:Certificate" }, { type: "azure-native:devices/v20220430preview:Certificate" }, { type: "azure-native:devices/v20221115preview:Certificate" }, { type: "azure-native:devices/v20230630:Certificate" }, { type: "azure-native:devices/v20230630preview:Certificate" }, { type: "azure-native:devices:Certificate" }, { type: "azure-native:iothub/v20170701:Certificate" }, { type: "azure-native:iothub/v20180122:Certificate" }, { type: "azure-native:iothub/v20180401:Certificate" }, { type: "azure-native:iothub/v20181201preview:Certificate" }, { type: "azure-native:iothub/v20190322:Certificate" }, { type: "azure-native:iothub/v20190322preview:Certificate" }, { type: "azure-native:iothub/v20190701preview:Certificate" }, { type: "azure-native:iothub/v20191104:Certificate" }, { type: "azure-native:iothub/v20200301:Certificate" }, { type: "azure-native:iothub/v20200401:Certificate" }, { type: "azure-native:iothub/v20200615:Certificate" }, { type: "azure-native:iothub/v20200710preview:Certificate" }, { type: "azure-native:iothub/v20200801:Certificate" }, { type: "azure-native:iothub/v20200831:Certificate" }, { type: "azure-native:iothub/v20200831preview:Certificate" }, { type: "azure-native:iothub/v20210201preview:Certificate" }, { type: "azure-native:iothub/v20210303preview:Certificate" }, { type: "azure-native:iothub/v20210331:Certificate" }, { type: "azure-native:iothub/v20210701:Certificate" }, { type: "azure-native:iothub/v20210701preview:Certificate" }, { type: "azure-native:iothub/v20210702:Certificate" }, { type: "azure-native:iothub/v20210702preview:Certificate" }, { type: "azure-native:iothub/v20220430preview:Certificate" }, { type: "azure-native:iothub/v20221115preview:Certificate" }, { type: "azure-native:iothub/v20230630:Certificate" }, { type: "azure-native:iothub/v20230630preview:Certificate" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Certificate.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Certificate = Certificate;
/** @internal */
Certificate.__pulumiType = 'azure-native:iothub:Certificate';
