// @pulumi/azure-native/iothub/iotHubResource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotHubResource = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * The description of the IoT hub.
 *
 * Uses Azure REST API version 2023-06-30.
 *
 * Other available API versions: 2016-02-03, 2017-01-19, 2017-07-01, 2018-01-22, 2018-04-01, 2018-12-01-preview, 2019-03-22, 2019-03-22-preview, 2019-07-01-preview, 2019-11-04, 2020-03-01, 2020-04-01, 2020-06-15, 2020-07-10-preview, 2020-08-01, 2020-08-31, 2020-08-31-preview, 2021-02-01-preview, 2021-03-03-preview, 2021-03-31, 2021-07-01, 2021-07-01-preview, 2021-07-02, 2021-07-02-preview, 2022-04-30-preview, 2022-11-15-preview, 2023-06-30-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native iothub [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class IotHubResource extends pulumi.CustomResource {
    /**
     * Get an existing IotHubResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IotHubResource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotHubResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotHubResource.__pulumiType;
    }
    /**
     * Create a IotHubResource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? (args.properties ? pulumi.output(args.properties).apply(inputs.iothub.iotHubPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devices/v20220430preview:IotHubResource" }, { type: "azure-native:devices/v20221115preview:IotHubResource" }, { type: "azure-native:devices/v20230630:IotHubResource" }, { type: "azure-native:devices/v20230630preview:IotHubResource" }, { type: "azure-native:devices:IotHubResource" }, { type: "azure-native:iothub/v20160203:IotHubResource" }, { type: "azure-native:iothub/v20170119:IotHubResource" }, { type: "azure-native:iothub/v20170701:IotHubResource" }, { type: "azure-native:iothub/v20180122:IotHubResource" }, { type: "azure-native:iothub/v20180401:IotHubResource" }, { type: "azure-native:iothub/v20181201preview:IotHubResource" }, { type: "azure-native:iothub/v20190322:IotHubResource" }, { type: "azure-native:iothub/v20190322preview:IotHubResource" }, { type: "azure-native:iothub/v20190701preview:IotHubResource" }, { type: "azure-native:iothub/v20191104:IotHubResource" }, { type: "azure-native:iothub/v20200301:IotHubResource" }, { type: "azure-native:iothub/v20200401:IotHubResource" }, { type: "azure-native:iothub/v20200615:IotHubResource" }, { type: "azure-native:iothub/v20200710preview:IotHubResource" }, { type: "azure-native:iothub/v20200801:IotHubResource" }, { type: "azure-native:iothub/v20200831:IotHubResource" }, { type: "azure-native:iothub/v20200831preview:IotHubResource" }, { type: "azure-native:iothub/v20210201preview:IotHubResource" }, { type: "azure-native:iothub/v20210303preview:IotHubResource" }, { type: "azure-native:iothub/v20210331:IotHubResource" }, { type: "azure-native:iothub/v20210701:IotHubResource" }, { type: "azure-native:iothub/v20210701preview:IotHubResource" }, { type: "azure-native:iothub/v20210702:IotHubResource" }, { type: "azure-native:iothub/v20210702preview:IotHubResource" }, { type: "azure-native:iothub/v20220430preview:IotHubResource" }, { type: "azure-native:iothub/v20221115preview:IotHubResource" }, { type: "azure-native:iothub/v20230630:IotHubResource" }, { type: "azure-native:iothub/v20230630preview:IotHubResource" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IotHubResource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IotHubResource = IotHubResource;
/** @internal */
IotHubResource.__pulumiType = 'azure-native:iothub:IotHubResource';
