// @pulumi/azure-native/types/enums/containerinstance/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Scheme = exports.ResourceIdentityType = exports.Priority = exports.OperatingSystemTypes = exports.LogAnalyticsLogType = exports.GpuSku = exports.DnsNameLabelReusePolicy = exports.ContainerNetworkProtocol = exports.ContainerGroupSku = exports.ContainerGroupRestartPolicy = exports.ContainerGroupPriority = exports.ContainerGroupNetworkProtocol = exports.ContainerGroupIpAddressType = void 0;
exports.ContainerGroupIpAddressType = {
    Public: "Public",
    Private: "Private",
};
exports.ContainerGroupNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.ContainerGroupPriority = {
    Regular: "Regular",
    Spot: "Spot",
};
exports.ContainerGroupRestartPolicy = {
    Always: "Always",
    OnFailure: "OnFailure",
    Never: "Never",
};
exports.ContainerGroupSku = {
    Standard: "Standard",
    Dedicated: "Dedicated",
    Confidential: "Confidential",
};
exports.ContainerNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.DnsNameLabelReusePolicy = {
    Unsecure: "Unsecure",
    TenantReuse: "TenantReuse",
    SubscriptionReuse: "SubscriptionReuse",
    ResourceGroupReuse: "ResourceGroupReuse",
    Noreuse: "Noreuse",
};
exports.GpuSku = {
    K80: "K80",
    P100: "P100",
    V100: "V100",
};
exports.LogAnalyticsLogType = {
    ContainerInsights: "ContainerInsights",
    ContainerInstanceLogs: "ContainerInstanceLogs",
};
exports.OperatingSystemTypes = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.Priority = {
    Regular: "Regular",
    Spot: "Spot",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.Scheme = {
    Http: "http",
    Https: "https",
};
