// @pulumi/azure-native/customerinsights/link.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Link = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The link resource format.
 *
 * Uses Azure REST API version 2017-04-26. In version 2.x of the Azure Native provider, it used API version 2017-04-26.
 */
class Link extends pulumi.CustomResource {
    /**
     * Get an existing Link resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Link(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Link.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Link.__pulumiType;
    }
    /**
     * Create a Link resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hubName'");
            }
            if ((!args || args.participantPropertyReferences === undefined) && !opts.urn) {
                throw new Error("Missing required property 'participantPropertyReferences'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sourceEntityType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceEntityType'");
            }
            if ((!args || args.sourceEntityTypeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceEntityTypeName'");
            }
            if ((!args || args.targetEntityType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetEntityType'");
            }
            if ((!args || args.targetEntityTypeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetEntityTypeName'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["hubName"] = args ? args.hubName : undefined;
            resourceInputs["linkName"] = args ? args.linkName : undefined;
            resourceInputs["mappings"] = args ? args.mappings : undefined;
            resourceInputs["operationType"] = args ? args.operationType : undefined;
            resourceInputs["participantPropertyReferences"] = args ? args.participantPropertyReferences : undefined;
            resourceInputs["referenceOnly"] = args ? args.referenceOnly : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sourceEntityType"] = args ? args.sourceEntityType : undefined;
            resourceInputs["sourceEntityTypeName"] = args ? args.sourceEntityTypeName : undefined;
            resourceInputs["targetEntityType"] = args ? args.targetEntityType : undefined;
            resourceInputs["targetEntityTypeName"] = args ? args.targetEntityTypeName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["linkName"] = undefined /*out*/;
            resourceInputs["mappings"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operationType"] = undefined /*out*/;
            resourceInputs["participantPropertyReferences"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["referenceOnly"] = undefined /*out*/;
            resourceInputs["sourceEntityType"] = undefined /*out*/;
            resourceInputs["sourceEntityTypeName"] = undefined /*out*/;
            resourceInputs["targetEntityType"] = undefined /*out*/;
            resourceInputs["targetEntityTypeName"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:customerinsights/v20170101:Link" }, { type: "azure-native:customerinsights/v20170426:Link" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Link.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Link = Link;
/** @internal */
Link.__pulumiType = 'azure-native:customerinsights:Link';
