// @pulumi/azure-native/customerinsights/prediction.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Prediction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The prediction resource format.
 *
 * Uses Azure REST API version 2017-04-26. In version 2.x of the Azure Native provider, it used API version 2017-04-26.
 */
class Prediction extends pulumi.CustomResource {
    /**
     * Get an existing Prediction resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Prediction(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Prediction.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Prediction.__pulumiType;
    }
    /**
     * Create a Prediction resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.autoAnalyze === undefined) && !opts.urn) {
                throw new Error("Missing required property 'autoAnalyze'");
            }
            if ((!args || args.hubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hubName'");
            }
            if ((!args || args.mappings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mappings'");
            }
            if ((!args || args.negativeOutcomeExpression === undefined) && !opts.urn) {
                throw new Error("Missing required property 'negativeOutcomeExpression'");
            }
            if ((!args || args.positiveOutcomeExpression === undefined) && !opts.urn) {
                throw new Error("Missing required property 'positiveOutcomeExpression'");
            }
            if ((!args || args.primaryProfileType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'primaryProfileType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scopeExpression === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scopeExpression'");
            }
            if ((!args || args.scoreLabel === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scoreLabel'");
            }
            resourceInputs["autoAnalyze"] = args ? args.autoAnalyze : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["grades"] = args ? args.grades : undefined;
            resourceInputs["hubName"] = args ? args.hubName : undefined;
            resourceInputs["involvedInteractionTypes"] = args ? args.involvedInteractionTypes : undefined;
            resourceInputs["involvedKpiTypes"] = args ? args.involvedKpiTypes : undefined;
            resourceInputs["involvedRelationships"] = args ? args.involvedRelationships : undefined;
            resourceInputs["mappings"] = args ? args.mappings : undefined;
            resourceInputs["negativeOutcomeExpression"] = args ? args.negativeOutcomeExpression : undefined;
            resourceInputs["positiveOutcomeExpression"] = args ? args.positiveOutcomeExpression : undefined;
            resourceInputs["predictionName"] = args ? args.predictionName : undefined;
            resourceInputs["primaryProfileType"] = args ? args.primaryProfileType : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scopeExpression"] = args ? args.scopeExpression : undefined;
            resourceInputs["scoreLabel"] = args ? args.scoreLabel : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemGeneratedEntities"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["autoAnalyze"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["grades"] = undefined /*out*/;
            resourceInputs["involvedInteractionTypes"] = undefined /*out*/;
            resourceInputs["involvedKpiTypes"] = undefined /*out*/;
            resourceInputs["involvedRelationships"] = undefined /*out*/;
            resourceInputs["mappings"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["negativeOutcomeExpression"] = undefined /*out*/;
            resourceInputs["positiveOutcomeExpression"] = undefined /*out*/;
            resourceInputs["predictionName"] = undefined /*out*/;
            resourceInputs["primaryProfileType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["scopeExpression"] = undefined /*out*/;
            resourceInputs["scoreLabel"] = undefined /*out*/;
            resourceInputs["systemGeneratedEntities"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:customerinsights/v20170426:Prediction" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Prediction.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Prediction = Prediction;
/** @internal */
Prediction.__pulumiType = 'azure-native:customerinsights:Prediction';
