// @pulumi/azure-native/customerinsights/relationship.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Relationship = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The relationship resource format.
 *
 * Uses Azure REST API version 2017-04-26. In version 2.x of the Azure Native provider, it used API version 2017-04-26.
 */
class Relationship extends pulumi.CustomResource {
    /**
     * Get an existing Relationship resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Relationship(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Relationship.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Relationship.__pulumiType;
    }
    /**
     * Create a Relationship resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hubName'");
            }
            if ((!args || args.profileType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'profileType'");
            }
            if ((!args || args.relatedProfileType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'relatedProfileType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["cardinality"] = args ? args.cardinality : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["expiryDateTimeUtc"] = args ? args.expiryDateTimeUtc : undefined;
            resourceInputs["fields"] = args ? args.fields : undefined;
            resourceInputs["hubName"] = args ? args.hubName : undefined;
            resourceInputs["lookupMappings"] = args ? args.lookupMappings : undefined;
            resourceInputs["profileType"] = args ? args.profileType : undefined;
            resourceInputs["relatedProfileType"] = args ? args.relatedProfileType : undefined;
            resourceInputs["relationshipName"] = args ? args.relationshipName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["relationshipGuidId"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cardinality"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["expiryDateTimeUtc"] = undefined /*out*/;
            resourceInputs["fields"] = undefined /*out*/;
            resourceInputs["lookupMappings"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["profileType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["relatedProfileType"] = undefined /*out*/;
            resourceInputs["relationshipGuidId"] = undefined /*out*/;
            resourceInputs["relationshipName"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:customerinsights/v20170101:Relationship" }, { type: "azure-native:customerinsights/v20170426:Relationship" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Relationship.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Relationship = Relationship;
/** @internal */
Relationship.__pulumiType = 'azure-native:customerinsights:Relationship';
