// @pulumi/azure-native/types/enums/hybridcompute/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusLevelTypes = exports.ResourceIdentityType = exports.PublicNetworkAccessType = exports.ProgramYear = exports.PatchModeTypes = exports.LicenseType = exports.LicenseTarget = exports.LicenseState = exports.LicenseProfileSubscriptionStatus = exports.LicenseProfileProductType = exports.LicenseEdition = exports.LicenseCoreType = exports.LicenseAssignmentState = exports.GatewayType = exports.AssessmentModeTypes = exports.ArcKindEnum = void 0;
exports.ArcKindEnum = {
    AVS: "AVS",
    HCI: "HCI",
    SCVMM: "SCVMM",
    VMware: "VMware",
    EPS: "EPS",
    GCP: "GCP",
    AWS: "AWS",
};
exports.AssessmentModeTypes = {
    ImageDefault: "ImageDefault",
    AutomaticByPlatform: "AutomaticByPlatform",
};
exports.GatewayType = {
    Public: "Public",
};
exports.LicenseAssignmentState = {
    Assigned: "Assigned",
    NotAssigned: "NotAssigned",
};
exports.LicenseCoreType = {
    PCore: "pCore",
    VCore: "vCore",
};
exports.LicenseEdition = {
    Standard: "Standard",
    Datacenter: "Datacenter",
};
exports.LicenseProfileProductType = {
    WindowsServer: "WindowsServer",
    WindowsIoTEnterprise: "WindowsIoTEnterprise",
};
exports.LicenseProfileSubscriptionStatus = {
    Unknown: "Unknown",
    Enabling: "Enabling",
    Enabled: "Enabled",
    Disabled: "Disabled",
    Disabling: "Disabling",
    Failed: "Failed",
};
exports.LicenseState = {
    Activated: "Activated",
    Deactivated: "Deactivated",
};
exports.LicenseTarget = {
    Windows_Server_2012: "Windows Server 2012",
    Windows_Server_2012_R2: "Windows Server 2012 R2",
};
exports.LicenseType = {
    ESU: "ESU",
};
exports.PatchModeTypes = {
    ImageDefault: "ImageDefault",
    AutomaticByPlatform: "AutomaticByPlatform",
    AutomaticByOS: "AutomaticByOS",
    Manual: "Manual",
};
exports.ProgramYear = {
    Year_1: "Year 1",
    Year_2: "Year 2",
    Year_3: "Year 3",
};
exports.PublicNetworkAccessType = {
    /**
     * Allows Azure Arc agents to communicate with Azure Arc services over both public (internet) and private endpoints.
     */
    Enabled: "Enabled",
    /**
     * Does not allow Azure Arc agents to communicate with Azure Arc services over public (internet) endpoints. The agents must use the private link.
     */
    Disabled: "Disabled",
    /**
     * Azure Arc agent communication with Azure Arc services over public (internet) is enforced by Network Security Perimeter (NSP)
     */
    SecuredByPerimeter: "SecuredByPerimeter",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.StatusLevelTypes = {
    Info: "Info",
    Warning: "Warning",
    Error: "Error",
};
