// @pulumi/azure-native/redisenterprise/privateEndpointConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Private Endpoint Connection resource.
 *
 * Uses Azure REST API version 2024-03-01-preview.
 *
 * Other available API versions: 2020-10-01-preview, 2021-02-01-preview, 2021-03-01, 2021-08-01, 2022-01-01, 2022-11-01-preview, 2023-03-01-preview, 2023-07-01, 2023-08-01-preview, 2023-10-01-preview, 2023-11-01, 2024-02-01, 2024-06-01-preview, 2024-09-01-preview, 2024-10-01, 2025-04-01, 2025-05-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native redisenterprise [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class PrivateEndpointConnection extends pulumi.CustomResource {
    /**
     * Get an existing PrivateEndpointConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateEndpointConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateEndpointConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateEndpointConnection.__pulumiType;
    }
    /**
     * Create a PrivateEndpointConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.privateLinkServiceConnectionState === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateLinkServiceConnectionState'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            resourceInputs["privateLinkServiceConnectionState"] = args ? args.privateLinkServiceConnectionState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["privateLinkServiceConnectionState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cache/v20230301preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20230701:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20230801preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20231001preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20231101:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240201:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240301preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240601preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20240901preview:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache/v20241001:EnterprisePrivateEndpointConnection" }, { type: "azure-native:cache:EnterprisePrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20201001preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20210201preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20210301:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20210801:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20220101:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20221101preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20230301preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20230701:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20230801preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20231001preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20231101:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20240201:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20240301preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20240601preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20240901preview:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20241001:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20250401:PrivateEndpointConnection" }, { type: "azure-native:redisenterprise/v20250501preview:PrivateEndpointConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateEndpointConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateEndpointConnection = PrivateEndpointConnection;
/** @internal */
PrivateEndpointConnection.__pulumiType = 'azure-native:redisenterprise:PrivateEndpointConnection';
