// @pulumi/azure-native/fluidrelay/fluidRelayServer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FluidRelayServer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A FluidRelay Server.
 *
 * Uses Azure REST API version 2022-06-01. In version 2.x of the Azure Native provider, it used API version 2022-06-01.
 */
class FluidRelayServer extends pulumi.CustomResource {
    /**
     * Get an existing FluidRelayServer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FluidRelayServer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FluidRelayServer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FluidRelayServer.__pulumiType;
    }
    /**
     * Create a FluidRelayServer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            resourceInputs["encryption"] = args ? args.encryption : undefined;
            resourceInputs["fluidRelayServerName"] = args ? args.fluidRelayServerName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            resourceInputs["storagesku"] = args ? args.storagesku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["fluidRelayEndpoints"] = undefined /*out*/;
            resourceInputs["frsTenantId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["encryption"] = undefined /*out*/;
            resourceInputs["fluidRelayEndpoints"] = undefined /*out*/;
            resourceInputs["frsTenantId"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["storagesku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:fluidrelay/v20210312preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20210615preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20210830preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20210910preview:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220215:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220421:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220511:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220526:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20220601:FluidRelayServer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FluidRelayServer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FluidRelayServer = FluidRelayServer;
/** @internal */
FluidRelayServer.__pulumiType = 'azure-native:fluidrelay:FluidRelayServer';
