// @pulumi/azure-native/types/enums/app/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekDay = exports.UnauthenticatedClientActionV2 = exports.Type = exports.TriggerType = exports.StorageType = exports.SessionNetworkStatus = exports.Scheme = exports.PrivateEndpointServiceConnectionStatus = exports.PoolManagementType = exports.ManagedServiceIdentityType = exports.ManagedCertificateDomainControlValidation = exports.LogLevel = exports.JavaComponentType = exports.IngressTransportMethod = exports.IngressClientCertificateMode = exports.IdentitySettingsLifeCycle = exports.ForwardProxyConvention = exports.ExtendedLocationTypes = exports.ExecutionType = exports.DotNetComponentType = exports.CookieExpirationConvention = exports.ContainerType = exports.ClientCredentialMethod = exports.BindingType = exports.AppProtocol = exports.Affinity = exports.ActiveRevisionsMode = exports.Action = exports.AccessMode = void 0;
exports.AccessMode = {
    ReadOnly: "ReadOnly",
    ReadWrite: "ReadWrite",
};
exports.Action = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.ActiveRevisionsMode = {
    Multiple: "Multiple",
    Single: "Single",
};
exports.Affinity = {
    Sticky: "sticky",
    None: "none",
};
exports.AppProtocol = {
    Http: "http",
    Grpc: "grpc",
};
exports.BindingType = {
    Disabled: "Disabled",
    SniEnabled: "SniEnabled",
    Auto: "Auto",
};
exports.ClientCredentialMethod = {
    ClientSecretPost: "ClientSecretPost",
};
exports.ContainerType = {
    CustomContainer: "CustomContainer",
    PythonLTS: "PythonLTS",
};
exports.CookieExpirationConvention = {
    FixedTime: "FixedTime",
    IdentityProviderDerived: "IdentityProviderDerived",
};
exports.DotNetComponentType = {
    AspireDashboard: "AspireDashboard",
};
exports.ExecutionType = {
    Timed: "Timed",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.ForwardProxyConvention = {
    NoProxy: "NoProxy",
    Standard: "Standard",
    Custom: "Custom",
};
exports.IdentitySettingsLifeCycle = {
    None: "None",
    Main: "Main",
};
exports.IngressClientCertificateMode = {
    Ignore: "ignore",
    Accept: "accept",
    Require: "require",
};
exports.IngressTransportMethod = {
    Auto: "auto",
    Http: "http",
    Http2: "http2",
    Tcp: "tcp",
};
exports.JavaComponentType = {
    SpringBootAdmin: "SpringBootAdmin",
    SpringCloudEureka: "SpringCloudEureka",
    SpringCloudConfig: "SpringCloudConfig",
    SpringCloudGateway: "SpringCloudGateway",
    Nacos: "Nacos",
};
exports.LogLevel = {
    Info: "info",
    Debug: "debug",
    Warn: "warn",
    Error: "error",
};
exports.ManagedCertificateDomainControlValidation = {
    CNAME: "CNAME",
    HTTP: "HTTP",
    TXT: "TXT",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PoolManagementType = {
    Manual: "Manual",
    Dynamic: "Dynamic",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.Scheme = {
    HTTP: "HTTP",
    HTTPS: "HTTPS",
};
exports.SessionNetworkStatus = {
    EgressEnabled: "EgressEnabled",
    EgressDisabled: "EgressDisabled",
};
exports.StorageType = {
    AzureFile: "AzureFile",
    EmptyDir: "EmptyDir",
    Secret: "Secret",
};
exports.TriggerType = {
    Schedule: "Schedule",
    Event: "Event",
    Manual: "Manual",
};
exports.Type = {
    Liveness: "Liveness",
    Readiness: "Readiness",
    Startup: "Startup",
};
exports.UnauthenticatedClientActionV2 = {
    RedirectToLoginPage: "RedirectToLoginPage",
    AllowAnonymous: "AllowAnonymous",
    Return401: "Return401",
    Return403: "Return403",
};
exports.WeekDay = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
