// @pulumi/azure-native/sql/extendedDatabaseBlobAuditingPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtendedDatabaseBlobAuditingPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An extended database blob auditing policy.
 *
 * Uses Azure REST API version 2023-08-01. In version 2.x of the Azure Native provider, it used API version 2021-11-01.
 *
 * Other available API versions: 2017-03-01-preview, 2020-02-02-preview, 2020-08-01-preview, 2020-11-01-preview, 2021-02-01-preview, 2021-05-01-preview, 2021-08-01-preview, 2021-11-01, 2021-11-01-preview, 2022-02-01-preview, 2022-05-01-preview, 2022-08-01-preview, 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01-preview, 2024-05-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native sql [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ExtendedDatabaseBlobAuditingPolicy extends pulumi.CustomResource {
    /**
     * Get an existing ExtendedDatabaseBlobAuditingPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExtendedDatabaseBlobAuditingPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExtendedDatabaseBlobAuditingPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExtendedDatabaseBlobAuditingPolicy.__pulumiType;
    }
    /**
     * Create a ExtendedDatabaseBlobAuditingPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            if ((!args || args.state === undefined) && !opts.urn) {
                throw new Error("Missing required property 'state'");
            }
            resourceInputs["auditActionsAndGroups"] = args ? args.auditActionsAndGroups : undefined;
            resourceInputs["blobAuditingPolicyName"] = args ? args.blobAuditingPolicyName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["isAzureMonitorTargetEnabled"] = args ? args.isAzureMonitorTargetEnabled : undefined;
            resourceInputs["isManagedIdentityInUse"] = args ? args.isManagedIdentityInUse : undefined;
            resourceInputs["isStorageSecondaryKeyInUse"] = args ? args.isStorageSecondaryKeyInUse : undefined;
            resourceInputs["predicateExpression"] = args ? args.predicateExpression : undefined;
            resourceInputs["queueDelayMs"] = args ? args.queueDelayMs : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["retentionDays"] = args ? args.retentionDays : undefined;
            resourceInputs["serverName"] = args ? args.serverName : undefined;
            resourceInputs["state"] = args ? args.state : undefined;
            resourceInputs["storageAccountAccessKey"] = args ? args.storageAccountAccessKey : undefined;
            resourceInputs["storageAccountSubscriptionId"] = args ? args.storageAccountSubscriptionId : undefined;
            resourceInputs["storageEndpoint"] = args ? args.storageEndpoint : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["auditActionsAndGroups"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["isAzureMonitorTargetEnabled"] = undefined /*out*/;
            resourceInputs["isManagedIdentityInUse"] = undefined /*out*/;
            resourceInputs["isStorageSecondaryKeyInUse"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["predicateExpression"] = undefined /*out*/;
            resourceInputs["queueDelayMs"] = undefined /*out*/;
            resourceInputs["retentionDays"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["storageAccountSubscriptionId"] = undefined /*out*/;
            resourceInputs["storageEndpoint"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20170301preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20200202preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20200801preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20201101preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20210201preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20210501preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20210801preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20211101:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20211101preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20220201preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20220501preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20220801preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20221101preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20230201preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20230501preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20230801:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20230801preview:ExtendedDatabaseBlobAuditingPolicy" }, { type: "azure-native:sql/v20240501preview:ExtendedDatabaseBlobAuditingPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExtendedDatabaseBlobAuditingPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExtendedDatabaseBlobAuditingPolicy = ExtendedDatabaseBlobAuditingPolicy;
/** @internal */
ExtendedDatabaseBlobAuditingPolicy.__pulumiType = 'azure-native:sql:ExtendedDatabaseBlobAuditingPolicy';
