// @pulumi/azure-native/sql/jobStep.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobStep = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * A job step.
 *
 * Uses Azure REST API version 2023-08-01. In version 2.x of the Azure Native provider, it used API version 2021-11-01.
 *
 * Other available API versions: 2017-03-01-preview, 2020-02-02-preview, 2020-08-01-preview, 2020-11-01-preview, 2021-02-01-preview, 2021-05-01-preview, 2021-08-01-preview, 2021-11-01, 2021-11-01-preview, 2022-02-01-preview, 2022-05-01-preview, 2022-08-01-preview, 2022-11-01-preview, 2023-02-01-preview, 2023-05-01-preview, 2023-08-01-preview, 2024-05-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native sql [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class JobStep extends pulumi.CustomResource {
    /**
     * Get an existing JobStep resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new JobStep(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of JobStep.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === JobStep.__pulumiType;
    }
    /**
     * Create a JobStep resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.action === undefined) && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if ((!args || args.jobAgentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobAgentName'");
            }
            if ((!args || args.jobName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            if ((!args || args.targetGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetGroup'");
            }
            resourceInputs["action"] = args ? (args.action ? pulumi.output(args.action).apply(inputs.sql.jobStepActionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["credential"] = args ? args.credential : undefined;
            resourceInputs["executionOptions"] = args ? (args.executionOptions ? pulumi.output(args.executionOptions).apply(inputs.sql.jobStepExecutionOptionsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["jobAgentName"] = args ? args.jobAgentName : undefined;
            resourceInputs["jobName"] = args ? args.jobName : undefined;
            resourceInputs["output"] = args ? (args.output ? pulumi.output(args.output).apply(inputs.sql.jobStepOutputArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverName"] = args ? args.serverName : undefined;
            resourceInputs["stepId"] = args ? args.stepId : undefined;
            resourceInputs["stepName"] = args ? args.stepName : undefined;
            resourceInputs["targetGroup"] = args ? args.targetGroup : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["action"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["credential"] = undefined /*out*/;
            resourceInputs["executionOptions"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["output"] = undefined /*out*/;
            resourceInputs["stepId"] = undefined /*out*/;
            resourceInputs["targetGroup"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sql/v20170301preview:JobStep" }, { type: "azure-native:sql/v20200202preview:JobStep" }, { type: "azure-native:sql/v20200801preview:JobStep" }, { type: "azure-native:sql/v20201101preview:JobStep" }, { type: "azure-native:sql/v20210201preview:JobStep" }, { type: "azure-native:sql/v20210501preview:JobStep" }, { type: "azure-native:sql/v20210801preview:JobStep" }, { type: "azure-native:sql/v20211101:JobStep" }, { type: "azure-native:sql/v20211101preview:JobStep" }, { type: "azure-native:sql/v20220201preview:JobStep" }, { type: "azure-native:sql/v20220501preview:JobStep" }, { type: "azure-native:sql/v20220801preview:JobStep" }, { type: "azure-native:sql/v20221101preview:JobStep" }, { type: "azure-native:sql/v20230201preview:JobStep" }, { type: "azure-native:sql/v20230501preview:JobStep" }, { type: "azure-native:sql/v20230801:JobStep" }, { type: "azure-native:sql/v20230801preview:JobStep" }, { type: "azure-native:sql/v20240501preview:JobStep" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(JobStep.__pulumiType, name, resourceInputs, opts);
    }
}
exports.JobStep = JobStep;
/** @internal */
JobStep.__pulumiType = 'azure-native:sql:JobStep';
