// @pulumi/azure-native/security/pricing.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Pricing = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Microsoft Defender for Cloud is provided in two pricing tiers: free and standard. The standard tier offers advanced security capabilities, while the free tier offers basic security features.
 *
 * Uses Azure REST API version 2024-01-01. In version 2.x of the Azure Native provider, it used API version 2024-01-01.
 */
class Pricing extends pulumi.CustomResource {
    /**
     * Get an existing Pricing resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Pricing(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Pricing.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Pricing.__pulumiType;
    }
    /**
     * Create a Pricing resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.pricingTier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pricingTier'");
            }
            if ((!args || args.scopeId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scopeId'");
            }
            resourceInputs["enforce"] = args ? args.enforce : undefined;
            resourceInputs["extensions"] = args ? args.extensions : undefined;
            resourceInputs["pricingName"] = args ? args.pricingName : undefined;
            resourceInputs["pricingTier"] = args ? args.pricingTier : undefined;
            resourceInputs["scopeId"] = args ? args.scopeId : undefined;
            resourceInputs["subPlan"] = args ? args.subPlan : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deprecated"] = undefined /*out*/;
            resourceInputs["enablementTime"] = undefined /*out*/;
            resourceInputs["freeTrialRemainingTime"] = undefined /*out*/;
            resourceInputs["inherited"] = undefined /*out*/;
            resourceInputs["inheritedFrom"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["replacedBy"] = undefined /*out*/;
            resourceInputs["resourcesCoverageStatus"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deprecated"] = undefined /*out*/;
            resourceInputs["enablementTime"] = undefined /*out*/;
            resourceInputs["enforce"] = undefined /*out*/;
            resourceInputs["extensions"] = undefined /*out*/;
            resourceInputs["freeTrialRemainingTime"] = undefined /*out*/;
            resourceInputs["inherited"] = undefined /*out*/;
            resourceInputs["inheritedFrom"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pricingTier"] = undefined /*out*/;
            resourceInputs["replacedBy"] = undefined /*out*/;
            resourceInputs["resourcesCoverageStatus"] = undefined /*out*/;
            resourceInputs["subPlan"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:security/v20240101:Pricing" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Pricing.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Pricing = Pricing;
/** @internal */
Pricing.__pulumiType = 'azure-native:security:Pricing';
