// @pulumi/azure-native/databasewatcher/alertRuleResource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertRuleResource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Concrete proxy resource types can be created by aliasing this type using a specific property type.
 *
 * Uses Azure REST API version 2024-10-01-preview. In version 2.x of the Azure Native provider, it used API version 2024-07-19-preview.
 *
 * Other available API versions: 2024-07-19-preview, 2025-01-02. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native databasewatcher [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AlertRuleResource extends pulumi.CustomResource {
    /**
     * Get an existing AlertRuleResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AlertRuleResource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AlertRuleResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AlertRuleResource.__pulumiType;
    }
    /**
     * Create a AlertRuleResource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.alertRuleResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertRuleResourceId'");
            }
            if ((!args || args.alertRuleTemplateId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertRuleTemplateId'");
            }
            if ((!args || args.alertRuleTemplateVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertRuleTemplateVersion'");
            }
            if ((!args || args.createdWithProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'createdWithProperties'");
            }
            if ((!args || args.creationTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'creationTime'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.watcherName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'watcherName'");
            }
            resourceInputs["alertRuleResourceId"] = args ? args.alertRuleResourceId : undefined;
            resourceInputs["alertRuleResourceName"] = args ? args.alertRuleResourceName : undefined;
            resourceInputs["alertRuleTemplateId"] = args ? args.alertRuleTemplateId : undefined;
            resourceInputs["alertRuleTemplateVersion"] = args ? args.alertRuleTemplateVersion : undefined;
            resourceInputs["createdWithProperties"] = args ? args.createdWithProperties : undefined;
            resourceInputs["creationTime"] = args ? args.creationTime : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["watcherName"] = args ? args.watcherName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["alertRuleResourceId"] = undefined /*out*/;
            resourceInputs["alertRuleTemplateId"] = undefined /*out*/;
            resourceInputs["alertRuleTemplateVersion"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdWithProperties"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databasewatcher/v20240719preview:AlertRuleResource" }, { type: "azure-native:databasewatcher/v20241001preview:AlertRuleResource" }, { type: "azure-native:databasewatcher/v20250102:AlertRuleResource" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AlertRuleResource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AlertRuleResource = AlertRuleResource;
/** @internal */
AlertRuleResource.__pulumiType = 'azure-native:databasewatcher:AlertRuleResource';
