// @pulumi/azure-native/devcenter/attachedNetworkByDevCenter.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachedNetworkByDevCenter = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents an attached NetworkConnection.
 *
 * Uses Azure REST API version 2024-02-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2023-04-01, 2023-08-01-preview, 2023-10-01-preview, 2024-05-01-preview, 2024-06-01-preview, 2024-07-01-preview, 2024-08-01-preview, 2024-10-01-preview, 2025-02-01, 2025-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native devcenter [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AttachedNetworkByDevCenter extends pulumi.CustomResource {
    /**
     * Get an existing AttachedNetworkByDevCenter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AttachedNetworkByDevCenter(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AttachedNetworkByDevCenter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AttachedNetworkByDevCenter.__pulumiType;
    }
    /**
     * Create a AttachedNetworkByDevCenter resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.devCenterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'devCenterName'");
            }
            if ((!args || args.networkConnectionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkConnectionId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["attachedNetworkConnectionName"] = args ? args.attachedNetworkConnectionName : undefined;
            resourceInputs["devCenterName"] = args ? args.devCenterName : undefined;
            resourceInputs["networkConnectionId"] = args ? args.networkConnectionId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["domainJoinType"] = undefined /*out*/;
            resourceInputs["healthCheckStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkConnectionLocation"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["domainJoinType"] = undefined /*out*/;
            resourceInputs["healthCheckStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkConnectionId"] = undefined /*out*/;
            resourceInputs["networkConnectionLocation"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devcenter/v20220801preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20220901preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20221012preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20221111preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20230101preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20230401:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20230801preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20231001preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20240201:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20240501preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20240601preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20240701preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20240801preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20241001preview:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20250201:AttachedNetworkByDevCenter" }, { type: "azure-native:devcenter/v20250401preview:AttachedNetworkByDevCenter" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AttachedNetworkByDevCenter.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AttachedNetworkByDevCenter = AttachedNetworkByDevCenter;
/** @internal */
AttachedNetworkByDevCenter.__pulumiType = 'azure-native:devcenter:AttachedNetworkByDevCenter';
