// @pulumi/azure-native/types/enums/migrate/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointServiceConnectionStatus = exports.PricingTier = exports.Percentile = exports.OsLicense = exports.OptimizationLogic = exports.OperatingSystemType = exports.MultiSubnetIntent = exports.MoveType = exports.MigrationStrategy = exports.LoadBalancerType = exports.LicensingProgram = exports.HyperVLicenseType = exports.GroupType = exports.FttAndRaidLevel = exports.ExternalStorageType = exports.EnvironmentType = exports.DiscoverySource = exports.Currency = exports.ConsolidationType = exports.ConfigurationType = exports.ComputeTier = exports.BusinessCaseCurrency = exports.AzureVmSecurityType = exports.AzureVmFamily = exports.AzureVmCategory = exports.AzureTarget = exports.AzureStorageRedundancy = exports.AzureSqlServiceTierV3 = exports.AzureSqlServiceTier = exports.AzureSqlPurchaseModel = exports.AzureSqlInstanceType = exports.AzureSqlDataBaseType = exports.AzureSecurityOfferingType = exports.AzureReservedInstance = exports.AzurePricingTier = exports.AzureOfferCode = exports.AzureOffer = exports.AzureLocation = exports.AzureHybridUseBenefit = exports.AzureEnvironmentType = exports.AzureDiskType = exports.AzureCurrency = exports.AzureAvsNodeType = exports.AzureArcState = exports.AvsAssessmentScenario = exports.AutomationArtifactStatus = exports.AsyncCommitModeIntent = exports.AssessmentType = exports.AssessmentStage = exports.AssessmentSizingCriterion = void 0;
exports.ZoneRedundant = exports.WorkloadType = exports.WorkloadDeploymentType = exports.WorkloadDeploymentTarget = exports.TimeRange = exports.TargetStorageProjectionType = exports.TargetStorageAccessType = exports.TargetHydrationStorageProviderType = exports.TargetAvailabilityZone = exports.Status = exports.SqlServerLicenseType = exports.SqlServerLicense = exports.SecretStoreType = exports.ScopeType = exports.SavingsOptions = exports.SavingsOption = exports.ResourceIdentityTypes = exports.ResourceIdentityType = exports.ReservedInstance = exports.ProvisioningState = exports.ProjectStatus = void 0;
exports.AssessmentSizingCriterion = {
    /**
     * Performance Data based Sizing.
     */
    PerformanceBased: "PerformanceBased",
    /**
     * As On Premises or Static Data based Sizing.
     */
    AsOnPremises: "AsOnPremises",
};
exports.AssessmentStage = {
    InProgress: "InProgress",
    UnderReview: "UnderReview",
    Approved: "Approved",
};
exports.AssessmentType = {
    Unknown: "Unknown",
    MachineAssessment: "MachineAssessment",
    AvsAssessment: "AvsAssessment",
    SqlAssessment: "SqlAssessment",
    WebAppAssessment: "WebAppAssessment",
};
exports.AsyncCommitModeIntent = {
    /**
     * None - Async Commit Mode Intent
     */
    None: "None",
    /**
     * HighAvailability - Async Commit Mode Intent
     */
    HighAvailability: "HighAvailability",
    /**
     * DisasterRecovery - Async Commit Mode Intent
     */
    DisasterRecovery: "DisasterRecovery",
};
exports.AutomationArtifactStatus = {
    NotGenerated: "NotGenerated",
    Generated: "Generated",
};
exports.AvsAssessmentScenario = {
    /**
     * Unknown assessment scenario
     */
    Unknown: "Unknown",
    /**
     * Create Assessment for new Avs Sddc
     */
    NewAvsSddc: "NewAvsSddc",
    /**
     * Create Assessment for existing Avs Sddc
     */
    AvsSddcExpansion: "AvsSddcExpansion",
};
exports.AzureArcState = {
    /**
     * AzureArc state disabled.
     */
    Disabled: "Disabled",
    /**
     * AzureArc state enabled.
     */
    Enabled: "Enabled",
    /**
     * AzureArc state unknown.
     */
    Unknown: "Unknown",
};
exports.AzureAvsNodeType = {
    /**
     * Unknown Node type
     */
    Unknown: "Unknown",
    /**
     * AVS36 Node type
     */
    AV36: "AV36",
    /**
     * AV36P Node type
     */
    AV36P: "AV36P",
    /**
     * AV52 Node type
     */
    AV52: "AV52",
    /**
     * AV64 Node type
     */
    AV64: "AV64",
};
exports.AzureCurrency = {
    /**
     * Unknown Currency for Azure
     */
    Unknown: "Unknown",
    /**
     * USD Currency for Azure
     */
    USD: "USD",
    /**
     * DKK Currency for Azure
     */
    DKK: "DKK",
    /**
     * CAD Currency for Azure
     */
    CAD: "CAD",
    /**
     * IDR Currency for Azure
     */
    IDR: "IDR",
    /**
     * JPY Currency for Azure
     */
    JPY: "JPY",
    /**
     * KRW Currency for Azure
     */
    KRW: "KRW",
    /**
     * NZD Currency for Azure
     */
    NZD: "NZD",
    /**
     * NOK Currency for Azure
     */
    NOK: "NOK",
    /**
     * RUB Currency for Azure
     */
    RUB: "RUB",
    /**
     * SAR Currency for Azure
     */
    SAR: "SAR",
    /**
     * ZAR Currency for Azure
     */
    ZAR: "ZAR",
    /**
     * SEK Currency for Azure
     */
    SEK: "SEK",
    /**
     * TRY Currency for Azure
     */
    TRY: "TRY",
    /**
     * GBP Currency for Azure
     */
    GBP: "GBP",
    /**
     * MXN Currency for Azure
     */
    MXN: "MXN",
    /**
     * MYR Currency for Azure
     */
    MYR: "MYR",
    /**
     * INR Currency for Azure
     */
    INR: "INR",
    /**
     * HKD Currency for Azure
     */
    HKD: "HKD",
    /**
     * BRL Currency for Azure
     */
    BRL: "BRL",
    /**
     * TWD Currency for Azure
     */
    TWD: "TWD",
    /**
     * EUR Currency for Azure
     */
    EUR: "EUR",
    /**
     * CHF Currency for Azure
     */
    CHF: "CHF",
    /**
     * ARS Currency for Azure
     */
    ARS: "ARS",
    /**
     * AUD Currency for Azure
     */
    AUD: "AUD",
    /**
     * CNY Currency for Azure
     */
    CNY: "CNY",
};
exports.AzureDiskType = {
    /**
     * Unknown Azure Disk Type
     */
    Unknown: "Unknown",
    /**
     * Standard Azure Disk Type
     */
    Standard: "Standard",
    /**
     * StandardSSD Azure Disk Type
     */
    StandardSSD: "StandardSSD",
    /**
     * Premium Azure Disk Type
     */
    Premium: "Premium",
    /**
     * StandardOrPremium Azure Disk Type
     */
    StandardOrPremium: "StandardOrPremium",
    /**
     * Ultra Azure Disk Type
     */
    Ultra: "Ultra",
    /**
     * PremiumV2 Azure Disk Type
     */
    PremiumV2: "PremiumV2",
};
exports.AzureEnvironmentType = {
    /**
     * Unknown. Indicates missing data.
     */
    Unknown: "Unknown",
    /**
     * Development or Test Environment.
     */
    DevTest: "DevTest",
    /**
     * Production Environment.
     */
    Production: "Production",
};
exports.AzureHybridUseBenefit = {
    /**
     * Unknown Azure Hybrid Use Benefit
     */
    Unknown: "Unknown",
    /**
     * Yes Azure Hybrid Use Benefit
     */
    Yes: "Yes",
    /**
     * No Azure Hybrid Use Benefit
     */
    No: "No",
};
exports.AzureLocation = {
    /**
     * Unknown Location for Azure
     */
    Unknown: "Unknown",
    /**
     * EastAsia Location for Azure
     */
    EastAsia: "EastAsia",
    /**
     * SoutheastAsia Location for Azure
     */
    SoutheastAsia: "SoutheastAsia",
    /**
     * AustraliaEast Location for Azure
     */
    AustraliaEast: "AustraliaEast",
    /**
     * AustraliaSoutheast Location for Azure
     */
    AustraliaSoutheast: "AustraliaSoutheast",
    /**
     * BrazilSouth Location for Azure
     */
    BrazilSouth: "BrazilSouth",
    /**
     * CanadaCentral Location for Azure
     */
    CanadaCentral: "CanadaCentral",
    /**
     * CanadaEast Location for Azure
     */
    CanadaEast: "CanadaEast",
    /**
     * WestEurope Location for Azure
     */
    WestEurope: "WestEurope",
    /**
     * NorthEurope Location for Azure
     */
    NorthEurope: "NorthEurope",
    /**
     * CentralIndia Location for Azure
     */
    CentralIndia: "CentralIndia",
    /**
     * SouthIndia Location for Azure
     */
    SouthIndia: "SouthIndia",
    /**
     * WestIndia Location for Azure
     */
    WestIndia: "WestIndia",
    /**
     * JapanEast Location for Azure
     */
    JapanEast: "JapanEast",
    /**
     * JapanWest Location for Azure
     */
    JapanWest: "JapanWest",
    /**
     * KoreaCentral Location for Azure
     */
    KoreaCentral: "KoreaCentral",
    /**
     * KoreaSouth Location for Azure
     */
    KoreaSouth: "KoreaSouth",
    /**
     * UkWest Location for Azure
     */
    UkWest: "UkWest",
    /**
     * UkSouth Location for Azure
     */
    UkSouth: "UkSouth",
    /**
     * NorthCentralUs Location for Azure
     */
    NorthCentralUs: "NorthCentralUs",
    /**
     * EastUs Location for Azure
     */
    EastUs: "EastUs",
    /**
     * WestUs2 Location for Azure
     */
    WestUs2: "WestUs2",
    /**
     * SouthCentralUs Location for Azure
     */
    SouthCentralUs: "SouthCentralUs",
    /**
     * CentralUs Location for Azure
     */
    CentralUs: "CentralUs",
    /**
     * EastUs2 Location for Azure
     */
    EastUs2: "EastUs2",
    /**
     * WestUs Location for Azure
     */
    WestUs: "WestUs",
    /**
     * WestCentralUs Location for Azure
     */
    WestCentralUs: "WestCentralUs",
    /**
     * GermanyCentral Location for Azure
     */
    GermanyCentral: "GermanyCentral",
    /**
     * GermanyNortheast Location for Azure
     */
    GermanyNortheast: "GermanyNortheast",
    /**
     * ChinaNorth Location for Azure
     */
    ChinaNorth: "ChinaNorth",
    /**
     * ChinaEast Location for Azure
     */
    ChinaEast: "ChinaEast",
    /**
     * USGovArizona Location for Azure
     */
    USGovArizona: "USGovArizona",
    /**
     * USGovTexas Location for Azure
     */
    USGovTexas: "USGovTexas",
    /**
     * USGovIowa Location for Azure
     */
    USGovIowa: "USGovIowa",
    /**
     * USGovVirginia Location for Azure
     */
    USGovVirginia: "USGovVirginia",
    /**
     * USDoDCentral Location for Azure
     */
    USDoDCentral: "USDoDCentral",
    /**
     * USDoDEast Location for Azure
     */
    USDoDEast: "USDoDEast",
    /**
     * FranceCentral Location for Azure
     */
    FranceCentral: "FranceCentral",
    /**
     * AustraliaCentral Location for Azure
     */
    AustraliaCentral: "AustraliaCentral",
    /**
     * SouthAfricaNorth Location for Azure
     */
    SouthAfricaNorth: "SouthAfricaNorth",
    /**
     * FranceSouth Location for Azure
     */
    FranceSouth: "FranceSouth",
    /**
     * AustraliaCentral2 Location for Azure
     */
    AustraliaCentral2: "AustraliaCentral2",
    /**
     * SouthAfricaWest Location for Azure
     */
    SouthAfricaWest: "SouthAfricaWest",
    /**
     * GermanyNorth Location for Azure
     */
    GermanyNorth: "GermanyNorth",
    /**
     * GermanyWestCentral Location for Azure
     */
    GermanyWestCentral: "GermanyWestCentral",
    /**
     * NorwayEast Location for Azure
     */
    NorwayEast: "NorwayEast",
    /**
     * NorwayWest Location for Azure
     */
    NorwayWest: "NorwayWest",
    /**
     * ChinaEast2 Location for Azure
     */
    ChinaEast2: "ChinaEast2",
    /**
     * ChinaNorth2 Location for Azure
     */
    ChinaNorth2: "ChinaNorth2",
    /**
     * SwitzerlandNorth Location for Azure
     */
    SwitzerlandNorth: "SwitzerlandNorth",
    /**
     * SwitzerlandWest Location for Azure
     */
    SwitzerlandWest: "SwitzerlandWest",
    /**
     * UAENorth Location for Azure
     */
    UAENorth: "UAENorth",
    /**
     * UAECentral Location for Azure
     */
    UAECentral: "UAECentral",
    /**
     * UsNatEast Location for Azure
     */
    UsNatEast: "UsNatEast",
    /**
     * UsNatWest Location for Azure
     */
    UsNatWest: "UsNatWest",
    /**
     * UsSecEast Location for Azure
     */
    UsSecEast: "UsSecEast",
    /**
     * UsSecCentral Location for Azure
     */
    UsSecCentral: "UsSecCentral",
    /**
     * UsSecWest Location for Azure
     */
    UsSecWest: "UsSecWest",
    /**
     * SwedenCentral Location for Azure
     */
    SwedenCentral: "SwedenCentral",
    /**
     * QatarCentral Location for Azure
     */
    QatarCentral: "QatarCentral",
    /**
     * JioIndiaWest Location for Azure
     */
    JioIndiaWest: "JioIndiaWest",
    /**
     * ItalyNorth Location for Azure
     */
    ItalyNorth: "ItalyNorth",
    /**
     * PolandCentral Location for Azure
     */
    PolandCentral: "PolandCentral",
    /**
     * IsraelCentral Location for Azure
     */
    IsraelCentral: "IsraelCentral",
    /**
     * MexicoCentral Location for Azure
     */
    MexicoCentral: "MexicoCentral",
    /**
     * NewZealandNorth Location for Azure
     */
    NewZealandNorth: "NewZealandNorth",
    /**
     * SpainCentral Location for Azure
     */
    SpainCentral: "SpainCentral",
};
exports.AzureOffer = {
    /**
     * Unknown - Azure Offer
     */
    Unknown: "Unknown",
    /**
     * MSAZR0003P Azure Offer
     */
    Msazr0003P: "MSAZR0003P",
    /**
     * MSAZR0023P Azure Offer
     */
    Msazr0023P: "MSAZR0023P",
    /**
     * MSMCAZR0044P Azure Offer
     */
    Msmcazr0044P: "MSMCAZR0044P",
    /**
     * MSMCAZR0059P Azure Offer
     */
    Msmcazr0059P: "MSMCAZR0059P",
    /**
     * MSMCAZR0060P Azure Offer
     */
    Msmcazr0060P: "MSMCAZR0060P",
    /**
     * MSMCAZR0063P Azure Offer
     */
    Msmcazr0063P: "MSMCAZR0063P",
    /**
     * MSAZRUSGOV0003P Azure Offer
     */
    Msazrusgov0003P: "MSAZRUSGOV0003P",
    /**
     * EA Azure Offer
     */
    EA: "EA",
    /**
     * SavingsPlan1Year Azure Offer
     */
    SavingsPlan1Year: "SavingsPlan1Year",
    /**
     * SavingsPlan3Year Azure Offer
     */
    SavingsPlan3Year: "SavingsPlan3Year",
};
exports.AzureOfferCode = {
    Unknown: "Unknown",
    Msazr0003P: "MSAZR0003P",
    Msazr0044P: "MSAZR0044P",
    Msazr0059P: "MSAZR0059P",
    Msazr0060P: "MSAZR0060P",
    Msazr0062P: "MSAZR0062P",
    Msazr0063P: "MSAZR0063P",
    Msazr0064P: "MSAZR0064P",
    Msazr0029P: "MSAZR0029P",
    Msazr0022P: "MSAZR0022P",
    Msazr0023P: "MSAZR0023P",
    Msazr0148P: "MSAZR0148P",
    Msazr0025P: "MSAZR0025P",
    Msazr0036P: "MSAZR0036P",
    Msazr0120P: "MSAZR0120P",
    Msazr0121P: "MSAZR0121P",
    Msazr0122P: "MSAZR0122P",
    Msazr0123P: "MSAZR0123P",
    Msazr0124P: "MSAZR0124P",
    Msazr0125P: "MSAZR0125P",
    Msazr0126P: "MSAZR0126P",
    Msazr0127P: "MSAZR0127P",
    Msazr0128P: "MSAZR0128P",
    Msazr0129P: "MSAZR0129P",
    Msazr0130P: "MSAZR0130P",
    Msazr0111P: "MSAZR0111P",
    Msazr0144P: "MSAZR0144P",
    Msazr0149P: "MSAZR0149P",
    Msmcazr0044P: "MSMCAZR0044P",
    Msmcazr0059P: "MSMCAZR0059P",
    Msmcazr0060P: "MSMCAZR0060P",
    Msmcazr0063P: "MSMCAZR0063P",
    Msmcazr0120P: "MSMCAZR0120P",
    Msmcazr0121P: "MSMCAZR0121P",
    Msmcazr0125P: "MSMCAZR0125P",
    Msmcazr0128P: "MSMCAZR0128P",
    Msazrde0003P: "MSAZRDE0003P",
    Msazrde0044P: "MSAZRDE0044P",
    Msazrusgov0003P: "MSAZRUSGOV0003P",
    EA: "EA",
    Msazr0243P: "MSAZR0243P",
    SavingsPlan1Year: "SavingsPlan1Year",
    SavingsPlan3Year: "SavingsPlan3Year",
};
exports.AzurePricingTier = {
    /**
     * Standard Azure Pricing Tier
     */
    Standard: "Standard",
    /**
     * Basic Azure Pricing Tier
     */
    Basic: "Basic",
};
exports.AzureReservedInstance = {
    None: "None",
    RI1Year: "RI1Year",
    RI3Year: "RI3Year",
};
exports.AzureSecurityOfferingType = {
    /**
     * NO - Azure Security Offering Type
     */
    NO: "NO",
    /**
     * MDC - Azure Security Offering Type
     */
    MDC: "MDC",
};
exports.AzureSqlDataBaseType = {
    /**
     * Unknown Data Base Type
     */
    Unknown: "Unknown",
    /**
     * Automatic Data Base Type
     */
    Automatic: "Automatic",
    /**
     * SingleDatabase Data Base Type
     */
    SingleDatabase: "SingleDatabase",
    /**
     * ElasticPool Data Base Type
     */
    ElasticPool: "ElasticPool",
};
exports.AzureSqlInstanceType = {
    /**
     * Unknown Azure Sql Instance Type
     */
    Unknown: "Unknown",
    /**
     * Automatic Azure Sql Instance Type
     */
    Automatic: "Automatic",
    /**
     * SingleInstance Azure Sql Instance Type
     */
    SingleInstance: "SingleInstance",
    /**
     * InstancePools Azure Sql Instance Type
     */
    InstancePools: "InstancePools",
};
exports.AzureSqlPurchaseModel = {
    /**
     * Unknown Purchase Model
     */
    Unknown: "Unknown",
    /**
     * VCore Purchase Model
     */
    VCore: "VCore",
    /**
     * DTU Purchase Model
     */
    DTU: "DTU",
};
exports.AzureSqlServiceTier = {
    Unknown: "Unknown",
    Automatic: "Automatic",
    GeneralPurpose: "GeneralPurpose",
    BusinessCritical: "BusinessCritical",
    HyperScale: "HyperScale",
};
exports.AzureSqlServiceTierV3 = {
    /**
     * SqlService_Unknown Sql Service Tier V3
     */
    SqlService_Unknown: "SqlService_Unknown",
    /**
     * SqlService_Automatic Sql Service Tier V3
     */
    SqlService_Automatic: "SqlService_Automatic",
    /**
     * SqlService_GeneralPurpose Sql Service Tier V3
     */
    SqlService_GeneralPurpose: "SqlService_GeneralPurpose",
    /**
     * SqlService_BusinessCritical Sql Service Tier V3
     */
    SqlService_BusinessCritical: "SqlService_BusinessCritical",
    /**
     * SqlService_HyperScale Sql Service Tier V3
     */
    SqlService_HyperScale: "SqlService_HyperScale",
};
exports.AzureStorageRedundancy = {
    /**
     * Unknown Azure Storage Redundancy
     */
    Unknown: "Unknown",
    /**
     * LocallyRedundant Azure Storage Redundancy
     */
    LocallyRedundant: "LocallyRedundant",
    /**
     * ZoneRedundant Azure Storage Redundancy
     */
    ZoneRedundant: "ZoneRedundant",
    /**
     * GeoRedundant Azure Storage Redundancy
     */
    GeoRedundant: "GeoRedundant",
    /**
     * ReadAccessGeoRedundant Azure Storage Redundancy
     */
    ReadAccessGeoRedundant: "ReadAccessGeoRedundant",
};
exports.AzureTarget = {
    /**
     * Unknown - Azure Target
     */
    Unknown: "Unknown",
    /**
     * SqlDatabase - Azure Target
     */
    SqlDatabase: "SqlDatabase",
    /**
     * SqlMI - Azure Target
     */
    SqlMI: "SqlMI",
    /**
     * FlexServerPG - Azure Target
     */
    FlexServerPG: "FlexServerPG",
    /**
     * OracleIaasVM - Azure Target
     */
    OracleIaasVM: "OracleIaasVM",
    /**
     * AzureSpringApps - Azure Target
     */
    AzureSpringApps: "AzureSpringApps",
    /**
     * SAPAzureInstance - Azure Target
     */
    SAPAzureInstance: "SAPAzureInstance",
    /**
     * AKS - Azure Target
     */
    AKS: "AKS",
    /**
     * MySQLAzureFlexServer - Azure Target
     */
    MySQLAzureFlexServer: "MySQLAzureFlexServer",
    /**
     * AzureSQLVM - Azure Target
     */
    AzureSQLVM: "AzureSQLVM",
    /**
     * AzureVM - Azure Target
     */
    AzureVM: "AzureVM",
    /**
     * AzureAppService - Azure Target
     */
    AzureAppService: "AzureAppService",
    /**
     * AzureAppServiceContainer - Azure Target
     */
    AzureAppServiceContainer: "AzureAppServiceContainer",
    /**
     * Avs - Azure Target
     */
    Avs: "Avs",
};
exports.AzureVmCategory = {
    /**
     * Indicates All categories of VM.
     */
    All: "All",
    /**
     * Compute Optimized.
     */
    ComputeOptimized: "ComputeOptimized",
    /**
     * General Purpose.
     */
    GeneralPurpose: "GeneralPurpose",
    /**
     * GPU Optimized.
     */
    GpuOptimized: "GpuOptimized",
    /**
     * High Performance Compute.
     */
    HighPerformanceCompute: "HighPerformanceCompute",
    /**
     * Memory Optimized.
     */
    MemoryOptimized: "MemoryOptimized",
    /**
     * Storage Optimized.
     */
    StorageOptimized: "StorageOptimized",
    /**
     * Isolated VM.
     */
    Isolated: "Isolated",
};
exports.AzureVmFamily = {
    /**
     * Unknown - Azure VM Family
     */
    Unknown: "Unknown",
    /**
     * Basic_A0_A4
     */
    BasicA0A4: "Basic_A0_A4",
    /**
     * Standard_A0_A7
     */
    StandardA0A7: "Standard_A0_A7",
    /**
     * Standard_A8_A11
     */
    StandardA8A11: "Standard_A8_A11",
    /**
     * Av2_series
     */
    Av2Series: "Av2_series",
    /**
     * D_series
     */
    DSeries: "D_series",
    /**
     * Dv2_series
     */
    Dv2Series: "Dv2_series",
    /**
     * DS_series
     */
    DSSeries: "DS_series",
    /**
     * DSv2_series
     */
    DSv2Series: "DSv2_series",
    /**
     * F_series
     */
    FSeries: "F_series",
    /**
     * Fs_series
     */
    FsSeries: "Fs_series",
    /**
     * G_series
     */
    GSeries: "G_series",
    /**
     * GS_series
     */
    GSSeries: "GS_series",
    /**
     * H_series
     */
    HSeries: "H_series",
    /**
     * Ls_series
     */
    LsSeries: "Ls_series",
    /**
     * Dsv3_series
     */
    Dsv3Series: "Dsv3_series",
    /**
     * Dv3_series
     */
    Dv3Series: "Dv3_series",
    /**
     * Fsv2_series
     */
    Fsv2Series: "Fsv2_series",
    /**
     * Ev3_series
     */
    Ev3Series: "Ev3_series",
    /**
     * Esv3_series
     */
    Esv3Series: "Esv3_series",
    /**
     * M_series
     */
    MSeries: "M_series",
    /**
     * DC_Series
     */
    DCSeries: "DC_Series",
    /**
     * Lsv2_series
     */
    Lsv2Series: "Lsv2_series",
    /**
     * Ev4_series
     */
    Ev4Series: "Ev4_series",
    /**
     * Esv4_series
     */
    Esv4Series: "Esv4_series",
    /**
     * Edv4_series
     */
    Edv4Series: "Edv4_series",
    /**
     * Edsv4_series
     */
    Edsv4Series: "Edsv4_series",
    /**
     * Dv4_series
     */
    Dv4Series: "Dv4_series",
    /**
     * Dsv4_series
     */
    Dsv4Series: "Dsv4_series",
    /**
     * Ddv4_series
     */
    Ddv4Series: "Ddv4_series",
    /**
     * Ddsv4_series
     */
    Ddsv4Series: "Ddsv4_series",
    /**
     * Easv4_series
     */
    Easv4Series: "Easv4_series",
    /**
     * Dasv4_series
     */
    Dasv4Series: "Dasv4_series",
    /**
     * Mv2_series
     */
    Mv2Series: "Mv2_series",
    /**
     * Eav4_series
     */
    Eav4Series: "Eav4_series",
    /**
     * Dav4_series
     */
    Dav4Series: "Dav4_series",
    /**
     * Msv2_series
     */
    Msv2Series: "Msv2_series",
    /**
     * Mdsv2_series
     */
    Mdsv2Series: "Mdsv2_series",
    /**
     * Dv5_series
     */
    Dv5Series: "Dv5_series",
    /**
     * Dsv5_series
     */
    Dsv5Series: "Dsv5_series",
    /**
     * Ddv5_series
     */
    Ddv5Series: "Ddv5_series",
    /**
     * Ddsv5_series
     */
    Ddsv5Series: "Ddsv5_series",
    /**
     * Dasv5_series
     */
    Dasv5Series: "Dasv5_series",
    /**
     * Dadsv5_series
     */
    Dadsv5Series: "Dadsv5_series",
    /**
     * Ev5_series
     */
    Ev5Series: "Ev5_series",
    /**
     * Esv5_series
     */
    Esv5Series: "Esv5_series",
    /**
     * Edv5_series
     */
    Edv5Series: "Edv5_series",
    /**
     * Edsv5_series
     */
    Edsv5Series: "Edsv5_series",
    /**
     * Easv5_series
     */
    Easv5Series: "Easv5_series",
    /**
     * Eadsv5_series
     */
    Eadsv5Series: "Eadsv5_series",
    /**
     * Ebsv5_series
     */
    Ebsv5Series: "Ebsv5_series",
    /**
     * Ebdsv5_series
     */
    Ebdsv5Series: "Ebdsv5_series",
};
exports.AzureVmSecurityType = {
    /**
     * Unknown - Azure Vm Security Type
     */
    Unknown: "Unknown",
    /**
     * Standard Azure Vm Security Type
     */
    Standard: "Standard",
    /**
     * TVM Azure Vm Security Type
     */
    TVM: "TVM",
    /**
     * CVM Azure Vm Security Type
     */
    CVM: "CVM",
};
exports.BusinessCaseCurrency = {
    /**
     * Currency Unknown.
     */
    Unknown: "Unknown",
    /**
     * Currency USD.
     */
    USD: "USD",
    /**
     * Currency DKK.
     */
    DKK: "DKK",
    /**
     * Currency CAD.
     */
    CAD: "CAD",
    /**
     * Currency IDR.
     */
    IDR: "IDR",
    /**
     * Currency JPY.
     */
    JPY: "JPY",
    /**
     * Currency KRW.
     */
    KRW: "KRW",
    /**
     * Currency NZD.
     */
    NZD: "NZD",
    /**
     * Currency NOK.
     */
    NOK: "NOK",
    /**
     * Currency RUB.
     */
    RUB: "RUB",
    /**
     * Currency SAR.
     */
    SAR: "SAR",
    /**
     * Currency ZAR.
     */
    ZAR: "ZAR",
    /**
     * Currency SEK.
     */
    SEK: "SEK",
    /**
     * Currency TRY.
     */
    TRY: "TRY",
    /**
     * Currency GBP.
     */
    GBP: "GBP",
    /**
     * Currency MXN.
     */
    MXN: "MXN",
    /**
     * Currency MYR.
     */
    MYR: "MYR",
    /**
     * Currency INR.
     */
    INR: "INR",
    /**
     * Currency HKD.
     */
    HKD: "HKD",
    /**
     * Currency BRL.
     */
    BRL: "BRL",
    /**
     * Currency TWD.
     */
    TWD: "TWD",
    /**
     * Currency EUR.
     */
    EUR: "EUR",
    /**
     * Currency CHF.
     */
    CHF: "CHF",
    /**
     * Currency ARS.
     */
    ARS: "ARS",
    /**
     * Currency AUD.
     */
    AUD: "AUD",
    /**
     * Currency CNY.
     */
    CNY: "CNY",
};
exports.ComputeTier = {
    /**
     * Unknown - Compute Tier
     */
    Unknown: "Unknown",
    /**
     * Automatic - Compute Tier
     */
    Automatic: "Automatic",
    /**
     * Provisioned - Compute Tier
     */
    Provisioned: "Provisioned",
    /**
     * Serverless - Compute Tier
     */
    Serverless: "Serverless",
};
exports.ConfigurationType = {
    IISConnectionString: "IISConnectionString",
    IISAuthentication: "IISAuthentication",
    ApacheTomcatContextResource: "ApacheTomcatContextResource",
};
exports.ConsolidationType = {
    /**
     * Full Consolidation.
     */
    Full: "Full",
    /**
     * As On Source or On Premises Consolidation.
     */
    AsOnSource: "AsOnSource",
};
exports.Currency = {
    Unknown: "Unknown",
    USD: "USD",
    DKK: "DKK",
    CAD: "CAD",
    IDR: "IDR",
    JPY: "JPY",
    KRW: "KRW",
    NZD: "NZD",
    NOK: "NOK",
    RUB: "RUB",
    SAR: "SAR",
    ZAR: "ZAR",
    SEK: "SEK",
    TRY: "TRY",
    GBP: "GBP",
    MXN: "MXN",
    MYR: "MYR",
    INR: "INR",
    HKD: "HKD",
    BRL: "BRL",
    TWD: "TWD",
    EUR: "EUR",
    CHF: "CHF",
    ARS: "ARS",
    AUD: "AUD",
    CNY: "CNY",
};
exports.DiscoverySource = {
    /**
     * Unknown Discovery Source.
     */
    Unknown: "Unknown",
    /**
     * Appliance Discovery Source.
     */
    Appliance: "Appliance",
    /**
     * Import Discovery Source.
     */
    Import: "Import",
};
exports.EnvironmentType = {
    /**
     * Production - Environment Type
     */
    Production: "Production",
    /**
     * Unknown - Environment Type
     */
    Unknown: "Unknown",
    /**
     * DevTest - Environment Type
     */
    DevTest: "DevTest",
};
exports.ExternalStorageType = {
    /**
     * Indicates un-supported external storage.
     */
    Unknown: "Unknown",
    /**
     * Indicates No external storage.
     */
    None: "None",
    /**
     * ANF external storage with Standard SKU.
     */
    AnfStandard: "AnfStandard",
    /**
     * ANF external storage with Premium SKU.
     */
    AnfPremium: "AnfPremium",
    /**
     * ANF external storage with Ultra SKU.
     */
    AnfUltra: "AnfUltra",
};
exports.FttAndRaidLevel = {
    /**
     * Unknown FTT and RAID Level.
     */
    Unknown: "Unknown",
    /**
     * FTT 1 and RAID Level 1.
     */
    Ftt1Raid1: "Ftt1Raid1",
    /**
     * FTT 1 and RAID Level 5.
     */
    Ftt1Raid5: "Ftt1Raid5",
    /**
     * FTT 2 and RAID Level 1.
     */
    Ftt2Raid1: "Ftt2Raid1",
    /**
     * FTT 2 and RAID Level 6.
     */
    Ftt2Raid6: "Ftt2Raid6",
    /**
     * FTT 3 and RAID Level 1.
     */
    Ftt3Raid1: "Ftt3Raid1",
};
exports.GroupType = {
    Default: "Default",
    Import: "Import",
};
exports.HyperVLicenseType = {
    /**
     * Unknown HyperV License.
     */
    Unknown: "Unknown",
    /**
     * Datacentre HyperV License.
     */
    Datacentre: "Datacentre",
    /**
     * Standard HyperV License.
     */
    Standard: "Standard",
};
exports.LicensingProgram = {
    /**
     * Retail/Pay as you go.
     */
    Retail: "Retail",
    /**
     * Not known
     */
    Unknown: "Unknown",
    /**
     * Enterprise agreement.
     */
    EA: "EA",
    /**
     * Microsoft customer agreement.
     */
    MCA: "MCA",
};
exports.LoadBalancerType = {
    Private: "Private",
    Public: "Public",
};
exports.MigrationStrategy = {
    /**
     * Unknown Migration Strategy.
     */
    Unknown: "Unknown",
    /**
     * Optimize for cost.
     */
    OptimizeForCost: "OptimizeForCost",
    /**
     * IaaS only.
     */
    IaaSOnly: "IaaSOnly",
    /**
     * Optimize for PaaS.
     */
    OptimizeForPaas: "OptimizeForPaas",
    /**
     * Avs only.
     */
    AVSOnly: "AVSOnly",
};
exports.MoveType = {
    RegionToRegion: "RegionToRegion",
    RegionToZone: "RegionToZone",
};
exports.MultiSubnetIntent = {
    /**
     * None - Multi Subnet Intent
     */
    None: "None",
    /**
     * HighAvailability - Multi Subnet Intent
     */
    HighAvailability: "HighAvailability",
    /**
     * DisasterRecovery - Multi Subnet Intent
     */
    DisasterRecovery: "DisasterRecovery",
};
exports.OperatingSystemType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.OptimizationLogic = {
    MinimizeCost: "MinimizeCost",
    ModernizeToPaaS: "ModernizeToPaaS",
    ModernizeToAzureSqlMi: "ModernizeToAzureSqlMi",
    ModernizeToAzureSqlDb: "ModernizeToAzureSqlDb",
};
exports.OsLicense = {
    /**
     * Unknown Os License
     */
    Unknown: "Unknown",
    /**
     * Yes Os License
     */
    Yes: "Yes",
    /**
     * No Os License
     */
    No: "No",
};
exports.Percentile = {
    /**
     * Percentile 50.
     */
    Percentile50: "Percentile50",
    /**
     * Percentile 90.
     */
    Percentile90: "Percentile90",
    /**
     * Percentile 95.
     */
    Percentile95: "Percentile95",
    /**
     * Percentile 99.
     */
    Percentile99: "Percentile99",
    /**
     * Percentile unknown.
     */
    PercentileUnknown: "PercentileUnknown",
};
exports.PricingTier = {
    /**
     * Standard Pricing Tier.
     */
    Standard: "Standard",
    /**
     * Free Pricing Tier.
     */
    Free: "Free",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.ProjectStatus = {
    Active: "Active",
    Inactive: "Inactive",
};
exports.ProvisioningState = {
    /**
     * Resource has been created.
     */
    Succeeded: "Succeeded",
    /**
     * Resource creation failed.
     */
    Failed: "Failed",
    /**
     * Resource creation was canceled.
     */
    Canceled: "Canceled",
    /**
     * Resource is being Provisioned.
     */
    Provisioning: "Provisioning",
    /**
     * Resource is being Updated.
     */
    Updating: "Updating",
    /**
     * Resource is being Deleted.
     */
    Deleting: "Deleting",
    /**
     * Resource is being Accepted.
     */
    Accepted: "Accepted",
};
exports.ReservedInstance = {
    None: "None",
    RI1Year: "RI1Year",
    RI3Year: "RI3Year",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.ResourceIdentityTypes = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.SavingsOption = {
    /**
     * Unknown Savings Option.
     */
    Unknown: "Unknown",
    /**
     * Reserved Instance 3 Year.
     */
    RI3Year: "RI3Year",
    /**
     * Azure Savings Plan 3 Year.
     */
    SavingsPlan3Year: "SavingsPlan3Year",
};
exports.SavingsOptions = {
    /**
     * No savings options.
     */
    None: "None",
    /**
     * Reserved Instance for 1 year.
     */
    RI1Year: "RI1Year",
    /**
     * Reserved Instance for 3 years.
     */
    RI3Year: "RI3Year",
    /**
     * The savings plan for 1 year.
     */
    SavingsPlan1Year: "SavingsPlan1Year",
    /**
     * The savings plan for 3  years.
     */
    SavingsPlan3Year: "SavingsPlan3Year",
    /**
     * Custom azure offer code.
     */
    CustomAzureOfferCode: "CustomAzureOfferCode",
};
exports.ScopeType = {
    /**
     * ServerGroupId - Scope type
     */
    ServerGroupId: "ServerGroupId",
    /**
     * AzureResourceGraphQuery - Scope type
     */
    AzureResourceGraphQuery: "AzureResourceGraphQuery",
};
exports.SecretStoreType = {
    None: "None",
    KubeSecret: "KubeSecret",
    KeyVaultSecret: "KeyVaultSecret",
    AppServiceAppSettings: "AppServiceAppSettings",
};
exports.SqlServerLicense = {
    /**
     * Unknown - Sql Server License
     */
    Unknown: "Unknown",
    /**
     * Has Sql Server License
     */
    Yes: "Yes",
    /**
     * No Sql Server License
     */
    No: "No",
};
exports.SqlServerLicenseType = {
    /**
     * Unknown Sql Server License.
     */
    Unknown: "Unknown",
    /**
     * Enterprise Sql Server License.
     */
    Enterprise: "Enterprise",
    /**
     * Standard Sql Server License.
     */
    Standard: "Standard",
};
exports.Status = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.TargetAvailabilityZone = {
    One: "1",
    Two: "2",
    Three: "3",
    NA: "NA",
};
exports.TargetHydrationStorageProviderType = {
    AzureFileShare: "AzureFileShare",
};
exports.TargetStorageAccessType = {
    Shared: "Shared",
    Exclusive: "Exclusive",
};
exports.TargetStorageProjectionType = {
    ContainerFileSystem: "ContainerFileSystem",
    PersistentVolume: "PersistentVolume",
};
exports.TimeRange = {
    /**
     * Daily.
     */
    Day: "Day",
    /**
     * Weekly.
     */
    Week: "Week",
    /**
     * Monthly.
     */
    Month: "Month",
    /**
     * Custom Time Range.
     */
    Custom: "Custom",
};
exports.WorkloadDeploymentTarget = {
    AzureKubernetesService: "AzureKubernetesService",
    AzureAppServiceContainer: "AzureAppServiceContainer",
    AzureAppServiceNative: "AzureAppServiceNative",
};
exports.WorkloadDeploymentType = {
    IISAKSWorkloadDeployment: "IISAKSWorkloadDeployment",
    ApacheTomcatAKSWorkloadDeployment: "ApacheTomcatAKSWorkloadDeployment",
};
exports.WorkloadType = {
    IISWorkload: "IISWorkload",
    ApacheTomcatWorkload: "ApacheTomcatWorkload",
};
exports.ZoneRedundant = {
    Enable: "Enable",
    Disable: "Disable",
};
