// @pulumi/azure-native/app/connectedEnvironmentsStorage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectedEnvironmentsStorage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Storage resource for connectedEnvironment.
 *
 * Uses Azure REST API version 2024-03-01. In version 2.x of the Azure Native provider, it used API version 2022-10-01.
 *
 * Other available API versions: 2022-10-01, 2022-11-01-preview, 2023-04-01-preview, 2023-05-01, 2023-05-02-preview, 2023-08-01-preview, 2023-11-02-preview, 2024-02-02-preview, 2024-08-02-preview, 2024-10-02-preview, 2025-01-01, 2025-02-02-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native app [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ConnectedEnvironmentsStorage extends pulumi.CustomResource {
    /**
     * Get an existing ConnectedEnvironmentsStorage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConnectedEnvironmentsStorage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConnectedEnvironmentsStorage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConnectedEnvironmentsStorage.__pulumiType;
    }
    /**
     * Create a ConnectedEnvironmentsStorage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.connectedEnvironmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectedEnvironmentName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["connectedEnvironmentName"] = args ? args.connectedEnvironmentName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storageName"] = args ? args.storageName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:app/v20220601preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20221001:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20221101preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20230401preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20230501:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20230502preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20230801preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20231102preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20240202preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20240301:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20240802preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20241002preview:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20250101:ConnectedEnvironmentsStorage" }, { type: "azure-native:app/v20250202preview:ConnectedEnvironmentsStorage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ConnectedEnvironmentsStorage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ConnectedEnvironmentsStorage = ConnectedEnvironmentsStorage;
/** @internal */
ConnectedEnvironmentsStorage.__pulumiType = 'azure-native:app:ConnectedEnvironmentsStorage';
