// @pulumi/azure-native/app/containerAppsAuthConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerAppsAuthConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Configuration settings for the Azure ContainerApp Service Authentication / Authorization feature.
 *
 * Uses Azure REST API version 2024-03-01. In version 2.x of the Azure Native provider, it used API version 2022-10-01.
 *
 * Other available API versions: 2022-10-01, 2022-11-01-preview, 2023-04-01-preview, 2023-05-01, 2023-05-02-preview, 2023-08-01-preview, 2023-11-02-preview, 2024-02-02-preview, 2024-08-02-preview, 2024-10-02-preview, 2025-01-01, 2025-02-02-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native app [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ContainerAppsAuthConfig extends pulumi.CustomResource {
    /**
     * Get an existing ContainerAppsAuthConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ContainerAppsAuthConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ContainerAppsAuthConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ContainerAppsAuthConfig.__pulumiType;
    }
    /**
     * Create a ContainerAppsAuthConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.containerAppName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerAppName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["authConfigName"] = args ? args.authConfigName : undefined;
            resourceInputs["containerAppName"] = args ? args.containerAppName : undefined;
            resourceInputs["encryptionSettings"] = args ? args.encryptionSettings : undefined;
            resourceInputs["globalValidation"] = args ? args.globalValidation : undefined;
            resourceInputs["httpSettings"] = args ? args.httpSettings : undefined;
            resourceInputs["identityProviders"] = args ? args.identityProviders : undefined;
            resourceInputs["login"] = args ? args.login : undefined;
            resourceInputs["platform"] = args ? args.platform : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["encryptionSettings"] = undefined /*out*/;
            resourceInputs["globalValidation"] = undefined /*out*/;
            resourceInputs["httpSettings"] = undefined /*out*/;
            resourceInputs["identityProviders"] = undefined /*out*/;
            resourceInputs["login"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["platform"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:app/v20220101preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20220301:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20220601preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20221001:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20221101preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20230401preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20230501:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20230502preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20230801preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20231102preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20240202preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20240301:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20240802preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20241002preview:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20250101:ContainerAppsAuthConfig" }, { type: "azure-native:app/v20250202preview:ContainerAppsAuthConfig" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ContainerAppsAuthConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ContainerAppsAuthConfig = ContainerAppsAuthConfig;
/** @internal */
ContainerAppsAuthConfig.__pulumiType = 'azure-native:app:ContainerAppsAuthConfig';
