// @pulumi/azure-native/app/daprComponent.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaprComponent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Dapr Component.
 *
 * Uses Azure REST API version 2024-03-01. In version 2.x of the Azure Native provider, it used API version 2022-10-01.
 *
 * Other available API versions: 2022-10-01, 2022-11-01-preview, 2023-04-01-preview, 2023-05-01, 2023-05-02-preview, 2023-08-01-preview, 2023-11-02-preview, 2024-02-02-preview, 2024-08-02-preview, 2024-10-02-preview, 2025-01-01, 2025-02-02-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native app [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DaprComponent extends pulumi.CustomResource {
    /**
     * Get an existing DaprComponent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DaprComponent(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DaprComponent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DaprComponent.__pulumiType;
    }
    /**
     * Create a DaprComponent resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.environmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["componentName"] = args ? args.componentName : undefined;
            resourceInputs["componentType"] = args ? args.componentType : undefined;
            resourceInputs["environmentName"] = args ? args.environmentName : undefined;
            resourceInputs["ignoreErrors"] = (_a = (args ? args.ignoreErrors : undefined)) !== null && _a !== void 0 ? _a : false;
            resourceInputs["initTimeout"] = args ? args.initTimeout : undefined;
            resourceInputs["metadata"] = args ? args.metadata : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scopes"] = args ? args.scopes : undefined;
            resourceInputs["secretStoreComponent"] = args ? args.secretStoreComponent : undefined;
            resourceInputs["secrets"] = args ? args.secrets : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["componentType"] = undefined /*out*/;
            resourceInputs["ignoreErrors"] = undefined /*out*/;
            resourceInputs["initTimeout"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["scopes"] = undefined /*out*/;
            resourceInputs["secretStoreComponent"] = undefined /*out*/;
            resourceInputs["secrets"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:app/v20220101preview:DaprComponent" }, { type: "azure-native:app/v20220301:DaprComponent" }, { type: "azure-native:app/v20220601preview:DaprComponent" }, { type: "azure-native:app/v20221001:DaprComponent" }, { type: "azure-native:app/v20221101preview:DaprComponent" }, { type: "azure-native:app/v20230401preview:DaprComponent" }, { type: "azure-native:app/v20230501:DaprComponent" }, { type: "azure-native:app/v20230502preview:DaprComponent" }, { type: "azure-native:app/v20230801preview:DaprComponent" }, { type: "azure-native:app/v20231102preview:DaprComponent" }, { type: "azure-native:app/v20240202preview:DaprComponent" }, { type: "azure-native:app/v20240301:DaprComponent" }, { type: "azure-native:app/v20240802preview:DaprComponent" }, { type: "azure-native:app/v20241002preview:DaprComponent" }, { type: "azure-native:app/v20250101:DaprComponent" }, { type: "azure-native:app/v20250202preview:DaprComponent" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DaprComponent.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DaprComponent = DaprComponent;
/** @internal */
DaprComponent.__pulumiType = 'azure-native:app:DaprComponent';
