// @pulumi/azure-native/types/enums/elastic/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagAction = exports.Status = exports.ProvisioningState = exports.Operation = exports.MonitoringStatus = exports.ManagedIdentityTypes = void 0;
exports.ManagedIdentityTypes = {
    SystemAssigned: "SystemAssigned",
};
exports.MonitoringStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.Operation = {
    AddBegin: "AddBegin",
    AddComplete: "AddComplete",
    DeleteBegin: "DeleteBegin",
    DeleteComplete: "DeleteComplete",
    Active: "Active",
};
exports.ProvisioningState = {
    Accepted: "Accepted",
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
    Deleted: "Deleted",
    NotSpecified: "NotSpecified",
};
exports.Status = {
    InProgress: "InProgress",
    Active: "Active",
    Failed: "Failed",
    Deleting: "Deleting",
};
exports.TagAction = {
    Include: "Include",
    Exclude: "Exclude",
};
