// @pulumi/azure-native/types/enums/dbformysql/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageRedundancyEnum = exports.StorageAutogrow = exports.SslEnforcementEnum = exports.SingleServerVersion = exports.SingleServerSkuTier = exports.ServerVersion = exports.ServerSkuTier = exports.ReplicationRole = exports.PublicNetworkAccessEnum = exports.PrivateEndpointServiceConnectionStatus = exports.MinimalTlsVersionEnum = exports.ManagedServiceIdentityType = exports.InfrastructureEncryption = exports.ImportSourceStorageType = exports.IdentityType = exports.HighAvailabilityMode = exports.GeoRedundantBackup = exports.EnableStatusEnum = exports.DataEncryptionType = exports.CreateMode = exports.ConfigurationSource = exports.AdministratorType = void 0;
exports.AdministratorType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.ConfigurationSource = {
    System_default: "system-default",
    User_override: "user-override",
};
exports.CreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    GeoRestore: "GeoRestore",
    Replica: "Replica",
};
exports.DataEncryptionType = {
    AzureKeyVault: "AzureKeyVault",
    SystemManaged: "SystemManaged",
};
exports.EnableStatusEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.GeoRedundantBackup = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HighAvailabilityMode = {
    Disabled: "Disabled",
    ZoneRedundant: "ZoneRedundant",
    SameZone: "SameZone",
};
exports.IdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.ImportSourceStorageType = {
    AzureBlob: "AzureBlob",
};
exports.InfrastructureEncryption = {
    /**
     * Default value for single layer of encryption for data at rest.
     */
    Enabled: "Enabled",
    /**
     * Additional (2nd) layer of encryption for data at rest
     */
    Disabled: "Disabled",
};
exports.ManagedServiceIdentityType = {
    UserAssigned: "UserAssigned",
};
exports.MinimalTlsVersionEnum = {
    TLS1_0: "TLS1_0",
    TLS1_1: "TLS1_1",
    TLS1_2: "TLS1_2",
    TLSEnforcementDisabled: "TLSEnforcementDisabled",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccessEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ReplicationRole = {
    None: "None",
    Source: "Source",
    Replica: "Replica",
};
exports.ServerSkuTier = {
    Burstable: "Burstable",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.ServerVersion = {
    ServerVersion_5_7: "5.7",
    ServerVersion_8_0_21: "8.0.21",
};
exports.SingleServerSkuTier = {
    Basic: "Basic",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.SingleServerVersion = {
    SingleServerVersion_5_6: "5.6",
    SingleServerVersion_5_7: "5.7",
    SingleServerVersion_8_0: "8.0",
};
exports.SslEnforcementEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageAutogrow = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageRedundancyEnum = {
    LocalRedundancy: "LocalRedundancy",
    ZoneRedundancy: "ZoneRedundancy",
};
