// @pulumi/azure-native/containerservice/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Namespace = exports.ManagedClusterSnapshot = exports.ManagedCluster = exports.MaintenanceConfiguration = exports.LoadBalancer = exports.listNamespaceCredentialOutput = exports.listNamespaceCredential = exports.listManagedClusterUserCredentialsOutput = exports.listManagedClusterUserCredentials = exports.listManagedClusterMonitoringUserCredentialsOutput = exports.listManagedClusterMonitoringUserCredentials = exports.listManagedClusterAdminCredentialsOutput = exports.listManagedClusterAdminCredentials = exports.listManagedClusterAccessProfileOutput = exports.listManagedClusterAccessProfile = exports.listFleetCredentialsOutput = exports.listFleetCredentials = exports.getUpdateRunOutput = exports.getUpdateRun = exports.getTrustedAccessRoleBindingOutput = exports.getTrustedAccessRoleBinding = exports.getSnapshotOutput = exports.getSnapshot = exports.getPrivateEndpointConnectionOutput = exports.getPrivateEndpointConnection = exports.getNamespaceOutput = exports.getNamespace = exports.getManagedClusterSnapshotOutput = exports.getManagedClusterSnapshot = exports.getManagedClusterOutput = exports.getManagedCluster = exports.getMaintenanceConfigurationOutput = exports.getMaintenanceConfiguration = exports.getLoadBalancerOutput = exports.getLoadBalancer = exports.getFleetUpdateStrategyOutput = exports.getFleetUpdateStrategy = exports.getFleetMemberOutput = exports.getFleetMember = exports.getFleetOutput = exports.getFleet = exports.getAutoUpgradeProfileOutput = exports.getAutoUpgradeProfile = exports.getAgentPoolOutput = exports.getAgentPool = exports.FleetUpdateStrategy = exports.FleetMember = exports.Fleet = exports.AutoUpgradeProfile = exports.AgentPool = void 0;
exports.UpdateRun = exports.TrustedAccessRoleBinding = exports.Snapshot = exports.PrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AgentPool = null;
utilities.lazyLoad(exports, ["AgentPool"], () => require("./agentPool"));
exports.AutoUpgradeProfile = null;
utilities.lazyLoad(exports, ["AutoUpgradeProfile"], () => require("./autoUpgradeProfile"));
exports.Fleet = null;
utilities.lazyLoad(exports, ["Fleet"], () => require("./fleet"));
exports.FleetMember = null;
utilities.lazyLoad(exports, ["FleetMember"], () => require("./fleetMember"));
exports.FleetUpdateStrategy = null;
utilities.lazyLoad(exports, ["FleetUpdateStrategy"], () => require("./fleetUpdateStrategy"));
exports.getAgentPool = null;
exports.getAgentPoolOutput = null;
utilities.lazyLoad(exports, ["getAgentPool", "getAgentPoolOutput"], () => require("./getAgentPool"));
exports.getAutoUpgradeProfile = null;
exports.getAutoUpgradeProfileOutput = null;
utilities.lazyLoad(exports, ["getAutoUpgradeProfile", "getAutoUpgradeProfileOutput"], () => require("./getAutoUpgradeProfile"));
exports.getFleet = null;
exports.getFleetOutput = null;
utilities.lazyLoad(exports, ["getFleet", "getFleetOutput"], () => require("./getFleet"));
exports.getFleetMember = null;
exports.getFleetMemberOutput = null;
utilities.lazyLoad(exports, ["getFleetMember", "getFleetMemberOutput"], () => require("./getFleetMember"));
exports.getFleetUpdateStrategy = null;
exports.getFleetUpdateStrategyOutput = null;
utilities.lazyLoad(exports, ["getFleetUpdateStrategy", "getFleetUpdateStrategyOutput"], () => require("./getFleetUpdateStrategy"));
exports.getLoadBalancer = null;
exports.getLoadBalancerOutput = null;
utilities.lazyLoad(exports, ["getLoadBalancer", "getLoadBalancerOutput"], () => require("./getLoadBalancer"));
exports.getMaintenanceConfiguration = null;
exports.getMaintenanceConfigurationOutput = null;
utilities.lazyLoad(exports, ["getMaintenanceConfiguration", "getMaintenanceConfigurationOutput"], () => require("./getMaintenanceConfiguration"));
exports.getManagedCluster = null;
exports.getManagedClusterOutput = null;
utilities.lazyLoad(exports, ["getManagedCluster", "getManagedClusterOutput"], () => require("./getManagedCluster"));
exports.getManagedClusterSnapshot = null;
exports.getManagedClusterSnapshotOutput = null;
utilities.lazyLoad(exports, ["getManagedClusterSnapshot", "getManagedClusterSnapshotOutput"], () => require("./getManagedClusterSnapshot"));
exports.getNamespace = null;
exports.getNamespaceOutput = null;
utilities.lazyLoad(exports, ["getNamespace", "getNamespaceOutput"], () => require("./getNamespace"));
exports.getPrivateEndpointConnection = null;
exports.getPrivateEndpointConnectionOutput = null;
utilities.lazyLoad(exports, ["getPrivateEndpointConnection", "getPrivateEndpointConnectionOutput"], () => require("./getPrivateEndpointConnection"));
exports.getSnapshot = null;
exports.getSnapshotOutput = null;
utilities.lazyLoad(exports, ["getSnapshot", "getSnapshotOutput"], () => require("./getSnapshot"));
exports.getTrustedAccessRoleBinding = null;
exports.getTrustedAccessRoleBindingOutput = null;
utilities.lazyLoad(exports, ["getTrustedAccessRoleBinding", "getTrustedAccessRoleBindingOutput"], () => require("./getTrustedAccessRoleBinding"));
exports.getUpdateRun = null;
exports.getUpdateRunOutput = null;
utilities.lazyLoad(exports, ["getUpdateRun", "getUpdateRunOutput"], () => require("./getUpdateRun"));
exports.listFleetCredentials = null;
exports.listFleetCredentialsOutput = null;
utilities.lazyLoad(exports, ["listFleetCredentials", "listFleetCredentialsOutput"], () => require("./listFleetCredentials"));
exports.listManagedClusterAccessProfile = null;
exports.listManagedClusterAccessProfileOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterAccessProfile", "listManagedClusterAccessProfileOutput"], () => require("./listManagedClusterAccessProfile"));
exports.listManagedClusterAdminCredentials = null;
exports.listManagedClusterAdminCredentialsOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterAdminCredentials", "listManagedClusterAdminCredentialsOutput"], () => require("./listManagedClusterAdminCredentials"));
exports.listManagedClusterMonitoringUserCredentials = null;
exports.listManagedClusterMonitoringUserCredentialsOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterMonitoringUserCredentials", "listManagedClusterMonitoringUserCredentialsOutput"], () => require("./listManagedClusterMonitoringUserCredentials"));
exports.listManagedClusterUserCredentials = null;
exports.listManagedClusterUserCredentialsOutput = null;
utilities.lazyLoad(exports, ["listManagedClusterUserCredentials", "listManagedClusterUserCredentialsOutput"], () => require("./listManagedClusterUserCredentials"));
exports.listNamespaceCredential = null;
exports.listNamespaceCredentialOutput = null;
utilities.lazyLoad(exports, ["listNamespaceCredential", "listNamespaceCredentialOutput"], () => require("./listNamespaceCredential"));
exports.LoadBalancer = null;
utilities.lazyLoad(exports, ["LoadBalancer"], () => require("./loadBalancer"));
exports.MaintenanceConfiguration = null;
utilities.lazyLoad(exports, ["MaintenanceConfiguration"], () => require("./maintenanceConfiguration"));
exports.ManagedCluster = null;
utilities.lazyLoad(exports, ["ManagedCluster"], () => require("./managedCluster"));
exports.ManagedClusterSnapshot = null;
utilities.lazyLoad(exports, ["ManagedClusterSnapshot"], () => require("./managedClusterSnapshot"));
exports.Namespace = null;
utilities.lazyLoad(exports, ["Namespace"], () => require("./namespace"));
exports.PrivateEndpointConnection = null;
utilities.lazyLoad(exports, ["PrivateEndpointConnection"], () => require("./privateEndpointConnection"));
exports.Snapshot = null;
utilities.lazyLoad(exports, ["Snapshot"], () => require("./snapshot"));
exports.TrustedAccessRoleBinding = null;
utilities.lazyLoad(exports, ["TrustedAccessRoleBinding"], () => require("./trustedAccessRoleBinding"));
exports.UpdateRun = null;
utilities.lazyLoad(exports, ["UpdateRun"], () => require("./updateRun"));
// Export enums:
__exportStar(require("../types/enums/containerservice"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerservice:AgentPool":
                return new exports.AgentPool(name, undefined, { urn });
            case "azure-native:containerservice:AutoUpgradeProfile":
                return new exports.AutoUpgradeProfile(name, undefined, { urn });
            case "azure-native:containerservice:Fleet":
                return new exports.Fleet(name, undefined, { urn });
            case "azure-native:containerservice:FleetMember":
                return new exports.FleetMember(name, undefined, { urn });
            case "azure-native:containerservice:FleetUpdateStrategy":
                return new exports.FleetUpdateStrategy(name, undefined, { urn });
            case "azure-native:containerservice:LoadBalancer":
                return new exports.LoadBalancer(name, undefined, { urn });
            case "azure-native:containerservice:MaintenanceConfiguration":
                return new exports.MaintenanceConfiguration(name, undefined, { urn });
            case "azure-native:containerservice:ManagedCluster":
                return new exports.ManagedCluster(name, undefined, { urn });
            case "azure-native:containerservice:ManagedClusterSnapshot":
                return new exports.ManagedClusterSnapshot(name, undefined, { urn });
            case "azure-native:containerservice:Namespace":
                return new exports.Namespace(name, undefined, { urn });
            case "azure-native:containerservice:PrivateEndpointConnection":
                return new exports.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:containerservice:Snapshot":
                return new exports.Snapshot(name, undefined, { urn });
            case "azure-native:containerservice:TrustedAccessRoleBinding":
                return new exports.TrustedAccessRoleBinding(name, undefined, { urn });
            case "azure-native:containerservice:UpdateRun":
                return new exports.UpdateRun(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerservice", _module);
