// @pulumi/azure-native/baremetalinfrastructure/azureBareMetalStorageInstance.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureBareMetalStorageInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * AzureBareMetalStorageInstance info on Azure (ARM properties and AzureBareMetalStorage properties)
 *
 * Uses Azure REST API version 2024-08-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-04-06.
 *
 * Other available API versions: 2023-04-06, 2023-08-04-preview, 2023-11-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native baremetalinfrastructure [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class AzureBareMetalStorageInstance extends pulumi.CustomResource {
    /**
     * Get an existing AzureBareMetalStorageInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AzureBareMetalStorageInstance(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AzureBareMetalStorageInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AzureBareMetalStorageInstance.__pulumiType;
    }
    /**
     * Create a AzureBareMetalStorageInstance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["azureBareMetalStorageInstanceName"] = args ? args.azureBareMetalStorageInstanceName : undefined;
            resourceInputs["azureBareMetalStorageInstanceUniqueIdentifier"] = args ? args.azureBareMetalStorageInstanceUniqueIdentifier : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storageProperties"] = args ? args.storageProperties : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["azureBareMetalStorageInstanceUniqueIdentifier"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["storageProperties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:baremetalinfrastructure/v20230406:AzureBareMetalStorageInstance" }, { type: "azure-native:baremetalinfrastructure/v20230804preview:AzureBareMetalStorageInstance" }, { type: "azure-native:baremetalinfrastructure/v20231101preview:AzureBareMetalStorageInstance" }, { type: "azure-native:baremetalinfrastructure/v20240801preview:AzureBareMetalStorageInstance" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AzureBareMetalStorageInstance.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AzureBareMetalStorageInstance = AzureBareMetalStorageInstance;
/** @internal */
AzureBareMetalStorageInstance.__pulumiType = 'azure-native:baremetalinfrastructure:AzureBareMetalStorageInstance';
