// @pulumi/azure-native/types/enums/dbforpostgresql/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualEndpointType = exports.TriggerCutoverEnum = exports.StorageType = exports.StorageAutogrow = exports.StorageAutoGrow = exports.StartDataMigrationEnum = exports.SslMode = exports.SslEnforcementEnum = exports.SourceType = exports.SkuTier = exports.SingleServerVersion = exports.SingleServerSkuTier = exports.SingleServerIdentityProperties = exports.SingleServerCreateMode = exports.ServerVersion = exports.ServerSecurityAlertPolicyState = exports.ServerPublicNetworkAccessState = exports.RoleType = exports.ReplicationRole = exports.ReplicationPromoteOption = exports.ReadReplicaPromoteMode = exports.PublicNetworkAccessEnum = exports.PrivateEndpointServiceConnectionStatus = exports.PrincipalType = exports.PasswordAuthEnum = exports.PasswordAuth = exports.OverwriteDbsInTargetEnum = exports.MinimalTlsVersionEnum = exports.MigrationOption = exports.MigrationMode = exports.MigrateRolesEnum = exports.LogicalReplicationOnSourceDbEnum = exports.KeyStatusEnum = exports.InfrastructureEncryption = exports.IdentityType = exports.HighAvailabilityMode = exports.GeoRedundantBackupEnum = exports.GeoRedundantBackup = exports.DataEncryptionType = exports.CreateMode = exports.CancelEnum = exports.AzureManagedDiskPerformanceTiers = exports.ArmServerKeyType = exports.AdministratorType = exports.ActiveDirectoryAuthEnum = exports.ActiveDirectoryAuth = void 0;
exports.ActiveDirectoryAuth = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ActiveDirectoryAuthEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.AdministratorType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.ArmServerKeyType = {
    SystemManaged: "SystemManaged",
    AzureKeyVault: "AzureKeyVault",
};
exports.AzureManagedDiskPerformanceTiers = {
    P1: "P1",
    P2: "P2",
    P3: "P3",
    P4: "P4",
    P6: "P6",
    P10: "P10",
    P15: "P15",
    P20: "P20",
    P30: "P30",
    P40: "P40",
    P50: "P50",
    P60: "P60",
    P70: "P70",
    P80: "P80",
};
exports.CancelEnum = {
    True: "True",
    False: "False",
};
exports.CreateMode = {
    Default: "Default",
    Create: "Create",
    Update: "Update",
    PointInTimeRestore: "PointInTimeRestore",
    GeoRestore: "GeoRestore",
    Replica: "Replica",
    ReviveDropped: "ReviveDropped",
};
exports.DataEncryptionType = {
    AzureKeyVault: "AzureKeyVault",
    SystemAssigned: "SystemAssigned",
};
exports.GeoRedundantBackup = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.GeoRedundantBackupEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HighAvailabilityMode = {
    Disabled: "Disabled",
    ZoneRedundant: "ZoneRedundant",
    SameZone: "SameZone",
};
exports.IdentityType = {
    UserAssigned: "UserAssigned",
    SystemAssigned: "SystemAssigned",
};
exports.InfrastructureEncryption = {
    /**
     * Default value for single layer of encryption for data at rest.
     */
    Enabled: "Enabled",
    /**
     * Additional (2nd) layer of encryption for data at rest
     */
    Disabled: "Disabled",
};
exports.KeyStatusEnum = {
    Valid: "Valid",
    Invalid: "Invalid",
};
exports.LogicalReplicationOnSourceDbEnum = {
    True: "True",
    False: "False",
};
exports.MigrateRolesEnum = {
    True: "True",
    False: "False",
};
exports.MigrationMode = {
    Offline: "Offline",
    Online: "Online",
};
exports.MigrationOption = {
    Validate: "Validate",
    Migrate: "Migrate",
    ValidateAndMigrate: "ValidateAndMigrate",
};
exports.MinimalTlsVersionEnum = {
    TLS1_0: "TLS1_0",
    TLS1_1: "TLS1_1",
    TLS1_2: "TLS1_2",
    TLSEnforcementDisabled: "TLSEnforcementDisabled",
};
exports.OverwriteDbsInTargetEnum = {
    True: "True",
    False: "False",
};
exports.PasswordAuth = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.PasswordAuthEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PrincipalType = {
    User: "user",
    ServicePrincipal: "servicePrincipal",
    Group: "group",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccessEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ReadReplicaPromoteMode = {
    Standalone: "standalone",
    Switchover: "switchover",
};
exports.ReplicationPromoteOption = {
    Planned: "planned",
    Forced: "forced",
};
exports.ReplicationRole = {
    None: "None",
    Primary: "Primary",
    AsyncReplica: "AsyncReplica",
    GeoAsyncReplica: "GeoAsyncReplica",
};
exports.RoleType = {
    User: "user",
    Admin: "admin",
};
exports.ServerPublicNetworkAccessState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ServerSecurityAlertPolicyState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ServerVersion = {
    ServerVersion_16: "16",
    ServerVersion_15: "15",
    ServerVersion_14: "14",
    ServerVersion_13: "13",
    ServerVersion_12: "12",
    ServerVersion_11: "11",
};
exports.SingleServerCreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    GeoRestore: "GeoRestore",
    Replica: "Replica",
};
exports.SingleServerIdentityProperties = {
    SystemAssigned: "SystemAssigned",
};
exports.SingleServerSkuTier = {
    Basic: "Basic",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.SingleServerVersion = {
    SingleServerVersion_9_5: "9.5",
    SingleServerVersion_9_6: "9.6",
    SingleServerVersion_10: "10",
    SingleServerVersion_10_0: "10.0",
    SingleServerVersion_10_2: "10.2",
    SingleServerVersion_11: "11",
};
exports.SkuTier = {
    Burstable: "Burstable",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.SourceType = {
    OnPremises: "OnPremises",
    AWS: "AWS",
    GCP: "GCP",
    AzureVM: "AzureVM",
    PostgreSQLSingleServer: "PostgreSQLSingleServer",
    AWS_RDS: "AWS_RDS",
    AWS_AURORA: "AWS_AURORA",
    AWS_EC2: "AWS_EC2",
    GCP_CloudSQL: "GCP_CloudSQL",
    GCP_AlloyDB: "GCP_AlloyDB",
    GCP_Compute: "GCP_Compute",
    EDB: "EDB",
};
exports.SslEnforcementEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SslMode = {
    Prefer: "Prefer",
    Require: "Require",
    VerifyCA: "VerifyCA",
    VerifyFull: "VerifyFull",
};
exports.StartDataMigrationEnum = {
    True: "True",
    False: "False",
};
exports.StorageAutoGrow = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageAutogrow = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageType = {
    Premium_LRS: "Premium_LRS",
    PremiumV2_LRS: "PremiumV2_LRS",
};
exports.TriggerCutoverEnum = {
    True: "True",
    False: "False",
};
exports.VirtualEndpointType = {
    ReadWrite: "ReadWrite",
};
