// @pulumi/azure-native/iotoperationsmq/broker.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Broker = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * MQ broker resource
 *
 * Uses Azure REST API version 2023-10-04-preview. In version 2.x of the Azure Native provider, it used API version 2023-10-04-preview.
 */
class Broker extends pulumi.CustomResource {
    /**
     * Get an existing Broker resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Broker(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Broker.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Broker.__pulumiType;
    }
    /**
     * Create a Broker resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.authImage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authImage'");
            }
            if ((!args || args.brokerImage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'brokerImage'");
            }
            if ((!args || args.extendedLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'extendedLocation'");
            }
            if ((!args || args.healthManagerImage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'healthManagerImage'");
            }
            if ((!args || args.mode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mode'");
            }
            if ((!args || args.mqName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mqName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["authImage"] = args ? args.authImage : undefined;
            resourceInputs["brokerImage"] = args ? args.brokerImage : undefined;
            resourceInputs["brokerName"] = args ? args.brokerName : undefined;
            resourceInputs["brokerNodeTolerations"] = args ? args.brokerNodeTolerations : undefined;
            resourceInputs["cardinality"] = args ? (args.cardinality ? pulumi.output(args.cardinality).apply(inputs.iotoperationsmq.cardinalityArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["diagnostics"] = args ? (args.diagnostics ? pulumi.output(args.diagnostics).apply(inputs.iotoperationsmq.brokerDiagnosticsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["diskBackedMessageBufferSettings"] = args ? args.diskBackedMessageBufferSettings : undefined;
            resourceInputs["encryptInternalTraffic"] = (_a = (args ? args.encryptInternalTraffic : undefined)) !== null && _a !== void 0 ? _a : true;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["healthManagerImage"] = args ? args.healthManagerImage : undefined;
            resourceInputs["healthManagerNodeTolerations"] = args ? args.healthManagerNodeTolerations : undefined;
            resourceInputs["internalCerts"] = args ? (args.internalCerts ? pulumi.output(args.internalCerts).apply(inputs.iotoperationsmq.certManagerCertOptionsArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["memoryProfile"] = (_b = (args ? args.memoryProfile : undefined)) !== null && _b !== void 0 ? _b : "medium";
            resourceInputs["mode"] = args ? args.mode : undefined;
            resourceInputs["mqName"] = args ? args.mqName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["authImage"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["brokerImage"] = undefined /*out*/;
            resourceInputs["brokerNodeTolerations"] = undefined /*out*/;
            resourceInputs["cardinality"] = undefined /*out*/;
            resourceInputs["diagnostics"] = undefined /*out*/;
            resourceInputs["diskBackedMessageBufferSettings"] = undefined /*out*/;
            resourceInputs["encryptInternalTraffic"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["healthManagerImage"] = undefined /*out*/;
            resourceInputs["healthManagerNodeTolerations"] = undefined /*out*/;
            resourceInputs["internalCerts"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["memoryProfile"] = undefined /*out*/;
            resourceInputs["mode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:iotoperationsmq/v20231004preview:Broker" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Broker.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Broker = Broker;
/** @internal */
Broker.__pulumiType = 'azure-native:iotoperationsmq:Broker';
