// @pulumi/azure-native/edge/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SitesBySubscription = exports.SitesByServiceGroup = exports.Site = exports.getSitesBySubscriptionOutput = exports.getSitesBySubscription = exports.getSitesByServiceGroupOutput = exports.getSitesByServiceGroup = exports.getSiteOutput = exports.getSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getSite = null;
exports.getSiteOutput = null;
utilities.lazyLoad(exports, ["getSite", "getSiteOutput"], () => require("./getSite"));
exports.getSitesByServiceGroup = null;
exports.getSitesByServiceGroupOutput = null;
utilities.lazyLoad(exports, ["getSitesByServiceGroup", "getSitesByServiceGroupOutput"], () => require("./getSitesByServiceGroup"));
exports.getSitesBySubscription = null;
exports.getSitesBySubscriptionOutput = null;
utilities.lazyLoad(exports, ["getSitesBySubscription", "getSitesBySubscriptionOutput"], () => require("./getSitesBySubscription"));
exports.Site = null;
utilities.lazyLoad(exports, ["Site"], () => require("./site"));
exports.SitesByServiceGroup = null;
utilities.lazyLoad(exports, ["SitesByServiceGroup"], () => require("./sitesByServiceGroup"));
exports.SitesBySubscription = null;
utilities.lazyLoad(exports, ["SitesBySubscription"], () => require("./sitesBySubscription"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:edge:Site":
                return new exports.Site(name, undefined, { urn });
            case "azure-native:edge:SitesByServiceGroup":
                return new exports.SitesByServiceGroup(name, undefined, { urn });
            case "azure-native:edge:SitesBySubscription":
                return new exports.SitesBySubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "edge", _module);
