// @pulumi/azure-native/edge/sitesByServiceGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SitesByServiceGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Site as ARM Resource
 *
 * Uses Azure REST API version 2025-03-01-preview.
 */
class SitesByServiceGroup extends pulumi.CustomResource {
    /**
     * Get an existing SitesByServiceGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SitesByServiceGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SitesByServiceGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SitesByServiceGroup.__pulumiType;
    }
    /**
     * Create a SitesByServiceGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.servicegroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'servicegroupName'");
            }
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["servicegroupName"] = args ? args.servicegroupName : undefined;
            resourceInputs["siteName"] = args ? args.siteName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:edge/v20250301preview:SitesByServiceGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SitesByServiceGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SitesByServiceGroup = SitesByServiceGroup;
/** @internal */
SitesByServiceGroup.__pulumiType = 'azure-native:edge:SitesByServiceGroup';
