// @pulumi/azure-native/recoveryservices/protectionIntent.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtectionIntent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Base class for backup ProtectionIntent.
 *
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2023-02-01, 2023-04-01, 2023-06-01, 2023-08-01, 2024-01-01, 2024-02-01, 2024-04-01, 2024-04-30-preview, 2024-07-30-preview, 2024-11-01-preview, 2025-01-01, 2025-02-01, 2025-02-28-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native recoveryservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ProtectionIntent extends pulumi.CustomResource {
    /**
     * Get an existing ProtectionIntent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ProtectionIntent(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ProtectionIntent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ProtectionIntent.__pulumiType;
    }
    /**
     * Create a ProtectionIntent resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vaultName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vaultName'");
            }
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["intentObjectName"] = args ? args.intentObjectName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["vaultName"] = args ? args.vaultName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20170701:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210201preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210210:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210301:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210401:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210601:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210701:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210801:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20211001:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20211201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220101:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220301:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220401:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220601preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220901preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220930preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20221001:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20230101:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20230201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20230401:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20230601:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20230801:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20240101:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20240201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20240401:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20240430preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20240730preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20241001:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20241101preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20250101:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20250201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20250228preview:ProtectionIntent" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ProtectionIntent.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ProtectionIntent = ProtectionIntent;
/** @internal */
ProtectionIntent.__pulumiType = 'azure-native:recoveryservices:ProtectionIntent';
