// @pulumi/azure-native/recoveryservices/replicationFabric.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationFabric = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Fabric definition.
 *
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2023-02-01, 2023-04-01, 2023-06-01, 2023-08-01, 2024-01-01, 2024-02-01, 2024-04-01, 2025-01-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native recoveryservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ReplicationFabric extends pulumi.CustomResource {
    /**
     * Get an existing ReplicationFabric resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReplicationFabric(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationFabric.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationFabric.__pulumiType;
    }
    /**
     * Create a ReplicationFabric resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices/v20160810:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20180110:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20180710:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20210210:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20210301:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20210401:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20210601:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20210701:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20210801:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20211001:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20211101:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20211201:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20220101:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20220201:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20220301:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20220401:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20220501:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20220801:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20220910:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20221001:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20230101:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20230201:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20230401:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20230601:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20230801:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20240101:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20240201:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20240401:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20241001:ReplicationFabric" }, { type: "azure-native:recoveryservices/v20250101:ReplicationFabric" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationFabric.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ReplicationFabric = ReplicationFabric;
/** @internal */
ReplicationFabric.__pulumiType = 'azure-native:recoveryservices:ReplicationFabric';
