// @pulumi/azure-native/durabletask/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskHub = exports.Scheduler = exports.RetentionPolicy = exports.getTaskHubOutput = exports.getTaskHub = exports.getSchedulerOutput = exports.getScheduler = exports.getRetentionPolicyOutput = exports.getRetentionPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getRetentionPolicy = null;
exports.getRetentionPolicyOutput = null;
utilities.lazyLoad(exports, ["getRetentionPolicy", "getRetentionPolicyOutput"], () => require("./getRetentionPolicy"));
exports.getScheduler = null;
exports.getSchedulerOutput = null;
utilities.lazyLoad(exports, ["getScheduler", "getSchedulerOutput"], () => require("./getScheduler"));
exports.getTaskHub = null;
exports.getTaskHubOutput = null;
utilities.lazyLoad(exports, ["getTaskHub", "getTaskHubOutput"], () => require("./getTaskHub"));
exports.RetentionPolicy = null;
utilities.lazyLoad(exports, ["RetentionPolicy"], () => require("./retentionPolicy"));
exports.Scheduler = null;
utilities.lazyLoad(exports, ["Scheduler"], () => require("./scheduler"));
exports.TaskHub = null;
utilities.lazyLoad(exports, ["TaskHub"], () => require("./taskHub"));
// Export enums:
__exportStar(require("../types/enums/durabletask"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:durabletask:RetentionPolicy":
                return new exports.RetentionPolicy(name, undefined, { urn });
            case "azure-native:durabletask:Scheduler":
                return new exports.Scheduler(name, undefined, { urn });
            case "azure-native:durabletask:TaskHub":
                return new exports.TaskHub(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "durabletask", _module);
