// @pulumi/azure-native/servicefabric/managedClusterApplicationTypeVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedClusterApplicationTypeVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An application type version resource for the specified application type name resource.
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2023-03-01-preview.
 *
 * Other available API versions: 2023-03-01-preview, 2023-07-01-preview, 2023-09-01-preview, 2023-11-01-preview, 2023-12-01-preview, 2024-02-01-preview, 2024-06-01-preview, 2024-09-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native servicefabric [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ManagedClusterApplicationTypeVersion extends pulumi.CustomResource {
    /**
     * Get an existing ManagedClusterApplicationTypeVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagedClusterApplicationTypeVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedClusterApplicationTypeVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedClusterApplicationTypeVersion.__pulumiType;
    }
    /**
     * Create a ManagedClusterApplicationTypeVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.appPackageUrl === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appPackageUrl'");
            }
            if ((!args || args.applicationTypeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationTypeName'");
            }
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["appPackageUrl"] = args ? args.appPackageUrl : undefined;
            resourceInputs["applicationTypeName"] = args ? args.applicationTypeName : undefined;
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["appPackageUrl"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:servicefabric/v20210101preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20210501:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20210701preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20210901privatepreview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20211101preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20220101:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20220201preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20220601preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20220801preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20221001preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20230201preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20230301preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20230701preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20230901preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20231101preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20231201preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20240201preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20240401:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20240601preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20240901preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20241101preview:ManagedClusterApplicationTypeVersion" }, { type: "azure-native:servicefabric/v20250301preview:ManagedClusterApplicationTypeVersion" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedClusterApplicationTypeVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ManagedClusterApplicationTypeVersion = ManagedClusterApplicationTypeVersion;
/** @internal */
ManagedClusterApplicationTypeVersion.__pulumiType = 'azure-native:servicefabric:ManagedClusterApplicationTypeVersion';
