// @pulumi/azure-native/frontdoor/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RulesEngine = exports.Policy = exports.NetworkExperimentProfile = exports.getRulesEngineOutput = exports.getRulesEngine = exports.getPolicyOutput = exports.getPolicy = exports.getNetworkExperimentProfileOutput = exports.getNetworkExperimentProfile = exports.getFrontDoorOutput = exports.getFrontDoor = exports.getExperimentOutput = exports.getExperiment = exports.FrontDoor = exports.Experiment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Experiment = null;
utilities.lazyLoad(exports, ["Experiment"], () => require("./experiment"));
exports.FrontDoor = null;
utilities.lazyLoad(exports, ["FrontDoor"], () => require("./frontDoor"));
exports.getExperiment = null;
exports.getExperimentOutput = null;
utilities.lazyLoad(exports, ["getExperiment", "getExperimentOutput"], () => require("./getExperiment"));
exports.getFrontDoor = null;
exports.getFrontDoorOutput = null;
utilities.lazyLoad(exports, ["getFrontDoor", "getFrontDoorOutput"], () => require("./getFrontDoor"));
exports.getNetworkExperimentProfile = null;
exports.getNetworkExperimentProfileOutput = null;
utilities.lazyLoad(exports, ["getNetworkExperimentProfile", "getNetworkExperimentProfileOutput"], () => require("./getNetworkExperimentProfile"));
exports.getPolicy = null;
exports.getPolicyOutput = null;
utilities.lazyLoad(exports, ["getPolicy", "getPolicyOutput"], () => require("./getPolicy"));
exports.getRulesEngine = null;
exports.getRulesEngineOutput = null;
utilities.lazyLoad(exports, ["getRulesEngine", "getRulesEngineOutput"], () => require("./getRulesEngine"));
exports.NetworkExperimentProfile = null;
utilities.lazyLoad(exports, ["NetworkExperimentProfile"], () => require("./networkExperimentProfile"));
exports.Policy = null;
utilities.lazyLoad(exports, ["Policy"], () => require("./policy"));
exports.RulesEngine = null;
utilities.lazyLoad(exports, ["RulesEngine"], () => require("./rulesEngine"));
// Export enums:
__exportStar(require("../types/enums/frontdoor"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:frontdoor:Experiment":
                return new exports.Experiment(name, undefined, { urn });
            case "azure-native:frontdoor:FrontDoor":
                return new exports.FrontDoor(name, undefined, { urn });
            case "azure-native:frontdoor:NetworkExperimentProfile":
                return new exports.NetworkExperimentProfile(name, undefined, { urn });
            case "azure-native:frontdoor:Policy":
                return new exports.Policy(name, undefined, { urn });
            case "azure-native:frontdoor:RulesEngine":
                return new exports.RulesEngine(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "frontdoor", _module);
