// @pulumi/azure-native/communication/smtpUsername.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmtpUsername = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The object describing the smtp username resource.
 *
 * Uses Azure REST API version 2024-09-01-preview.
 */
class SmtpUsername extends pulumi.CustomResource {
    /**
     * Get an existing SmtpUsername resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SmtpUsername(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SmtpUsername.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SmtpUsername.__pulumiType;
    }
    /**
     * Create a SmtpUsername resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.communicationServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'communicationServiceName'");
            }
            if ((!args || args.entraApplicationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'entraApplicationId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.tenantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tenantId'");
            }
            if ((!args || args.username === undefined) && !opts.urn) {
                throw new Error("Missing required property 'username'");
            }
            resourceInputs["communicationServiceName"] = args ? args.communicationServiceName : undefined;
            resourceInputs["entraApplicationId"] = args ? args.entraApplicationId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["smtpUsername"] = args ? args.smtpUsername : undefined;
            resourceInputs["tenantId"] = args ? args.tenantId : undefined;
            resourceInputs["username"] = args ? args.username : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["entraApplicationId"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["username"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:communication/v20240901preview:SmtpUsername" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SmtpUsername.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SmtpUsername = SmtpUsername;
/** @internal */
SmtpUsername.__pulumiType = 'azure-native:communication:SmtpUsername';
