// @pulumi/azure-native/sovereign/landingZoneRegistrationOperation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LandingZoneRegistrationOperation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Landing zone registration resource type.
 *
 * Uses Azure REST API version 2025-02-27-preview. In version 2.x of the Azure Native provider, it used API version 2025-02-27-preview.
 */
class LandingZoneRegistrationOperation extends pulumi.CustomResource {
    /**
     * Get an existing LandingZoneRegistrationOperation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LandingZoneRegistrationOperation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LandingZoneRegistrationOperation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LandingZoneRegistrationOperation.__pulumiType;
    }
    /**
     * Create a LandingZoneRegistrationOperation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.landingZoneAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'landingZoneAccountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["landingZoneAccountName"] = args ? args.landingZoneAccountName : undefined;
            resourceInputs["landingZoneRegistrationName"] = args ? args.landingZoneRegistrationName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sovereign/v20250227preview:LandingZoneRegistrationOperation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(LandingZoneRegistrationOperation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LandingZoneRegistrationOperation = LandingZoneRegistrationOperation;
/** @internal */
LandingZoneRegistrationOperation.__pulumiType = 'azure-native:sovereign:LandingZoneRegistrationOperation';
