// @pulumi/azure-native/storage/blobServiceProperties.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlobServiceProperties = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The properties of a storage account’s Blob service.
 *
 * Uses Azure REST API version 2024-01-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2022-09-01, 2023-01-01, 2023-04-01, 2023-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native storage [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class BlobServiceProperties extends pulumi.CustomResource {
    /**
     * Get an existing BlobServiceProperties resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BlobServiceProperties(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BlobServiceProperties.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BlobServiceProperties.__pulumiType;
    }
    /**
     * Create a BlobServiceProperties resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["automaticSnapshotPolicyEnabled"] = args ? args.automaticSnapshotPolicyEnabled : undefined;
            resourceInputs["blobServicesName"] = args ? args.blobServicesName : undefined;
            resourceInputs["changeFeed"] = args ? args.changeFeed : undefined;
            resourceInputs["containerDeleteRetentionPolicy"] = args ? args.containerDeleteRetentionPolicy : undefined;
            resourceInputs["cors"] = args ? args.cors : undefined;
            resourceInputs["defaultServiceVersion"] = args ? args.defaultServiceVersion : undefined;
            resourceInputs["deleteRetentionPolicy"] = args ? args.deleteRetentionPolicy : undefined;
            resourceInputs["isVersioningEnabled"] = args ? args.isVersioningEnabled : undefined;
            resourceInputs["lastAccessTimeTrackingPolicy"] = args ? args.lastAccessTimeTrackingPolicy : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["restorePolicy"] = args ? args.restorePolicy : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["automaticSnapshotPolicyEnabled"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["changeFeed"] = undefined /*out*/;
            resourceInputs["containerDeleteRetentionPolicy"] = undefined /*out*/;
            resourceInputs["cors"] = undefined /*out*/;
            resourceInputs["defaultServiceVersion"] = undefined /*out*/;
            resourceInputs["deleteRetentionPolicy"] = undefined /*out*/;
            resourceInputs["isVersioningEnabled"] = undefined /*out*/;
            resourceInputs["lastAccessTimeTrackingPolicy"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["restorePolicy"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storage/v20180701:BlobServiceProperties" }, { type: "azure-native:storage/v20181101:BlobServiceProperties" }, { type: "azure-native:storage/v20190401:BlobServiceProperties" }, { type: "azure-native:storage/v20190601:BlobServiceProperties" }, { type: "azure-native:storage/v20200801preview:BlobServiceProperties" }, { type: "azure-native:storage/v20210101:BlobServiceProperties" }, { type: "azure-native:storage/v20210201:BlobServiceProperties" }, { type: "azure-native:storage/v20210401:BlobServiceProperties" }, { type: "azure-native:storage/v20210601:BlobServiceProperties" }, { type: "azure-native:storage/v20210801:BlobServiceProperties" }, { type: "azure-native:storage/v20210901:BlobServiceProperties" }, { type: "azure-native:storage/v20220501:BlobServiceProperties" }, { type: "azure-native:storage/v20220901:BlobServiceProperties" }, { type: "azure-native:storage/v20230101:BlobServiceProperties" }, { type: "azure-native:storage/v20230401:BlobServiceProperties" }, { type: "azure-native:storage/v20230501:BlobServiceProperties" }, { type: "azure-native:storage/v20240101:BlobServiceProperties" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BlobServiceProperties.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BlobServiceProperties = BlobServiceProperties;
/** @internal */
BlobServiceProperties.__pulumiType = 'azure-native:storage:BlobServiceProperties';
