// @pulumi/azure-native/storage/fileShare.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileShare = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Properties of the file share, including Id, resource name, resource type, Etag.
 *
 * Uses Azure REST API version 2024-01-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2022-09-01, 2023-01-01, 2023-04-01, 2023-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native storage [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class FileShare extends pulumi.CustomResource {
    /**
     * Get an existing FileShare resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FileShare(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FileShare.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FileShare.__pulumiType;
    }
    /**
     * Create a FileShare resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accessTier"] = args ? args.accessTier : undefined;
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["enabledProtocols"] = args ? args.enabledProtocols : undefined;
            resourceInputs["expand"] = args ? args.expand : undefined;
            resourceInputs["fileSharePaidBursting"] = args ? args.fileSharePaidBursting : undefined;
            resourceInputs["metadata"] = args ? args.metadata : undefined;
            resourceInputs["provisionedBandwidthMibps"] = args ? args.provisionedBandwidthMibps : undefined;
            resourceInputs["provisionedIops"] = args ? args.provisionedIops : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["rootSquash"] = args ? args.rootSquash : undefined;
            resourceInputs["shareName"] = args ? args.shareName : undefined;
            resourceInputs["shareQuota"] = args ? args.shareQuota : undefined;
            resourceInputs["signedIdentifiers"] = args ? args.signedIdentifiers : undefined;
            resourceInputs["accessTierChangeTime"] = undefined /*out*/;
            resourceInputs["accessTierStatus"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deleted"] = undefined /*out*/;
            resourceInputs["deletedTime"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["includedBurstIops"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["leaseDuration"] = undefined /*out*/;
            resourceInputs["leaseState"] = undefined /*out*/;
            resourceInputs["leaseStatus"] = undefined /*out*/;
            resourceInputs["maxBurstCreditsForIops"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nextAllowedProvisionedBandwidthDowngradeTime"] = undefined /*out*/;
            resourceInputs["nextAllowedProvisionedIopsDowngradeTime"] = undefined /*out*/;
            resourceInputs["nextAllowedQuotaDowngradeTime"] = undefined /*out*/;
            resourceInputs["remainingRetentionDays"] = undefined /*out*/;
            resourceInputs["shareUsageBytes"] = undefined /*out*/;
            resourceInputs["snapshotTime"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessTier"] = undefined /*out*/;
            resourceInputs["accessTierChangeTime"] = undefined /*out*/;
            resourceInputs["accessTierStatus"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deleted"] = undefined /*out*/;
            resourceInputs["deletedTime"] = undefined /*out*/;
            resourceInputs["enabledProtocols"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fileSharePaidBursting"] = undefined /*out*/;
            resourceInputs["includedBurstIops"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["leaseDuration"] = undefined /*out*/;
            resourceInputs["leaseState"] = undefined /*out*/;
            resourceInputs["leaseStatus"] = undefined /*out*/;
            resourceInputs["maxBurstCreditsForIops"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nextAllowedProvisionedBandwidthDowngradeTime"] = undefined /*out*/;
            resourceInputs["nextAllowedProvisionedIopsDowngradeTime"] = undefined /*out*/;
            resourceInputs["nextAllowedQuotaDowngradeTime"] = undefined /*out*/;
            resourceInputs["provisionedBandwidthMibps"] = undefined /*out*/;
            resourceInputs["provisionedIops"] = undefined /*out*/;
            resourceInputs["remainingRetentionDays"] = undefined /*out*/;
            resourceInputs["rootSquash"] = undefined /*out*/;
            resourceInputs["shareQuota"] = undefined /*out*/;
            resourceInputs["shareUsageBytes"] = undefined /*out*/;
            resourceInputs["signedIdentifiers"] = undefined /*out*/;
            resourceInputs["snapshotTime"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storage/v20190401:FileShare" }, { type: "azure-native:storage/v20190601:FileShare" }, { type: "azure-native:storage/v20200801preview:FileShare" }, { type: "azure-native:storage/v20210101:FileShare" }, { type: "azure-native:storage/v20210201:FileShare" }, { type: "azure-native:storage/v20210401:FileShare" }, { type: "azure-native:storage/v20210601:FileShare" }, { type: "azure-native:storage/v20210801:FileShare" }, { type: "azure-native:storage/v20210901:FileShare" }, { type: "azure-native:storage/v20220501:FileShare" }, { type: "azure-native:storage/v20220901:FileShare" }, { type: "azure-native:storage/v20230101:FileShare" }, { type: "azure-native:storage/v20230401:FileShare" }, { type: "azure-native:storage/v20230501:FileShare" }, { type: "azure-native:storage/v20240101:FileShare" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FileShare.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FileShare = FileShare;
/** @internal */
FileShare.__pulumiType = 'azure-native:storage:FileShare';
