// @pulumi/azure-native/storage/tableServiceProperties.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableServiceProperties = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The properties of a storage account’s Table service.
 *
 * Uses Azure REST API version 2024-01-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2022-09-01, 2023-01-01, 2023-04-01, 2023-05-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native storage [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class TableServiceProperties extends pulumi.CustomResource {
    /**
     * Get an existing TableServiceProperties resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TableServiceProperties(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TableServiceProperties.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TableServiceProperties.__pulumiType;
    }
    /**
     * Create a TableServiceProperties resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["cors"] = args ? args.cors : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tableServiceName"] = args ? args.tableServiceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cors"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storage/v20190601:TableServiceProperties" }, { type: "azure-native:storage/v20200801preview:TableServiceProperties" }, { type: "azure-native:storage/v20210101:TableServiceProperties" }, { type: "azure-native:storage/v20210201:TableServiceProperties" }, { type: "azure-native:storage/v20210401:TableServiceProperties" }, { type: "azure-native:storage/v20210601:TableServiceProperties" }, { type: "azure-native:storage/v20210801:TableServiceProperties" }, { type: "azure-native:storage/v20210901:TableServiceProperties" }, { type: "azure-native:storage/v20220501:TableServiceProperties" }, { type: "azure-native:storage/v20220901:TableServiceProperties" }, { type: "azure-native:storage/v20230101:TableServiceProperties" }, { type: "azure-native:storage/v20230401:TableServiceProperties" }, { type: "azure-native:storage/v20230501:TableServiceProperties" }, { type: "azure-native:storage/v20240101:TableServiceProperties" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(TableServiceProperties.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TableServiceProperties = TableServiceProperties;
/** @internal */
TableServiceProperties.__pulumiType = 'azure-native:storage:TableServiceProperties';
