// @pulumi/azure-native/logic/rosettaNetProcessConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RosettaNetProcessConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The integration account RosettaNet process configuration.
 *
 * Uses Azure REST API version 2016-06-01. In version 2.x of the Azure Native provider, it used API version 2016-06-01.
 */
class RosettaNetProcessConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing RosettaNetProcessConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RosettaNetProcessConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RosettaNetProcessConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RosettaNetProcessConfiguration.__pulumiType;
    }
    /**
     * Create a RosettaNetProcessConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.activitySettings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'activitySettings'");
            }
            if ((!args || args.initiatorRoleSettings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'initiatorRoleSettings'");
            }
            if ((!args || args.integrationAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integrationAccountName'");
            }
            if ((!args || args.processCode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'processCode'");
            }
            if ((!args || args.processName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'processName'");
            }
            if ((!args || args.processVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'processVersion'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.responderRoleSettings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'responderRoleSettings'");
            }
            resourceInputs["activitySettings"] = args ? args.activitySettings : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["initiatorRoleSettings"] = args ? args.initiatorRoleSettings : undefined;
            resourceInputs["integrationAccountName"] = args ? args.integrationAccountName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["metadata"] = args ? args.metadata : undefined;
            resourceInputs["processCode"] = args ? args.processCode : undefined;
            resourceInputs["processName"] = args ? args.processName : undefined;
            resourceInputs["processVersion"] = args ? args.processVersion : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["responderRoleSettings"] = args ? args.responderRoleSettings : undefined;
            resourceInputs["rosettaNetProcessConfigurationName"] = args ? args.rosettaNetProcessConfigurationName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["changedTime"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["activitySettings"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["changedTime"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["initiatorRoleSettings"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["processCode"] = undefined /*out*/;
            resourceInputs["processName"] = undefined /*out*/;
            resourceInputs["processVersion"] = undefined /*out*/;
            resourceInputs["responderRoleSettings"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:logic/v20160601:RosettaNetProcessConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RosettaNetProcessConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RosettaNetProcessConfiguration = RosettaNetProcessConfiguration;
/** @internal */
RosettaNetProcessConfiguration.__pulumiType = 'azure-native:logic:RosettaNetProcessConfiguration';
