// @pulumi/azure-native/dbforpostgresql/serverGroupCluster.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerGroupCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a cluster.
 *
 * Uses Azure REST API version 2023-03-02-preview.
 *
 * Other available API versions: 2022-11-08. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native dbforpostgresql [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ServerGroupCluster extends pulumi.CustomResource {
    /**
     * Get an existing ServerGroupCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServerGroupCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServerGroupCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerGroupCluster.__pulumiType;
    }
    /**
     * Create a ServerGroupCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["administratorLoginPassword"] = args ? args.administratorLoginPassword : undefined;
            resourceInputs["authConfig"] = args ? args.authConfig : undefined;
            resourceInputs["citusVersion"] = args ? args.citusVersion : undefined;
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["coordinatorEnablePublicIpAccess"] = args ? args.coordinatorEnablePublicIpAccess : undefined;
            resourceInputs["coordinatorServerEdition"] = args ? args.coordinatorServerEdition : undefined;
            resourceInputs["coordinatorStorageQuotaInMb"] = args ? args.coordinatorStorageQuotaInMb : undefined;
            resourceInputs["coordinatorVCores"] = args ? args.coordinatorVCores : undefined;
            resourceInputs["dataEncryption"] = args ? args.dataEncryption : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["enableGeoBackup"] = args ? args.enableGeoBackup : undefined;
            resourceInputs["enableHa"] = args ? args.enableHa : undefined;
            resourceInputs["enableShardsOnCoordinator"] = args ? args.enableShardsOnCoordinator : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["maintenanceWindow"] = args ? args.maintenanceWindow : undefined;
            resourceInputs["nodeCount"] = args ? args.nodeCount : undefined;
            resourceInputs["nodeEnablePublicIpAccess"] = args ? args.nodeEnablePublicIpAccess : undefined;
            resourceInputs["nodeServerEdition"] = args ? args.nodeServerEdition : undefined;
            resourceInputs["nodeStorageQuotaInMb"] = args ? args.nodeStorageQuotaInMb : undefined;
            resourceInputs["nodeVCores"] = args ? args.nodeVCores : undefined;
            resourceInputs["pointInTimeUTC"] = args ? args.pointInTimeUTC : undefined;
            resourceInputs["postgresqlVersion"] = args ? args.postgresqlVersion : undefined;
            resourceInputs["preferredPrimaryZone"] = args ? args.preferredPrimaryZone : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sourceLocation"] = args ? args.sourceLocation : undefined;
            resourceInputs["sourceResourceId"] = args ? args.sourceResourceId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["aadAuthEnabled"] = undefined /*out*/;
            resourceInputs["administratorLogin"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["earliestRestoreTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["passwordEnabled"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["readReplicas"] = undefined /*out*/;
            resourceInputs["serverNames"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["aadAuthEnabled"] = undefined /*out*/;
            resourceInputs["administratorLogin"] = undefined /*out*/;
            resourceInputs["authConfig"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["citusVersion"] = undefined /*out*/;
            resourceInputs["coordinatorEnablePublicIpAccess"] = undefined /*out*/;
            resourceInputs["coordinatorServerEdition"] = undefined /*out*/;
            resourceInputs["coordinatorStorageQuotaInMb"] = undefined /*out*/;
            resourceInputs["coordinatorVCores"] = undefined /*out*/;
            resourceInputs["dataEncryption"] = undefined /*out*/;
            resourceInputs["databaseName"] = undefined /*out*/;
            resourceInputs["earliestRestoreTime"] = undefined /*out*/;
            resourceInputs["enableGeoBackup"] = undefined /*out*/;
            resourceInputs["enableHa"] = undefined /*out*/;
            resourceInputs["enableShardsOnCoordinator"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maintenanceWindow"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nodeCount"] = undefined /*out*/;
            resourceInputs["nodeEnablePublicIpAccess"] = undefined /*out*/;
            resourceInputs["nodeServerEdition"] = undefined /*out*/;
            resourceInputs["nodeStorageQuotaInMb"] = undefined /*out*/;
            resourceInputs["nodeVCores"] = undefined /*out*/;
            resourceInputs["passwordEnabled"] = undefined /*out*/;
            resourceInputs["pointInTimeUTC"] = undefined /*out*/;
            resourceInputs["postgresqlVersion"] = undefined /*out*/;
            resourceInputs["preferredPrimaryZone"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["readReplicas"] = undefined /*out*/;
            resourceInputs["serverNames"] = undefined /*out*/;
            resourceInputs["sourceLocation"] = undefined /*out*/;
            resourceInputs["sourceResourceId"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:dbforpostgresql/v20201005privatepreview:ServerGroup" }, { type: "azure-native:dbforpostgresql/v20201005privatepreview:ServerGroupCluster" }, { type: "azure-native:dbforpostgresql/v20221108:Cluster" }, { type: "azure-native:dbforpostgresql/v20221108:ServerGroupCluster" }, { type: "azure-native:dbforpostgresql/v20230302preview:Cluster" }, { type: "azure-native:dbforpostgresql/v20230302preview:ServerGroupCluster" }, { type: "azure-native:dbforpostgresql:Cluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerGroupCluster.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServerGroupCluster = ServerGroupCluster;
/** @internal */
ServerGroupCluster.__pulumiType = 'azure-native:dbforpostgresql:ServerGroupCluster';
