// @pulumi/azure-native/migrate/sqlAssessmentV2Operation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlAssessmentV2Operation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * SQL Assessment REST resource.
 *
 * Uses Azure REST API version 2024-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-03-15.
 *
 * Other available API versions: 2023-03-15, 2023-04-01-preview, 2023-05-01-preview, 2023-09-09-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native migrate [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SqlAssessmentV2Operation extends pulumi.CustomResource {
    /**
     * Get an existing SqlAssessmentV2Operation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlAssessmentV2Operation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlAssessmentV2Operation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlAssessmentV2Operation.__pulumiType;
    }
    /**
     * Create a SqlAssessmentV2Operation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.groupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupName'");
            }
            if ((!args || args.projectName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["assessmentName"] = args ? args.assessmentName : undefined;
            resourceInputs["assessmentType"] = args ? args.assessmentType : undefined;
            resourceInputs["asyncCommitModeIntent"] = args ? args.asyncCommitModeIntent : undefined;
            resourceInputs["azureLocation"] = args ? args.azureLocation : undefined;
            resourceInputs["azureOfferCode"] = args ? args.azureOfferCode : undefined;
            resourceInputs["azureOfferCodeForVm"] = args ? args.azureOfferCodeForVm : undefined;
            resourceInputs["azureSecurityOfferingType"] = args ? args.azureSecurityOfferingType : undefined;
            resourceInputs["azureSqlDatabaseSettings"] = args ? args.azureSqlDatabaseSettings : undefined;
            resourceInputs["azureSqlManagedInstanceSettings"] = args ? args.azureSqlManagedInstanceSettings : undefined;
            resourceInputs["azureSqlVmSettings"] = args ? args.azureSqlVmSettings : undefined;
            resourceInputs["confidenceRatingInPercentage"] = args ? args.confidenceRatingInPercentage : undefined;
            resourceInputs["currency"] = args ? args.currency : undefined;
            resourceInputs["disasterRecoveryLocation"] = args ? args.disasterRecoveryLocation : undefined;
            resourceInputs["discountPercentage"] = args ? args.discountPercentage : undefined;
            resourceInputs["eaSubscriptionId"] = args ? args.eaSubscriptionId : undefined;
            resourceInputs["enableHadrAssessment"] = args ? args.enableHadrAssessment : undefined;
            resourceInputs["entityUptime"] = args ? args.entityUptime : undefined;
            resourceInputs["environmentType"] = args ? args.environmentType : undefined;
            resourceInputs["groupName"] = args ? args.groupName : undefined;
            resourceInputs["groupType"] = args ? args.groupType : undefined;
            resourceInputs["isInternetAccessAvailable"] = args ? args.isInternetAccessAvailable : undefined;
            resourceInputs["multiSubnetIntent"] = args ? args.multiSubnetIntent : undefined;
            resourceInputs["optimizationLogic"] = args ? args.optimizationLogic : undefined;
            resourceInputs["osLicense"] = args ? args.osLicense : undefined;
            resourceInputs["percentile"] = args ? args.percentile : undefined;
            resourceInputs["perfDataEndTime"] = args ? args.perfDataEndTime : undefined;
            resourceInputs["perfDataStartTime"] = args ? args.perfDataStartTime : undefined;
            resourceInputs["projectName"] = args ? args.projectName : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["reservedInstance"] = args ? args.reservedInstance : undefined;
            resourceInputs["reservedInstanceForVm"] = args ? args.reservedInstanceForVm : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scalingFactor"] = args ? args.scalingFactor : undefined;
            resourceInputs["sizingCriterion"] = args ? args.sizingCriterion : undefined;
            resourceInputs["sqlServerLicense"] = args ? args.sqlServerLicense : undefined;
            resourceInputs["timeRange"] = args ? args.timeRange : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["schemaVersion"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
        }
        else {
            resourceInputs["assessmentType"] = undefined /*out*/;
            resourceInputs["asyncCommitModeIntent"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["azureLocation"] = undefined /*out*/;
            resourceInputs["azureOfferCode"] = undefined /*out*/;
            resourceInputs["azureOfferCodeForVm"] = undefined /*out*/;
            resourceInputs["azureSecurityOfferingType"] = undefined /*out*/;
            resourceInputs["azureSqlDatabaseSettings"] = undefined /*out*/;
            resourceInputs["azureSqlManagedInstanceSettings"] = undefined /*out*/;
            resourceInputs["azureSqlVmSettings"] = undefined /*out*/;
            resourceInputs["confidenceRatingInPercentage"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["disasterRecoveryLocation"] = undefined /*out*/;
            resourceInputs["discountPercentage"] = undefined /*out*/;
            resourceInputs["eaSubscriptionId"] = undefined /*out*/;
            resourceInputs["enableHadrAssessment"] = undefined /*out*/;
            resourceInputs["entityUptime"] = undefined /*out*/;
            resourceInputs["environmentType"] = undefined /*out*/;
            resourceInputs["groupType"] = undefined /*out*/;
            resourceInputs["isInternetAccessAvailable"] = undefined /*out*/;
            resourceInputs["multiSubnetIntent"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["optimizationLogic"] = undefined /*out*/;
            resourceInputs["osLicense"] = undefined /*out*/;
            resourceInputs["percentile"] = undefined /*out*/;
            resourceInputs["perfDataEndTime"] = undefined /*out*/;
            resourceInputs["perfDataStartTime"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["reservedInstance"] = undefined /*out*/;
            resourceInputs["reservedInstanceForVm"] = undefined /*out*/;
            resourceInputs["scalingFactor"] = undefined /*out*/;
            resourceInputs["schemaVersion"] = undefined /*out*/;
            resourceInputs["sizingCriterion"] = undefined /*out*/;
            resourceInputs["sqlServerLicense"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["timeRange"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:migrate/v20230315:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20230401preview:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20230501preview:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20230909preview:SqlAssessmentV2Operation" }, { type: "azure-native:migrate/v20240101preview:SqlAssessmentV2Operation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlAssessmentV2Operation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SqlAssessmentV2Operation = SqlAssessmentV2Operation;
/** @internal */
SqlAssessmentV2Operation.__pulumiType = 'azure-native:migrate:SqlAssessmentV2Operation';
