// @pulumi/azure-native/cloudngfw/certificateObjectLocalRulestack.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateObjectLocalRulestack = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * LocalRulestack Certificate Object
 *
 * Uses Azure REST API version 2025-02-06-preview. In version 2.x of the Azure Native provider, it used API version 2023-09-01.
 *
 * Other available API versions: 2023-09-01, 2023-10-10-preview, 2024-01-19-preview, 2024-02-07-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cloudngfw [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class CertificateObjectLocalRulestack extends pulumi.CustomResource {
    /**
     * Get an existing CertificateObjectLocalRulestack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CertificateObjectLocalRulestack(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CertificateObjectLocalRulestack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CertificateObjectLocalRulestack.__pulumiType;
    }
    /**
     * Create a CertificateObjectLocalRulestack resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.certificateSelfSigned === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateSelfSigned'");
            }
            if ((!args || args.localRulestackName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'localRulestackName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["auditComment"] = args ? args.auditComment : undefined;
            resourceInputs["certificateSelfSigned"] = args ? args.certificateSelfSigned : undefined;
            resourceInputs["certificateSignerResourceId"] = args ? args.certificateSignerResourceId : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["localRulestackName"] = args ? args.localRulestackName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["auditComment"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["certificateSelfSigned"] = undefined /*out*/;
            resourceInputs["certificateSignerResourceId"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cloudngfw/v20220829:CertificateObjectLocalRulestack" }, { type: "azure-native:cloudngfw/v20220829preview:CertificateObjectLocalRulestack" }, { type: "azure-native:cloudngfw/v20230901:CertificateObjectLocalRulestack" }, { type: "azure-native:cloudngfw/v20230901preview:CertificateObjectLocalRulestack" }, { type: "azure-native:cloudngfw/v20231010preview:CertificateObjectLocalRulestack" }, { type: "azure-native:cloudngfw/v20240119preview:CertificateObjectLocalRulestack" }, { type: "azure-native:cloudngfw/v20240207preview:CertificateObjectLocalRulestack" }, { type: "azure-native:cloudngfw/v20250206preview:CertificateObjectLocalRulestack" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CertificateObjectLocalRulestack.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CertificateObjectLocalRulestack = CertificateObjectLocalRulestack;
/** @internal */
CertificateObjectLocalRulestack.__pulumiType = 'azure-native:cloudngfw:CertificateObjectLocalRulestack';
