// @pulumi/azure-native/cloudngfw/firewall.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Firewall = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * PaloAltoNetworks Firewall
 *
 * Uses Azure REST API version 2025-02-06-preview. In version 2.x of the Azure Native provider, it used API version 2023-09-01.
 *
 * Other available API versions: 2023-09-01, 2023-10-10-preview, 2024-01-19-preview, 2024-02-07-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cloudngfw [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Firewall extends pulumi.CustomResource {
    /**
     * Get an existing Firewall resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Firewall(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Firewall.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Firewall.__pulumiType;
    }
    /**
     * Create a Firewall resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dnsSettings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dnsSettings'");
            }
            if ((!args || args.marketplaceDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'marketplaceDetails'");
            }
            if ((!args || args.networkProfile === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkProfile'");
            }
            if ((!args || args.planData === undefined) && !opts.urn) {
                throw new Error("Missing required property 'planData'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["associatedRulestack"] = args ? args.associatedRulestack : undefined;
            resourceInputs["dnsSettings"] = args ? args.dnsSettings : undefined;
            resourceInputs["firewallName"] = args ? args.firewallName : undefined;
            resourceInputs["frontEndSettings"] = args ? args.frontEndSettings : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["isPanoramaManaged"] = args ? args.isPanoramaManaged : undefined;
            resourceInputs["isStrataCloudManaged"] = args ? args.isStrataCloudManaged : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["marketplaceDetails"] = args ? args.marketplaceDetails : undefined;
            resourceInputs["networkProfile"] = args ? args.networkProfile : undefined;
            resourceInputs["panEtag"] = args ? args.panEtag : undefined;
            resourceInputs["panoramaConfig"] = args ? args.panoramaConfig : undefined;
            resourceInputs["planData"] = args ? args.planData : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["strataCloudManagerConfig"] = args ? args.strataCloudManagerConfig : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["associatedRulestack"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["dnsSettings"] = undefined /*out*/;
            resourceInputs["frontEndSettings"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["isPanoramaManaged"] = undefined /*out*/;
            resourceInputs["isStrataCloudManaged"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["marketplaceDetails"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkProfile"] = undefined /*out*/;
            resourceInputs["panEtag"] = undefined /*out*/;
            resourceInputs["panoramaConfig"] = undefined /*out*/;
            resourceInputs["planData"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["strataCloudManagerConfig"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cloudngfw/v20220829:Firewall" }, { type: "azure-native:cloudngfw/v20220829preview:Firewall" }, { type: "azure-native:cloudngfw/v20230901:Firewall" }, { type: "azure-native:cloudngfw/v20230901preview:Firewall" }, { type: "azure-native:cloudngfw/v20231010preview:Firewall" }, { type: "azure-native:cloudngfw/v20240119preview:Firewall" }, { type: "azure-native:cloudngfw/v20240207preview:Firewall" }, { type: "azure-native:cloudngfw/v20250206preview:Firewall" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Firewall.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Firewall = Firewall;
/** @internal */
Firewall.__pulumiType = 'azure-native:cloudngfw:Firewall';
