// @pulumi/azure-native/types/enums/hybridnetwork/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkFunctionNFVIType = exports.VirtualMachineSizeTypes = exports.VMSwitchType = exports.Type = exports.TemplateType = exports.SkuType = exports.SkuName = exports.SkuDeploymentMode = exports.PublisherScope = exports.OperatingSystemTypes = exports.NetworkFunctionType = exports.NetworkFunctionRoleConfigurationType = exports.NetworkFunctionConfigurationType = exports.NFVIType = exports.ManagedServiceIdentityType = exports.IdType = exports.IPVersion = exports.IPAllocationMethod = exports.DiskCreateOptionTypes = exports.DeviceType = exports.ContainerizedNetworkFunctionNFVIType = exports.ConfigurationGroupValueConfigurationType = exports.BackingResourcePublicNetworkAccess = exports.AzureOperatorNexusArtifactType = exports.AzureCoreArtifactType = exports.AzureArcKubernetesArtifactType = exports.ArtifactType = exports.ArtifactStoreType = exports.ArtifactReplicationStrategy = exports.ApplicationEnablement = void 0;
exports.ApplicationEnablement = {
    Unknown: "Unknown",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ArtifactReplicationStrategy = {
    Unknown: "Unknown",
    SingleReplication: "SingleReplication",
};
exports.ArtifactStoreType = {
    Unknown: "Unknown",
    AzureContainerRegistry: "AzureContainerRegistry",
    AzureStorageAccount: "AzureStorageAccount",
};
exports.ArtifactType = {
    Unknown: "Unknown",
    OCIArtifact: "OCIArtifact",
    VhdImageFile: "VhdImageFile",
    ArmTemplate: "ArmTemplate",
    ImageFile: "ImageFile",
};
exports.AzureArcKubernetesArtifactType = {
    Unknown: "Unknown",
    HelmPackage: "HelmPackage",
};
exports.AzureCoreArtifactType = {
    Unknown: "Unknown",
    VhdImageFile: "VhdImageFile",
    ArmTemplate: "ArmTemplate",
};
exports.AzureOperatorNexusArtifactType = {
    Unknown: "Unknown",
    ImageFile: "ImageFile",
    ArmTemplate: "ArmTemplate",
};
exports.BackingResourcePublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ConfigurationGroupValueConfigurationType = {
    Unknown: "Unknown",
    Secret: "Secret",
    Open: "Open",
};
exports.ContainerizedNetworkFunctionNFVIType = {
    Unknown: "Unknown",
    AzureArcKubernetes: "AzureArcKubernetes",
};
exports.DeviceType = {
    Unknown: "Unknown",
    AzureStackEdge: "AzureStackEdge",
};
exports.DiskCreateOptionTypes = {
    Unknown: "Unknown",
    Empty: "Empty",
};
exports.IPAllocationMethod = {
    Unknown: "Unknown",
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.IPVersion = {
    Unknown: "Unknown",
    IPv4: "IPv4",
};
exports.IdType = {
    Unknown: "Unknown",
    Open: "Open",
    Secret: "Secret",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NFVIType = {
    Unknown: "Unknown",
    AzureArcKubernetes: "AzureArcKubernetes",
    AzureCore: "AzureCore",
    AzureOperatorNexus: "AzureOperatorNexus",
};
exports.NetworkFunctionConfigurationType = {
    Unknown: "Unknown",
    Secret: "Secret",
    Open: "Open",
};
exports.NetworkFunctionRoleConfigurationType = {
    Unknown: "Unknown",
    VirtualMachine: "VirtualMachine",
};
exports.NetworkFunctionType = {
    Unknown: "Unknown",
    VirtualNetworkFunction: "VirtualNetworkFunction",
    ContainerizedNetworkFunction: "ContainerizedNetworkFunction",
};
exports.OperatingSystemTypes = {
    Unknown: "Unknown",
    Windows: "Windows",
    Linux: "Linux",
};
exports.PublisherScope = {
    Unknown: "Unknown",
    Private: "Private",
};
exports.SkuDeploymentMode = {
    Unknown: "Unknown",
    Azure: "Azure",
    PrivateEdgeZone: "PrivateEdgeZone",
};
exports.SkuName = {
    Basic: "Basic",
    Standard: "Standard",
};
exports.SkuType = {
    Unknown: "Unknown",
    EvolvedPacketCore: "EvolvedPacketCore",
    SDWAN: "SDWAN",
    Firewall: "Firewall",
};
exports.TemplateType = {
    Unknown: "Unknown",
    ArmTemplate: "ArmTemplate",
};
exports.Type = {
    Unknown: "Unknown",
    ArmResourceDefinition: "ArmResourceDefinition",
    NetworkFunctionDefinition: "NetworkFunctionDefinition",
};
exports.VMSwitchType = {
    Unknown: "Unknown",
    Management: "Management",
    Wan: "Wan",
    Lan: "Lan",
};
exports.VirtualMachineSizeTypes = {
    Unknown: "Unknown",
    Standard_D1_v2: "Standard_D1_v2",
    Standard_D2_v2: "Standard_D2_v2",
    Standard_D3_v2: "Standard_D3_v2",
    Standard_D4_v2: "Standard_D4_v2",
    Standard_D5_v2: "Standard_D5_v2",
    Standard_D11_v2: "Standard_D11_v2",
    Standard_D12_v2: "Standard_D12_v2",
    Standard_D13_v2: "Standard_D13_v2",
    Standard_DS1_v2: "Standard_DS1_v2",
    Standard_DS2_v2: "Standard_DS2_v2",
    Standard_DS3_v2: "Standard_DS3_v2",
    Standard_DS4_v2: "Standard_DS4_v2",
    Standard_DS5_v2: "Standard_DS5_v2",
    Standard_DS11_v2: "Standard_DS11_v2",
    Standard_DS12_v2: "Standard_DS12_v2",
    Standard_DS13_v2: "Standard_DS13_v2",
    Standard_F1: "Standard_F1",
    Standard_F2: "Standard_F2",
    Standard_F4: "Standard_F4",
    Standard_F8: "Standard_F8",
    Standard_F16: "Standard_F16",
    Standard_F1s: "Standard_F1s",
    Standard_F2s: "Standard_F2s",
    Standard_F4s: "Standard_F4s",
    Standard_F8s: "Standard_F8s",
    Standard_F16s: "Standard_F16s",
};
exports.VirtualNetworkFunctionNFVIType = {
    Unknown: "Unknown",
    AzureCore: "AzureCore",
    AzureOperatorNexus: "AzureOperatorNexus",
};
