// @pulumi/azure-native/orbital/groundStation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroundStation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Ground Station contains one or more antennas.
 *
 * Uses Azure REST API version 2024-03-01-preview. In version 2.x of the Azure Native provider, it used API version 2024-03-01-preview.
 *
 * Other available API versions: 2024-03-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native orbital [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class GroundStation extends pulumi.CustomResource {
    /**
     * Get an existing GroundStation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GroundStation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GroundStation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GroundStation.__pulumiType;
    }
    /**
     * Create a GroundStation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.capabilities === undefined) && !opts.urn) {
                throw new Error("Missing required property 'capabilities'");
            }
            if ((!args || args.globalCommunicationsSite === undefined) && !opts.urn) {
                throw new Error("Missing required property 'globalCommunicationsSite'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["altitudeMeters"] = args ? args.altitudeMeters : undefined;
            resourceInputs["capabilities"] = args ? args.capabilities : undefined;
            resourceInputs["city"] = args ? args.city : undefined;
            resourceInputs["globalCommunicationsSite"] = args ? args.globalCommunicationsSite : undefined;
            resourceInputs["groundStationName"] = args ? args.groundStationName : undefined;
            resourceInputs["latitudeDegrees"] = args ? args.latitudeDegrees : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["longitudeDegrees"] = args ? args.longitudeDegrees : undefined;
            resourceInputs["providerName"] = args ? args.providerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["releaseMode"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["altitudeMeters"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["capabilities"] = undefined /*out*/;
            resourceInputs["city"] = undefined /*out*/;
            resourceInputs["globalCommunicationsSite"] = undefined /*out*/;
            resourceInputs["latitudeDegrees"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["longitudeDegrees"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["providerName"] = undefined /*out*/;
            resourceInputs["releaseMode"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:orbital/v20240301:GroundStation" }, { type: "azure-native:orbital/v20240301preview:GroundStation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GroundStation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GroundStation = GroundStation;
/** @internal */
GroundStation.__pulumiType = 'azure-native:orbital:GroundStation';
