// @pulumi/azure-native/orbital/spacecraft.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Spacecraft = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Customer creates a spacecraft resource to schedule a contact.
 *
 * Uses Azure REST API version 2022-11-01. In version 2.x of the Azure Native provider, it used API version 2022-11-01.
 */
class Spacecraft extends pulumi.CustomResource {
    /**
     * Get an existing Spacecraft resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Spacecraft(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Spacecraft.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Spacecraft.__pulumiType;
    }
    /**
     * Create a Spacecraft resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.links === undefined) && !opts.urn) {
                throw new Error("Missing required property 'links'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.titleLine === undefined) && !opts.urn) {
                throw new Error("Missing required property 'titleLine'");
            }
            if ((!args || args.tleLine1 === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tleLine1'");
            }
            if ((!args || args.tleLine2 === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tleLine2'");
            }
            resourceInputs["links"] = args ? args.links : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["noradId"] = args ? args.noradId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["spacecraftName"] = args ? args.spacecraftName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["titleLine"] = args ? args.titleLine : undefined;
            resourceInputs["tleLine1"] = args ? args.tleLine1 : undefined;
            resourceInputs["tleLine2"] = args ? args.tleLine2 : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["links"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["noradId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["titleLine"] = undefined /*out*/;
            resourceInputs["tleLine1"] = undefined /*out*/;
            resourceInputs["tleLine2"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:orbital/v20220301:Spacecraft" }, { type: "azure-native:orbital/v20221101:Spacecraft" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Spacecraft.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Spacecraft = Spacecraft;
/** @internal */
Spacecraft.__pulumiType = 'azure-native:orbital:Spacecraft';
