// @pulumi/azure-native/cosmosdb/cassandraCluster.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CassandraCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Representation of a managed Cassandra cluster.
 *
 * Uses Azure REST API version 2024-11-15.
 *
 * Other available API versions: 2021-03-01-preview, 2021-04-01-preview, 2021-07-01-preview, 2021-10-15, 2021-10-15-preview, 2021-11-15-preview, 2022-02-15-preview, 2022-05-15, 2022-05-15-preview, 2022-08-15, 2022-08-15-preview, 2022-11-15, 2022-11-15-preview, 2023-03-01-preview, 2023-03-15, 2023-03-15-preview, 2023-04-15, 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-12-01-preview, 2025-04-15, 2025-05-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class CassandraCluster extends pulumi.CustomResource {
    /**
     * Get an existing CassandraCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CassandraCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CassandraCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CassandraCluster.__pulumiType;
    }
    /**
     * Create a CassandraCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20210301preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20210401preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20210701preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20211015:CassandraCluster" }, { type: "azure-native:cosmosdb/v20211015preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20211115preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20220215preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20220515:CassandraCluster" }, { type: "azure-native:cosmosdb/v20220515preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20220815:CassandraCluster" }, { type: "azure-native:cosmosdb/v20220815preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20221115:CassandraCluster" }, { type: "azure-native:cosmosdb/v20221115preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20230301preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20230315:CassandraCluster" }, { type: "azure-native:cosmosdb/v20230315preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20230415:CassandraCluster" }, { type: "azure-native:cosmosdb/v20230915:CassandraCluster" }, { type: "azure-native:cosmosdb/v20230915preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20231115:CassandraCluster" }, { type: "azure-native:cosmosdb/v20231115preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20240215preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20240515:CassandraCluster" }, { type: "azure-native:cosmosdb/v20240515preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20240815:CassandraCluster" }, { type: "azure-native:cosmosdb/v20240901preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20241115:CassandraCluster" }, { type: "azure-native:cosmosdb/v20241201preview:CassandraCluster" }, { type: "azure-native:cosmosdb/v20250415:CassandraCluster" }, { type: "azure-native:cosmosdb/v20250501preview:CassandraCluster" }, { type: "azure-native:documentdb/v20210701preview:CassandraCluster" }, { type: "azure-native:documentdb/v20230415:CassandraCluster" }, { type: "azure-native:documentdb/v20230915:CassandraCluster" }, { type: "azure-native:documentdb/v20230915preview:CassandraCluster" }, { type: "azure-native:documentdb/v20231115:CassandraCluster" }, { type: "azure-native:documentdb/v20231115preview:CassandraCluster" }, { type: "azure-native:documentdb/v20240215preview:CassandraCluster" }, { type: "azure-native:documentdb/v20240515:CassandraCluster" }, { type: "azure-native:documentdb/v20240515preview:CassandraCluster" }, { type: "azure-native:documentdb/v20240815:CassandraCluster" }, { type: "azure-native:documentdb/v20240901preview:CassandraCluster" }, { type: "azure-native:documentdb/v20241115:CassandraCluster" }, { type: "azure-native:documentdb/v20241201preview:CassandraCluster" }, { type: "azure-native:documentdb:CassandraCluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CassandraCluster.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CassandraCluster = CassandraCluster;
/** @internal */
CassandraCluster.__pulumiType = 'azure-native:cosmosdb:CassandraCluster';
