// @pulumi/azure-native/cosmosdb/databaseAccountSqlContainer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseAccountSqlContainer = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB container.
 *
 * Uses Azure REST API version 2016-03-31.
 *
 * Other available API versions: 2015-04-01, 2015-04-08, 2015-11-06, 2016-03-19. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DatabaseAccountSqlContainer extends pulumi.CustomResource {
    /**
     * Get an existing DatabaseAccountSqlContainer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DatabaseAccountSqlContainer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatabaseAccountSqlContainer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseAccountSqlContainer.__pulumiType;
    }
    /**
     * Create a DatabaseAccountSqlContainer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["containerName"] = args ? args.containerName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? (args.resource ? pulumi.output(args.resource).apply(inputs.cosmosdb.sqlContainerResourceArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["conflictResolutionPolicy"] = undefined /*out*/;
            resourceInputs["defaultTtl"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["indexingPolicy"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["partitionKey"] = undefined /*out*/;
            resourceInputs["rid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["ts"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueKeyPolicy"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["conflictResolutionPolicy"] = undefined /*out*/;
            resourceInputs["defaultTtl"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["indexingPolicy"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["partitionKey"] = undefined /*out*/;
            resourceInputs["rid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["ts"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueKeyPolicy"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20150401:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20150408:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20151106:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20160319:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20160331:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20190801:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20191212:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20200301:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20200401:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20200601preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20200901:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210115:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210301preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210315:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210401preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210415:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210515:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210615:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20210701preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20211015:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20211015preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20211115preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20220215preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20220515:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20220515preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20220815:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20220815preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20221115:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20221115preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20230301preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20230315:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20230315preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20230415:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20230915:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20230915preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20231115:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20231115preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20240215preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20240515:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20240515preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20240815:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20240901preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20241115:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20241201preview:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20250415:DatabaseAccountSqlContainer" }, { type: "azure-native:cosmosdb/v20250501preview:DatabaseAccountSqlContainer" }, { type: "azure-native:documentdb/v20230315preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20230415:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20230915:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20230915preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20231115:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20231115preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20240215preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20240515:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20240515preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20240815:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20240901preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20241115:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20241201preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb:SqlResourceSqlContainer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseAccountSqlContainer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DatabaseAccountSqlContainer = DatabaseAccountSqlContainer;
/** @internal */
DatabaseAccountSqlContainer.__pulumiType = 'azure-native:cosmosdb:DatabaseAccountSqlContainer';
