// @pulumi/azure-native/cosmosdb/databaseAccount.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB database account.
 *
 * Uses Azure REST API version 2024-11-15.
 *
 * Other available API versions: 2015-04-01, 2015-04-08, 2015-11-06, 2016-03-19, 2016-03-31, 2019-08-01, 2019-12-12, 2020-03-01, 2020-04-01, 2020-06-01-preview, 2020-09-01, 2021-01-15, 2021-03-01-preview, 2021-03-15, 2021-04-01-preview, 2021-04-15, 2021-05-15, 2021-06-15, 2021-07-01-preview, 2021-10-15, 2021-10-15-preview, 2021-11-15-preview, 2022-02-15-preview, 2022-05-15, 2022-05-15-preview, 2022-08-15, 2022-08-15-preview, 2022-11-15, 2022-11-15-preview, 2023-03-01-preview, 2023-03-15, 2023-03-15-preview, 2023-04-15, 2023-09-15, 2023-09-15-preview, 2023-11-15, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15, 2024-05-15-preview, 2024-08-15, 2024-09-01-preview, 2024-12-01-preview, 2025-04-15, 2025-05-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DatabaseAccount extends pulumi.CustomResource {
    /**
     * Get an existing DatabaseAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DatabaseAccount(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatabaseAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseAccount.__pulumiType;
    }
    /**
     * Create a DatabaseAccount resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseAccountOfferType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseAccountOfferType'");
            }
            if ((!args || args.locations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'locations'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["analyticalStorageConfiguration"] = args ? args.analyticalStorageConfiguration : undefined;
            resourceInputs["apiProperties"] = args ? args.apiProperties : undefined;
            resourceInputs["backupPolicy"] = args ? args.backupPolicy : undefined;
            resourceInputs["capabilities"] = args ? args.capabilities : undefined;
            resourceInputs["capacity"] = args ? args.capacity : undefined;
            resourceInputs["connectorOffer"] = args ? args.connectorOffer : undefined;
            resourceInputs["consistencyPolicy"] = args ? args.consistencyPolicy : undefined;
            resourceInputs["cors"] = args ? args.cors : undefined;
            resourceInputs["createMode"] = (_a = (args ? args.createMode : undefined)) !== null && _a !== void 0 ? _a : "Default";
            resourceInputs["customerManagedKeyStatus"] = args ? args.customerManagedKeyStatus : undefined;
            resourceInputs["databaseAccountOfferType"] = args ? args.databaseAccountOfferType : undefined;
            resourceInputs["defaultIdentity"] = args ? args.defaultIdentity : undefined;
            resourceInputs["disableKeyBasedMetadataWriteAccess"] = args ? args.disableKeyBasedMetadataWriteAccess : undefined;
            resourceInputs["disableLocalAuth"] = args ? args.disableLocalAuth : undefined;
            resourceInputs["enableAnalyticalStorage"] = args ? args.enableAnalyticalStorage : undefined;
            resourceInputs["enableAutomaticFailover"] = args ? args.enableAutomaticFailover : undefined;
            resourceInputs["enableBurstCapacity"] = args ? args.enableBurstCapacity : undefined;
            resourceInputs["enableCassandraConnector"] = args ? args.enableCassandraConnector : undefined;
            resourceInputs["enableFreeTier"] = args ? args.enableFreeTier : undefined;
            resourceInputs["enableMultipleWriteLocations"] = args ? args.enableMultipleWriteLocations : undefined;
            resourceInputs["enablePartitionMerge"] = args ? args.enablePartitionMerge : undefined;
            resourceInputs["enablePerRegionPerPartitionAutoscale"] = args ? args.enablePerRegionPerPartitionAutoscale : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["ipRules"] = args ? args.ipRules : undefined;
            resourceInputs["isVirtualNetworkFilterEnabled"] = args ? args.isVirtualNetworkFilterEnabled : undefined;
            resourceInputs["keyVaultKeyUri"] = args ? args.keyVaultKeyUri : undefined;
            resourceInputs["kind"] = (_b = (args ? args.kind : undefined)) !== null && _b !== void 0 ? _b : "GlobalDocumentDB";
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["locations"] = args ? args.locations : undefined;
            resourceInputs["minimalTlsVersion"] = args ? args.minimalTlsVersion : undefined;
            resourceInputs["networkAclBypass"] = args ? args.networkAclBypass : undefined;
            resourceInputs["networkAclBypassResourceIds"] = args ? args.networkAclBypassResourceIds : undefined;
            resourceInputs["publicNetworkAccess"] = args ? args.publicNetworkAccess : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["restoreParameters"] = args ? args.restoreParameters : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["virtualNetworkRules"] = args ? args.virtualNetworkRules : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["documentEndpoint"] = undefined /*out*/;
            resourceInputs["failoverPolicies"] = undefined /*out*/;
            resourceInputs["instanceId"] = undefined /*out*/;
            resourceInputs["keysMetadata"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["readLocations"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["writeLocations"] = undefined /*out*/;
        }
        else {
            resourceInputs["analyticalStorageConfiguration"] = undefined /*out*/;
            resourceInputs["apiProperties"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backupPolicy"] = undefined /*out*/;
            resourceInputs["capabilities"] = undefined /*out*/;
            resourceInputs["capacity"] = undefined /*out*/;
            resourceInputs["connectorOffer"] = undefined /*out*/;
            resourceInputs["consistencyPolicy"] = undefined /*out*/;
            resourceInputs["cors"] = undefined /*out*/;
            resourceInputs["createMode"] = undefined /*out*/;
            resourceInputs["customerManagedKeyStatus"] = undefined /*out*/;
            resourceInputs["databaseAccountOfferType"] = undefined /*out*/;
            resourceInputs["defaultIdentity"] = undefined /*out*/;
            resourceInputs["disableKeyBasedMetadataWriteAccess"] = undefined /*out*/;
            resourceInputs["disableLocalAuth"] = undefined /*out*/;
            resourceInputs["documentEndpoint"] = undefined /*out*/;
            resourceInputs["enableAnalyticalStorage"] = undefined /*out*/;
            resourceInputs["enableAutomaticFailover"] = undefined /*out*/;
            resourceInputs["enableBurstCapacity"] = undefined /*out*/;
            resourceInputs["enableCassandraConnector"] = undefined /*out*/;
            resourceInputs["enableFreeTier"] = undefined /*out*/;
            resourceInputs["enableMultipleWriteLocations"] = undefined /*out*/;
            resourceInputs["enablePartitionMerge"] = undefined /*out*/;
            resourceInputs["enablePerRegionPerPartitionAutoscale"] = undefined /*out*/;
            resourceInputs["failoverPolicies"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["instanceId"] = undefined /*out*/;
            resourceInputs["ipRules"] = undefined /*out*/;
            resourceInputs["isVirtualNetworkFilterEnabled"] = undefined /*out*/;
            resourceInputs["keyVaultKeyUri"] = undefined /*out*/;
            resourceInputs["keysMetadata"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["locations"] = undefined /*out*/;
            resourceInputs["minimalTlsVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkAclBypass"] = undefined /*out*/;
            resourceInputs["networkAclBypassResourceIds"] = undefined /*out*/;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicNetworkAccess"] = undefined /*out*/;
            resourceInputs["readLocations"] = undefined /*out*/;
            resourceInputs["restoreParameters"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualNetworkRules"] = undefined /*out*/;
            resourceInputs["writeLocations"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20150401:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20150408:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20151106:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20160319:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20160331:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20190801:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20191212:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20200301:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20200401:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20200601preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20200901:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210115:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210301preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210315:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210401preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210415:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210515:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210615:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20210701preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20211015:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20211015preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20211115preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20220215preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20220515:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20220515preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20220815:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20220815preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20221115:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20221115preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20230301preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20230315:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20230315preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20230415:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20230915:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20230915preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20231115:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20231115preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20240215preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20240515:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20240515preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20240815:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20240901preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20241115:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20241201preview:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20250415:DatabaseAccount" }, { type: "azure-native:cosmosdb/v20250501preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20210401preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20230315preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20230415:DatabaseAccount" }, { type: "azure-native:documentdb/v20230915:DatabaseAccount" }, { type: "azure-native:documentdb/v20230915preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20231115:DatabaseAccount" }, { type: "azure-native:documentdb/v20231115preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20240215preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20240515:DatabaseAccount" }, { type: "azure-native:documentdb/v20240515preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20240815:DatabaseAccount" }, { type: "azure-native:documentdb/v20240901preview:DatabaseAccount" }, { type: "azure-native:documentdb/v20241115:DatabaseAccount" }, { type: "azure-native:documentdb/v20241201preview:DatabaseAccount" }, { type: "azure-native:documentdb:DatabaseAccount" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseAccount.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DatabaseAccount = DatabaseAccount;
/** @internal */
DatabaseAccount.__pulumiType = 'azure-native:cosmosdb:DatabaseAccount';
