// @pulumi/azure-native/cosmosdb/graphResourceGraph.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GraphResourceGraph = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB Graph resource.
 *
 * Uses Azure REST API version 2024-12-01-preview.
 *
 * Other available API versions: 2021-07-01-preview, 2021-10-15-preview, 2021-11-15-preview, 2022-02-15-preview, 2022-05-15-preview, 2022-08-15-preview, 2022-11-15-preview, 2023-03-01-preview, 2023-03-15-preview, 2023-09-15-preview, 2023-11-15-preview, 2024-02-15-preview, 2024-05-15-preview, 2024-09-01-preview, 2025-05-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class GraphResourceGraph extends pulumi.CustomResource {
    /**
     * Get an existing GraphResourceGraph resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GraphResourceGraph(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GraphResourceGraph.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GraphResourceGraph.__pulumiType;
    }
    /**
     * Create a GraphResourceGraph resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["graphName"] = args ? args.graphName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? args.resource : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["options"] = undefined /*out*/;
            resourceInputs["resource"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20210701preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20211015preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20211115preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20220215preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20220515preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20220815preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20221115preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20230301preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20230315preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20230915preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20231115preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20240215preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20240515preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20240901preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20241201preview:GraphResourceGraph" }, { type: "azure-native:cosmosdb/v20250501preview:GraphResourceGraph" }, { type: "azure-native:documentdb/v20230315preview:GraphResourceGraph" }, { type: "azure-native:documentdb/v20230915preview:GraphResourceGraph" }, { type: "azure-native:documentdb/v20231115preview:GraphResourceGraph" }, { type: "azure-native:documentdb/v20240215preview:GraphResourceGraph" }, { type: "azure-native:documentdb/v20240515preview:GraphResourceGraph" }, { type: "azure-native:documentdb/v20240901preview:GraphResourceGraph" }, { type: "azure-native:documentdb/v20241201preview:GraphResourceGraph" }, { type: "azure-native:documentdb:GraphResourceGraph" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GraphResourceGraph.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GraphResourceGraph = GraphResourceGraph;
/** @internal */
GraphResourceGraph.__pulumiType = 'azure-native:cosmosdb:GraphResourceGraph';
