// @pulumi/azure-native/cosmosdb/throughputPool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThroughputPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Azure Cosmos DB Throughputpool.
 *
 * Uses Azure REST API version 2024-12-01-preview.
 *
 * Other available API versions: 2023-11-15-preview, 2024-02-15-preview, 2024-05-15-preview, 2024-09-01-preview, 2025-05-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native cosmosdb [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ThroughputPool extends pulumi.CustomResource {
    /**
     * Get an existing ThroughputPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ThroughputPool(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ThroughputPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ThroughputPool.__pulumiType;
    }
    /**
     * Create a ThroughputPool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["maxThroughput"] = args ? args.maxThroughput : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["throughputPoolName"] = args ? args.throughputPoolName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maxThroughput"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cosmosdb/v20231115preview:ThroughputPool" }, { type: "azure-native:cosmosdb/v20240215preview:ThroughputPool" }, { type: "azure-native:cosmosdb/v20240515preview:ThroughputPool" }, { type: "azure-native:cosmosdb/v20240901preview:ThroughputPool" }, { type: "azure-native:cosmosdb/v20241201preview:ThroughputPool" }, { type: "azure-native:cosmosdb/v20250501preview:ThroughputPool" }, { type: "azure-native:documentdb/v20231115preview:ThroughputPool" }, { type: "azure-native:documentdb/v20240215preview:ThroughputPool" }, { type: "azure-native:documentdb/v20240515preview:ThroughputPool" }, { type: "azure-native:documentdb/v20240901preview:ThroughputPool" }, { type: "azure-native:documentdb/v20241201preview:ThroughputPool" }, { type: "azure-native:documentdb:ThroughputPool" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ThroughputPool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ThroughputPool = ThroughputPool;
/** @internal */
ThroughputPool.__pulumiType = 'azure-native:cosmosdb:ThroughputPool';
