// @pulumi/azure-native/apimanagement/apiDiagnostic.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiDiagnostic = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Diagnostic details.
 *
 * Uses Azure REST API version 2022-09-01-preview. In version 2.x of the Azure Native provider, it used API version 2022-08-01.
 *
 * Other available API versions: 2021-04-01-preview, 2021-08-01, 2021-12-01-preview, 2022-04-01-preview, 2022-08-01, 2023-03-01-preview, 2023-05-01-preview, 2023-09-01-preview, 2024-05-01, 2024-06-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native apimanagement [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ApiDiagnostic extends pulumi.CustomResource {
    /**
     * Get an existing ApiDiagnostic resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ApiDiagnostic(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApiDiagnostic.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApiDiagnostic.__pulumiType;
    }
    /**
     * Create a ApiDiagnostic resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            if ((!args || args.loggerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'loggerId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["alwaysLog"] = args ? args.alwaysLog : undefined;
            resourceInputs["apiId"] = args ? args.apiId : undefined;
            resourceInputs["backend"] = args ? args.backend : undefined;
            resourceInputs["diagnosticId"] = args ? args.diagnosticId : undefined;
            resourceInputs["frontend"] = args ? args.frontend : undefined;
            resourceInputs["httpCorrelationProtocol"] = args ? args.httpCorrelationProtocol : undefined;
            resourceInputs["logClientIp"] = args ? args.logClientIp : undefined;
            resourceInputs["loggerId"] = args ? args.loggerId : undefined;
            resourceInputs["metrics"] = args ? args.metrics : undefined;
            resourceInputs["operationNameFormat"] = args ? args.operationNameFormat : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sampling"] = args ? args.sampling : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["verbosity"] = args ? args.verbosity : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["alwaysLog"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backend"] = undefined /*out*/;
            resourceInputs["frontend"] = undefined /*out*/;
            resourceInputs["httpCorrelationProtocol"] = undefined /*out*/;
            resourceInputs["logClientIp"] = undefined /*out*/;
            resourceInputs["loggerId"] = undefined /*out*/;
            resourceInputs["metrics"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operationNameFormat"] = undefined /*out*/;
            resourceInputs["sampling"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["verbosity"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:apimanagement/v20170301:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20180101:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20180601preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20190101:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20191201:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20191201preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20200601preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20201201:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20210101preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20210401preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20210801:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20211201preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20220401preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20220801:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20220901preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20230301preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20230501preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20230901preview:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20240501:ApiDiagnostic" }, { type: "azure-native:apimanagement/v20240601preview:ApiDiagnostic" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ApiDiagnostic.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ApiDiagnostic = ApiDiagnostic;
/** @internal */
ApiDiagnostic.__pulumiType = 'azure-native:apimanagement:ApiDiagnostic';
