// @pulumi/azure-native/managednetworkfabric/internalNetwork.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InternalNetwork = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Defines the Internal Network resource.
 *
 * Uses Azure REST API version 2023-06-15. In version 2.x of the Azure Native provider, it used API version 2023-02-01-preview.
 *
 * Other available API versions: 2023-02-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native managednetworkfabric [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class InternalNetwork extends pulumi.CustomResource {
    /**
     * Get an existing InternalNetwork resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new InternalNetwork(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of InternalNetwork.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InternalNetwork.__pulumiType;
    }
    /**
     * Create a InternalNetwork resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.l3IsolationDomainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'l3IsolationDomainName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vlanId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vlanId'");
            }
            resourceInputs["annotation"] = args ? args.annotation : undefined;
            resourceInputs["bgpConfiguration"] = args ? (args.bgpConfiguration ? pulumi.output(args.bgpConfiguration).apply(inputs.managednetworkfabric.internalNetworkPropertiesBgpConfigurationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["connectedIPv4Subnets"] = args ? args.connectedIPv4Subnets : undefined;
            resourceInputs["connectedIPv6Subnets"] = args ? args.connectedIPv6Subnets : undefined;
            resourceInputs["egressAclId"] = args ? args.egressAclId : undefined;
            resourceInputs["exportRoutePolicy"] = args ? args.exportRoutePolicy : undefined;
            resourceInputs["exportRoutePolicyId"] = args ? args.exportRoutePolicyId : undefined;
            resourceInputs["extension"] = (_a = (args ? args.extension : undefined)) !== null && _a !== void 0 ? _a : "NoExtension";
            resourceInputs["importRoutePolicy"] = args ? args.importRoutePolicy : undefined;
            resourceInputs["importRoutePolicyId"] = args ? args.importRoutePolicyId : undefined;
            resourceInputs["ingressAclId"] = args ? args.ingressAclId : undefined;
            resourceInputs["internalNetworkName"] = args ? args.internalNetworkName : undefined;
            resourceInputs["isMonitoringEnabled"] = (_b = (args ? args.isMonitoringEnabled : undefined)) !== null && _b !== void 0 ? _b : "False";
            resourceInputs["l3IsolationDomainName"] = args ? args.l3IsolationDomainName : undefined;
            resourceInputs["mtu"] = (_c = (args ? args.mtu : undefined)) !== null && _c !== void 0 ? _c : 1500;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["staticRouteConfiguration"] = args ? (args.staticRouteConfiguration ? pulumi.output(args.staticRouteConfiguration).apply(inputs.managednetworkfabric.internalNetworkPropertiesStaticRouteConfigurationArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["vlanId"] = args ? args.vlanId : undefined;
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["annotation"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["bgpConfiguration"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["connectedIPv4Subnets"] = undefined /*out*/;
            resourceInputs["connectedIPv6Subnets"] = undefined /*out*/;
            resourceInputs["egressAclId"] = undefined /*out*/;
            resourceInputs["exportRoutePolicy"] = undefined /*out*/;
            resourceInputs["exportRoutePolicyId"] = undefined /*out*/;
            resourceInputs["extension"] = undefined /*out*/;
            resourceInputs["importRoutePolicy"] = undefined /*out*/;
            resourceInputs["importRoutePolicyId"] = undefined /*out*/;
            resourceInputs["ingressAclId"] = undefined /*out*/;
            resourceInputs["isMonitoringEnabled"] = undefined /*out*/;
            resourceInputs["mtu"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["staticRouteConfiguration"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vlanId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:managednetworkfabric/v20230201preview:InternalNetwork" }, { type: "azure-native:managednetworkfabric/v20230615:InternalNetwork" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(InternalNetwork.__pulumiType, name, resourceInputs, opts);
    }
}
exports.InternalNetwork = InternalNetwork;
/** @internal */
InternalNetwork.__pulumiType = 'azure-native:managednetworkfabric:InternalNetwork';
