// @pulumi/azure-native/managednetworkfabric/networkFabric.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkFabric = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * The Network Fabric resource definition.
 *
 * Uses Azure REST API version 2023-06-15. In version 2.x of the Azure Native provider, it used API version 2023-02-01-preview.
 *
 * Other available API versions: 2023-02-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native managednetworkfabric [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class NetworkFabric extends pulumi.CustomResource {
    /**
     * Get an existing NetworkFabric resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkFabric(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkFabric.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkFabric.__pulumiType;
    }
    /**
     * Create a NetworkFabric resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricASN === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricASN'");
            }
            if ((!args || args.ipv4Prefix === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ipv4Prefix'");
            }
            if ((!args || args.managementNetworkConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managementNetworkConfiguration'");
            }
            if ((!args || args.networkFabricControllerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkFabricControllerId'");
            }
            if ((!args || args.networkFabricSku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkFabricSku'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverCountPerRack === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverCountPerRack'");
            }
            if ((!args || args.terminalServerConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'terminalServerConfiguration'");
            }
            resourceInputs["annotation"] = args ? args.annotation : undefined;
            resourceInputs["fabricASN"] = args ? args.fabricASN : undefined;
            resourceInputs["fabricVersion"] = args ? args.fabricVersion : undefined;
            resourceInputs["ipv4Prefix"] = args ? args.ipv4Prefix : undefined;
            resourceInputs["ipv6Prefix"] = args ? args.ipv6Prefix : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["managementNetworkConfiguration"] = args ? (args.managementNetworkConfiguration ? pulumi.output(args.managementNetworkConfiguration).apply(inputs.managednetworkfabric.managementNetworkConfigurationPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["networkFabricControllerId"] = args ? args.networkFabricControllerId : undefined;
            resourceInputs["networkFabricName"] = args ? args.networkFabricName : undefined;
            resourceInputs["networkFabricSku"] = args ? args.networkFabricSku : undefined;
            resourceInputs["rackCount"] = args ? args.rackCount : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverCountPerRack"] = args ? args.serverCountPerRack : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["terminalServerConfiguration"] = args ? args.terminalServerConfiguration : undefined;
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["l2IsolationDomains"] = undefined /*out*/;
            resourceInputs["l3IsolationDomains"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["racks"] = undefined /*out*/;
            resourceInputs["routerIds"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["administrativeState"] = undefined /*out*/;
            resourceInputs["annotation"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["configurationState"] = undefined /*out*/;
            resourceInputs["fabricASN"] = undefined /*out*/;
            resourceInputs["fabricVersion"] = undefined /*out*/;
            resourceInputs["ipv4Prefix"] = undefined /*out*/;
            resourceInputs["ipv6Prefix"] = undefined /*out*/;
            resourceInputs["l2IsolationDomains"] = undefined /*out*/;
            resourceInputs["l3IsolationDomains"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managementNetworkConfiguration"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkFabricControllerId"] = undefined /*out*/;
            resourceInputs["networkFabricSku"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["rackCount"] = undefined /*out*/;
            resourceInputs["racks"] = undefined /*out*/;
            resourceInputs["routerIds"] = undefined /*out*/;
            resourceInputs["serverCountPerRack"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["terminalServerConfiguration"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:managednetworkfabric/v20230201preview:NetworkFabric" }, { type: "azure-native:managednetworkfabric/v20230615:NetworkFabric" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkFabric.__pulumiType, name, resourceInputs, opts);
    }
}
exports.NetworkFabric = NetworkFabric;
/** @internal */
NetworkFabric.__pulumiType = 'azure-native:managednetworkfabric:NetworkFabric';
