// @pulumi/azure-native/web/webAppSourceControlSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSourceControlSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Source control configuration for an app.
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2016-08-01, 2018-02-01, 2018-11-01, 2019-08-01, 2020-06-01, 2020-09-01, 2020-10-01, 2020-12-01, 2021-01-01, 2021-01-15, 2021-02-01, 2021-03-01, 2022-03-01, 2022-09-01, 2023-01-01, 2023-12-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native web [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebAppSourceControlSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppSourceControlSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppSourceControlSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppSourceControlSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppSourceControlSlot.__pulumiType;
    }
    /**
     * Create a WebAppSourceControlSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["branch"] = args ? args.branch : undefined;
            resourceInputs["deploymentRollbackEnabled"] = args ? args.deploymentRollbackEnabled : undefined;
            resourceInputs["gitHubActionConfiguration"] = args ? args.gitHubActionConfiguration : undefined;
            resourceInputs["isGitHubAction"] = args ? args.isGitHubAction : undefined;
            resourceInputs["isManualIntegration"] = args ? args.isManualIntegration : undefined;
            resourceInputs["isMercurial"] = args ? args.isMercurial : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["repoUrl"] = args ? args.repoUrl : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["branch"] = undefined /*out*/;
            resourceInputs["deploymentRollbackEnabled"] = undefined /*out*/;
            resourceInputs["gitHubActionConfiguration"] = undefined /*out*/;
            resourceInputs["isGitHubAction"] = undefined /*out*/;
            resourceInputs["isManualIntegration"] = undefined /*out*/;
            resourceInputs["isMercurial"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["repoUrl"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20150801:WebAppSourceControlSlot" }, { type: "azure-native:web/v20160801:WebAppSourceControlSlot" }, { type: "azure-native:web/v20180201:WebAppSourceControlSlot" }, { type: "azure-native:web/v20181101:WebAppSourceControlSlot" }, { type: "azure-native:web/v20190801:WebAppSourceControlSlot" }, { type: "azure-native:web/v20200601:WebAppSourceControlSlot" }, { type: "azure-native:web/v20200901:WebAppSourceControlSlot" }, { type: "azure-native:web/v20201001:WebAppSourceControlSlot" }, { type: "azure-native:web/v20201201:WebAppSourceControlSlot" }, { type: "azure-native:web/v20210101:WebAppSourceControlSlot" }, { type: "azure-native:web/v20210115:WebAppSourceControlSlot" }, { type: "azure-native:web/v20210201:WebAppSourceControlSlot" }, { type: "azure-native:web/v20210301:WebAppSourceControlSlot" }, { type: "azure-native:web/v20220301:WebAppSourceControlSlot" }, { type: "azure-native:web/v20220901:WebAppSourceControlSlot" }, { type: "azure-native:web/v20230101:WebAppSourceControlSlot" }, { type: "azure-native:web/v20231201:WebAppSourceControlSlot" }, { type: "azure-native:web/v20240401:WebAppSourceControlSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppSourceControlSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppSourceControlSlot = WebAppSourceControlSlot;
/** @internal */
WebAppSourceControlSlot.__pulumiType = 'azure-native:web:WebAppSourceControlSlot';
