// @pulumi/azure-native/web/webAppSitePushSettingsSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSitePushSettingsSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Push settings for the App.
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2016-08-01, 2018-02-01, 2018-11-01, 2019-08-01, 2020-06-01, 2020-09-01, 2020-10-01, 2020-12-01, 2021-01-01, 2021-01-15, 2021-02-01, 2021-03-01, 2022-03-01, 2022-09-01, 2023-01-01, 2023-12-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native web [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebAppSitePushSettingsSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppSitePushSettingsSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppSitePushSettingsSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppSitePushSettingsSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppSitePushSettingsSlot.__pulumiType;
    }
    /**
     * Create a WebAppSitePushSettingsSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.isPushEnabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'isPushEnabled'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["dynamicTagsJson"] = args ? args.dynamicTagsJson : undefined;
            resourceInputs["isPushEnabled"] = args ? args.isPushEnabled : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["tagWhitelistJson"] = args ? args.tagWhitelistJson : undefined;
            resourceInputs["tagsRequiringAuth"] = args ? args.tagsRequiringAuth : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["dynamicTagsJson"] = undefined /*out*/;
            resourceInputs["isPushEnabled"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tagWhitelistJson"] = undefined /*out*/;
            resourceInputs["tagsRequiringAuth"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20160801:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20180201:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20181101:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20190801:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20200601:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20200901:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20201001:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20201201:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20210101:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20210115:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20210201:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20210301:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20220301:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20220901:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20230101:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20231201:WebAppSitePushSettingsSlot" }, { type: "azure-native:web/v20240401:WebAppSitePushSettingsSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppSitePushSettingsSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppSitePushSettingsSlot = WebAppSitePushSettingsSlot;
/** @internal */
WebAppSitePushSettingsSlot.__pulumiType = 'azure-native:web:WebAppSitePushSettingsSlot';
