// @pulumi/azure-native/web/webAppAuthSettingsSlot.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppAuthSettingsSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Configuration settings for the Azure App Service Authentication / Authorization feature.
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2022-09-01.
 *
 * Other available API versions: 2016-08-01, 2018-02-01, 2018-11-01, 2019-08-01, 2020-06-01, 2020-09-01, 2020-10-01, 2020-12-01, 2021-01-01, 2021-01-15, 2021-02-01, 2021-03-01, 2022-03-01, 2022-09-01, 2023-01-01, 2023-12-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native web [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class WebAppAuthSettingsSlot extends pulumi.CustomResource {
    /**
     * Get an existing WebAppAuthSettingsSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppAuthSettingsSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppAuthSettingsSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppAuthSettingsSlot.__pulumiType;
    }
    /**
     * Create a WebAppAuthSettingsSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["aadClaimsAuthorization"] = args ? args.aadClaimsAuthorization : undefined;
            resourceInputs["additionalLoginParams"] = args ? args.additionalLoginParams : undefined;
            resourceInputs["allowedAudiences"] = args ? args.allowedAudiences : undefined;
            resourceInputs["allowedExternalRedirectUrls"] = args ? args.allowedExternalRedirectUrls : undefined;
            resourceInputs["authFilePath"] = args ? args.authFilePath : undefined;
            resourceInputs["clientId"] = args ? args.clientId : undefined;
            resourceInputs["clientSecret"] = args ? args.clientSecret : undefined;
            resourceInputs["clientSecretCertificateThumbprint"] = args ? args.clientSecretCertificateThumbprint : undefined;
            resourceInputs["clientSecretSettingName"] = args ? args.clientSecretSettingName : undefined;
            resourceInputs["configVersion"] = args ? args.configVersion : undefined;
            resourceInputs["defaultProvider"] = args ? args.defaultProvider : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["facebookAppId"] = args ? args.facebookAppId : undefined;
            resourceInputs["facebookAppSecret"] = args ? args.facebookAppSecret : undefined;
            resourceInputs["facebookAppSecretSettingName"] = args ? args.facebookAppSecretSettingName : undefined;
            resourceInputs["facebookOAuthScopes"] = args ? args.facebookOAuthScopes : undefined;
            resourceInputs["gitHubClientId"] = args ? args.gitHubClientId : undefined;
            resourceInputs["gitHubClientSecret"] = args ? args.gitHubClientSecret : undefined;
            resourceInputs["gitHubClientSecretSettingName"] = args ? args.gitHubClientSecretSettingName : undefined;
            resourceInputs["gitHubOAuthScopes"] = args ? args.gitHubOAuthScopes : undefined;
            resourceInputs["googleClientId"] = args ? args.googleClientId : undefined;
            resourceInputs["googleClientSecret"] = args ? args.googleClientSecret : undefined;
            resourceInputs["googleClientSecretSettingName"] = args ? args.googleClientSecretSettingName : undefined;
            resourceInputs["googleOAuthScopes"] = args ? args.googleOAuthScopes : undefined;
            resourceInputs["isAuthFromFile"] = args ? args.isAuthFromFile : undefined;
            resourceInputs["issuer"] = args ? args.issuer : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["microsoftAccountClientId"] = args ? args.microsoftAccountClientId : undefined;
            resourceInputs["microsoftAccountClientSecret"] = args ? args.microsoftAccountClientSecret : undefined;
            resourceInputs["microsoftAccountClientSecretSettingName"] = args ? args.microsoftAccountClientSecretSettingName : undefined;
            resourceInputs["microsoftAccountOAuthScopes"] = args ? args.microsoftAccountOAuthScopes : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["runtimeVersion"] = args ? args.runtimeVersion : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["tokenRefreshExtensionHours"] = args ? args.tokenRefreshExtensionHours : undefined;
            resourceInputs["tokenStoreEnabled"] = args ? args.tokenStoreEnabled : undefined;
            resourceInputs["twitterConsumerKey"] = args ? args.twitterConsumerKey : undefined;
            resourceInputs["twitterConsumerSecret"] = args ? args.twitterConsumerSecret : undefined;
            resourceInputs["twitterConsumerSecretSettingName"] = args ? args.twitterConsumerSecretSettingName : undefined;
            resourceInputs["unauthenticatedClientAction"] = args ? args.unauthenticatedClientAction : undefined;
            resourceInputs["validateIssuer"] = args ? args.validateIssuer : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["aadClaimsAuthorization"] = undefined /*out*/;
            resourceInputs["additionalLoginParams"] = undefined /*out*/;
            resourceInputs["allowedAudiences"] = undefined /*out*/;
            resourceInputs["allowedExternalRedirectUrls"] = undefined /*out*/;
            resourceInputs["authFilePath"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["clientId"] = undefined /*out*/;
            resourceInputs["clientSecret"] = undefined /*out*/;
            resourceInputs["clientSecretCertificateThumbprint"] = undefined /*out*/;
            resourceInputs["clientSecretSettingName"] = undefined /*out*/;
            resourceInputs["configVersion"] = undefined /*out*/;
            resourceInputs["defaultProvider"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["facebookAppId"] = undefined /*out*/;
            resourceInputs["facebookAppSecret"] = undefined /*out*/;
            resourceInputs["facebookAppSecretSettingName"] = undefined /*out*/;
            resourceInputs["facebookOAuthScopes"] = undefined /*out*/;
            resourceInputs["gitHubClientId"] = undefined /*out*/;
            resourceInputs["gitHubClientSecret"] = undefined /*out*/;
            resourceInputs["gitHubClientSecretSettingName"] = undefined /*out*/;
            resourceInputs["gitHubOAuthScopes"] = undefined /*out*/;
            resourceInputs["googleClientId"] = undefined /*out*/;
            resourceInputs["googleClientSecret"] = undefined /*out*/;
            resourceInputs["googleClientSecretSettingName"] = undefined /*out*/;
            resourceInputs["googleOAuthScopes"] = undefined /*out*/;
            resourceInputs["isAuthFromFile"] = undefined /*out*/;
            resourceInputs["issuer"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["microsoftAccountClientId"] = undefined /*out*/;
            resourceInputs["microsoftAccountClientSecret"] = undefined /*out*/;
            resourceInputs["microsoftAccountClientSecretSettingName"] = undefined /*out*/;
            resourceInputs["microsoftAccountOAuthScopes"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["runtimeVersion"] = undefined /*out*/;
            resourceInputs["tokenRefreshExtensionHours"] = undefined /*out*/;
            resourceInputs["tokenStoreEnabled"] = undefined /*out*/;
            resourceInputs["twitterConsumerKey"] = undefined /*out*/;
            resourceInputs["twitterConsumerSecret"] = undefined /*out*/;
            resourceInputs["twitterConsumerSecretSettingName"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["unauthenticatedClientAction"] = undefined /*out*/;
            resourceInputs["validateIssuer"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web/v20150801:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20160801:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20180201:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20181101:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20190801:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20200601:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20200901:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20201001:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20201201:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20210101:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20210115:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20210201:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20210301:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20220301:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20220901:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20230101:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20231201:WebAppAuthSettingsSlot" }, { type: "azure-native:web/v20240401:WebAppAuthSettingsSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppAuthSettingsSlot.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAppAuthSettingsSlot = WebAppAuthSettingsSlot;
/** @internal */
WebAppAuthSettingsSlot.__pulumiType = 'azure-native:web:WebAppAuthSettingsSlot';
