// @pulumi/azure-native/signalrservice/signalRPrivateEndpointConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalRPrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A private endpoint connection to an azure resource
 *
 * Uses Azure REST API version 2024-03-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2024-01-01-preview, 2024-04-01-preview, 2024-08-01-preview, 2024-10-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native signalrservice [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SignalRPrivateEndpointConnection extends pulumi.CustomResource {
    /**
     * Get an existing SignalRPrivateEndpointConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SignalRPrivateEndpointConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SignalRPrivateEndpointConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SignalRPrivateEndpointConnection.__pulumiType;
    }
    /**
     * Create a SignalRPrivateEndpointConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["privateEndpoint"] = args ? args.privateEndpoint : undefined;
            resourceInputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            resourceInputs["privateLinkServiceConnectionState"] = args ? args.privateLinkServiceConnectionState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["groupIds"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["groupIds"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["privateLinkServiceConnectionState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:signalrservice/v20200501:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20200701preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20210401preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20210601preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20210901preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20211001:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20220201:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20220801preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20230201:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20230301preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20230601preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20230801preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20240101preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20240301:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20240401preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20240801preview:SignalRPrivateEndpointConnection" }, { type: "azure-native:signalrservice/v20241001preview:SignalRPrivateEndpointConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SignalRPrivateEndpointConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SignalRPrivateEndpointConnection = SignalRPrivateEndpointConnection;
/** @internal */
SignalRPrivateEndpointConnection.__pulumiType = 'azure-native:signalrservice:SignalRPrivateEndpointConnection';
