// @pulumi/azure-native/deviceprovisioningservices/iotDpsResourcePrivateEndpointConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotDpsResourcePrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The private endpoint connection of a provisioning service
 *
 * Uses Azure REST API version 2023-03-01-preview.
 *
 * Other available API versions: 2020-03-01, 2020-09-01-preview, 2021-10-15, 2022-02-05, 2022-12-12, 2025-02-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native deviceprovisioningservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class IotDpsResourcePrivateEndpointConnection extends pulumi.CustomResource {
    /**
     * Get an existing IotDpsResourcePrivateEndpointConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IotDpsResourcePrivateEndpointConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotDpsResourcePrivateEndpointConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotDpsResourcePrivateEndpointConnection.__pulumiType;
    }
    /**
     * Create a IotDpsResourcePrivateEndpointConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:deviceprovisioningservices/v20200301:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:deviceprovisioningservices/v20200901preview:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:deviceprovisioningservices/v20211015:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:deviceprovisioningservices/v20220205:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:deviceprovisioningservices/v20221212:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:deviceprovisioningservices/v20230301preview:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:deviceprovisioningservices/v20250201preview:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:devices/v20221212:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:devices/v20230301preview:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:devices/v20250201preview:IotDpsResourcePrivateEndpointConnection" }, { type: "azure-native:devices:IotDpsResourcePrivateEndpointConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IotDpsResourcePrivateEndpointConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IotDpsResourcePrivateEndpointConnection = IotDpsResourcePrivateEndpointConnection;
/** @internal */
IotDpsResourcePrivateEndpointConnection.__pulumiType = 'azure-native:deviceprovisioningservices:IotDpsResourcePrivateEndpointConnection';
