// @pulumi/azure-native/marketplace/privateStoreCollection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateStoreCollection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Collection data structure.
 *
 * Uses Azure REST API version 2023-01-01. In version 2.x of the Azure Native provider, it used API version 2023-01-01.
 *
 * Other available API versions: 2025-01-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native marketplace [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class PrivateStoreCollection extends pulumi.CustomResource {
    /**
     * Get an existing PrivateStoreCollection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateStoreCollection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateStoreCollection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateStoreCollection.__pulumiType;
    }
    /**
     * Create a PrivateStoreCollection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.privateStoreId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateStoreId'");
            }
            resourceInputs["allSubscriptions"] = args ? args.allSubscriptions : undefined;
            resourceInputs["claim"] = args ? args.claim : undefined;
            resourceInputs["collectionId"] = args ? args.collectionId : undefined;
            resourceInputs["collectionName"] = args ? args.collectionName : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["privateStoreId"] = args ? args.privateStoreId : undefined;
            resourceInputs["subscriptionsList"] = args ? args.subscriptionsList : undefined;
            resourceInputs["appliedRules"] = undefined /*out*/;
            resourceInputs["approveAllItems"] = undefined /*out*/;
            resourceInputs["approveAllItemsModifiedAt"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfOffers"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allSubscriptions"] = undefined /*out*/;
            resourceInputs["appliedRules"] = undefined /*out*/;
            resourceInputs["approveAllItems"] = undefined /*out*/;
            resourceInputs["approveAllItemsModifiedAt"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["claim"] = undefined /*out*/;
            resourceInputs["collectionId"] = undefined /*out*/;
            resourceInputs["collectionName"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfOffers"] = undefined /*out*/;
            resourceInputs["subscriptionsList"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:marketplace/v20210601:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20211201:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20220301:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20220901:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20230101:PrivateStoreCollection" }, { type: "azure-native:marketplace/v20250101:PrivateStoreCollection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateStoreCollection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateStoreCollection = PrivateStoreCollection;
/** @internal */
PrivateStoreCollection.__pulumiType = 'azure-native:marketplace:PrivateStoreCollection';
