// @pulumi/azure/compute/restorePoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RestorePoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Virtual Machine Restore Point.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as std from "@pulumi/std";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("example", {
 *     name: "example-network",
 *     addressSpaces: ["10.0.0.0/16"],
 *     location: example.location,
 *     resourceGroupName: example.name,
 * });
 * const exampleSubnet = new azure.network.Subnet("example", {
 *     name: "internal",
 *     resourceGroupName: example.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.0.2.0/24"],
 * });
 * const exampleNetworkInterface = new azure.network.NetworkInterface("example", {
 *     name: "example-nic",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     ipConfigurations: [{
 *         name: "internal",
 *         subnetId: exampleSubnet.id,
 *         privateIpAddressAllocation: "Dynamic",
 *     }],
 * });
 * const exampleLinuxVirtualMachine = new azure.compute.LinuxVirtualMachine("example", {
 *     name: "example-machine",
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     size: "Standard_F2",
 *     adminUsername: "adminuser",
 *     networkInterfaceIds: [exampleNetworkInterface.id],
 *     adminSshKeys: [{
 *         username: "adminuser",
 *         publicKey: std.file({
 *             input: "~/.ssh/id_rsa.pub",
 *         }).then(invoke => invoke.result),
 *     }],
 *     osDisk: {
 *         caching: "ReadWrite",
 *         storageAccountType: "Standard_LRS",
 *     },
 *     sourceImageReference: {
 *         publisher: "Canonical",
 *         offer: "0001-com-ubuntu-server-jammy",
 *         sku: "22_04-lts",
 *         version: "latest",
 *     },
 * });
 * const exampleRestorePointCollection = new azure.compute.RestorePointCollection("example", {
 *     name: "example-collection",
 *     resourceGroupName: example.name,
 *     location: exampleLinuxVirtualMachine.location,
 *     sourceVirtualMachineId: exampleLinuxVirtualMachine.id,
 * });
 * const exampleRestorePoint = new azure.compute.RestorePoint("example", {
 *     name: "example-restore-point",
 *     virtualMachineRestorePointCollectionId: exampleRestorePointCollection.id,
 * });
 * ```
 *
 * ## API Providers
 *
 * <!-- This section is generated, changes will be overwritten -->
 * This resource uses the following Azure API Providers:
 *
 * * `Microsoft.Compute`: 2024-03-01
 *
 * ## Import
 *
 * Virtual Machine Restore Point can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:compute/restorePoint:RestorePoint example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Compute/restorePointCollections/collection1/restorePoints/restorePoint1
 * ```
 */
class RestorePoint extends pulumi.CustomResource {
    /**
     * Get an existing RestorePoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RestorePoint(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RestorePoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RestorePoint.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["crashConsistencyModeEnabled"] = state ? state.crashConsistencyModeEnabled : undefined;
            resourceInputs["excludedDisks"] = state ? state.excludedDisks : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["virtualMachineRestorePointCollectionId"] = state ? state.virtualMachineRestorePointCollectionId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.virtualMachineRestorePointCollectionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'virtualMachineRestorePointCollectionId'");
            }
            resourceInputs["crashConsistencyModeEnabled"] = args ? args.crashConsistencyModeEnabled : undefined;
            resourceInputs["excludedDisks"] = args ? args.excludedDisks : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["virtualMachineRestorePointCollectionId"] = args ? args.virtualMachineRestorePointCollectionId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RestorePoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RestorePoint = RestorePoint;
/** @internal */
RestorePoint.__pulumiType = 'azure:compute/restorePoint:RestorePoint';
