// @pulumi/azure-native/types/enums/containerregistry/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundancy = exports.WebhookStatus = exports.WebhookAction = exports.Variant = exports.UpdateTriggerPayloadType = exports.TrustPolicyType = exports.TriggerStatus = exports.TokenType = exports.TokenStatus = exports.TokenPasswordName = exports.TokenCertificateName = exports.TaskStatus = exports.StepType = exports.SourceTriggerEvent = exports.SourceRegistryLoginMode = exports.SourceControlType = exports.SkuName = exports.SecretObjectType = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.PolicyStatus = exports.PipelineSourceType = exports.PipelineRunTargetType = exports.PipelineRunSourceType = exports.PipelineOptions = exports.PackageSourceType = exports.OS = exports.NetworkRuleBypassOptions = exports.LogLevel = exports.ExportPolicyStatus = exports.EncryptionStatus = exports.DefaultAction = exports.CredentialName = exports.ConnectionStatus = exports.ConnectedRegistryMode = exports.BaseImageTriggerType = exports.AzureADAuthenticationAsArmPolicyStatus = exports.AuditLogStatus = exports.Architecture = exports.ActionsRequired = exports.Action = void 0;
exports.Action = {
    Allow: "Allow",
};
exports.ActionsRequired = {
    None: "None",
    Recreate: "Recreate",
};
exports.Architecture = {
    Amd64: "amd64",
    X86: "x86",
    Architecture_386: "386",
    Arm: "arm",
    Arm64: "arm64",
};
exports.AuditLogStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.AzureADAuthenticationAsArmPolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.BaseImageTriggerType = {
    All: "All",
    Runtime: "Runtime",
};
exports.ConnectedRegistryMode = {
    ReadWrite: "ReadWrite",
    ReadOnly: "ReadOnly",
    Registry: "Registry",
    Mirror: "Mirror",
};
exports.ConnectionStatus = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.CredentialName = {
    Credential1: "Credential1",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EncryptionStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ExportPolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.LogLevel = {
    Debug: "Debug",
    Information: "Information",
    Warning: "Warning",
    Error: "Error",
    None: "None",
};
exports.NetworkRuleBypassOptions = {
    AzureServices: "AzureServices",
    None: "None",
};
exports.OS = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.PackageSourceType = {
    Remote: "remote",
};
exports.PipelineOptions = {
    OverwriteTags: "OverwriteTags",
    OverwriteBlobs: "OverwriteBlobs",
    DeleteSourceBlobOnSuccess: "DeleteSourceBlobOnSuccess",
    ContinueOnErrors: "ContinueOnErrors",
};
exports.PipelineRunSourceType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineRunTargetType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineSourceType = {
    AzureStorageBlobContainer: "AzureStorageBlobContainer",
};
exports.PolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.SecretObjectType = {
    Opaque: "Opaque",
    Vaultsecret: "Vaultsecret",
};
exports.SkuName = {
    Classic: "Classic",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SourceControlType = {
    Github: "Github",
    VisualStudioTeamService: "VisualStudioTeamService",
};
exports.SourceRegistryLoginMode = {
    None: "None",
    Default: "Default",
};
exports.SourceTriggerEvent = {
    Commit: "commit",
    Pullrequest: "pullrequest",
};
exports.StepType = {
    Docker: "Docker",
    FileTask: "FileTask",
    EncodedTask: "EncodedTask",
};
exports.TaskStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.TokenCertificateName = {
    Certificate1: "certificate1",
    Certificate2: "certificate2",
};
exports.TokenPasswordName = {
    Password1: "password1",
    Password2: "password2",
};
exports.TokenStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.TokenType = {
    PAT: "PAT",
    OAuth: "OAuth",
};
exports.TriggerStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.TrustPolicyType = {
    Notary: "Notary",
};
exports.UpdateTriggerPayloadType = {
    Default: "Default",
    Token: "Token",
};
exports.Variant = {
    V6: "v6",
    V7: "v7",
    V8: "v8",
};
exports.WebhookAction = {
    Push: "push",
    Delete: "delete",
    Quarantine: "quarantine",
    Chart_push: "chart_push",
    Chart_delete: "chart_delete",
};
exports.WebhookStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ZoneRedundancy = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
