// @pulumi/azure-native/authorization/scopeAccessReviewScheduleDefinitionById.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScopeAccessReviewScheduleDefinitionById = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Access Review Schedule Definition.
 *
 * Uses Azure REST API version 2021-12-01-preview. In version 2.x of the Azure Native provider, it used API version 2021-12-01-preview.
 */
class ScopeAccessReviewScheduleDefinitionById extends pulumi.CustomResource {
    /**
     * Get an existing ScopeAccessReviewScheduleDefinitionById resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScopeAccessReviewScheduleDefinitionById(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScopeAccessReviewScheduleDefinitionById.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScopeAccessReviewScheduleDefinitionById.__pulumiType;
    }
    /**
     * Create a ScopeAccessReviewScheduleDefinitionById resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            resourceInputs["autoApplyDecisionsEnabled"] = args ? args.autoApplyDecisionsEnabled : undefined;
            resourceInputs["backupReviewers"] = args ? args.backupReviewers : undefined;
            resourceInputs["defaultDecision"] = args ? args.defaultDecision : undefined;
            resourceInputs["defaultDecisionEnabled"] = args ? args.defaultDecisionEnabled : undefined;
            resourceInputs["descriptionForAdmins"] = args ? args.descriptionForAdmins : undefined;
            resourceInputs["descriptionForReviewers"] = args ? args.descriptionForReviewers : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["excludeResourceId"] = args ? args.excludeResourceId : undefined;
            resourceInputs["excludeRoleDefinitionId"] = args ? args.excludeRoleDefinitionId : undefined;
            resourceInputs["expandNestedMemberships"] = args ? args.expandNestedMemberships : undefined;
            resourceInputs["inactiveDuration"] = args ? args.inactiveDuration : undefined;
            resourceInputs["includeAccessBelowResource"] = args ? args.includeAccessBelowResource : undefined;
            resourceInputs["includeInheritedAccess"] = args ? args.includeInheritedAccess : undefined;
            resourceInputs["instanceDurationInDays"] = args ? args.instanceDurationInDays : undefined;
            resourceInputs["instances"] = args ? args.instances : undefined;
            resourceInputs["interval"] = args ? args.interval : undefined;
            resourceInputs["justificationRequiredOnApproval"] = args ? args.justificationRequiredOnApproval : undefined;
            resourceInputs["mailNotificationsEnabled"] = args ? args.mailNotificationsEnabled : undefined;
            resourceInputs["range"] = args ? args.range : undefined;
            resourceInputs["recommendationLookBackDuration"] = args ? args.recommendationLookBackDuration : undefined;
            resourceInputs["recommendationsEnabled"] = args ? args.recommendationsEnabled : undefined;
            resourceInputs["reminderNotificationsEnabled"] = args ? args.reminderNotificationsEnabled : undefined;
            resourceInputs["reviewers"] = args ? args.reviewers : undefined;
            resourceInputs["scheduleDefinitionId"] = args ? args.scheduleDefinitionId : undefined;
            resourceInputs["scope"] = args ? args.scope : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalId"] = undefined /*out*/;
            resourceInputs["principalName"] = undefined /*out*/;
            resourceInputs["principalType"] = undefined /*out*/;
            resourceInputs["reviewersType"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["userPrincipalName"] = undefined /*out*/;
        }
        else {
            resourceInputs["autoApplyDecisionsEnabled"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backupReviewers"] = undefined /*out*/;
            resourceInputs["defaultDecision"] = undefined /*out*/;
            resourceInputs["defaultDecisionEnabled"] = undefined /*out*/;
            resourceInputs["descriptionForAdmins"] = undefined /*out*/;
            resourceInputs["descriptionForReviewers"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["instanceDurationInDays"] = undefined /*out*/;
            resourceInputs["instances"] = undefined /*out*/;
            resourceInputs["interval"] = undefined /*out*/;
            resourceInputs["justificationRequiredOnApproval"] = undefined /*out*/;
            resourceInputs["mailNotificationsEnabled"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalId"] = undefined /*out*/;
            resourceInputs["principalName"] = undefined /*out*/;
            resourceInputs["principalType"] = undefined /*out*/;
            resourceInputs["range"] = undefined /*out*/;
            resourceInputs["recommendationLookBackDuration"] = undefined /*out*/;
            resourceInputs["recommendationsEnabled"] = undefined /*out*/;
            resourceInputs["reminderNotificationsEnabled"] = undefined /*out*/;
            resourceInputs["reviewers"] = undefined /*out*/;
            resourceInputs["reviewersType"] = undefined /*out*/;
            resourceInputs["scope"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userPrincipalName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:authorization/v20211201preview:ScopeAccessReviewScheduleDefinitionById" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ScopeAccessReviewScheduleDefinitionById.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ScopeAccessReviewScheduleDefinitionById = ScopeAccessReviewScheduleDefinitionById;
/** @internal */
ScopeAccessReviewScheduleDefinitionById.__pulumiType = 'azure-native:authorization:ScopeAccessReviewScheduleDefinitionById';
