// @pulumi/azure-native/authorization/scopeAccessReviewHistoryDefinitionById.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScopeAccessReviewHistoryDefinitionById = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Access Review History Definition.
 *
 * Uses Azure REST API version 2021-12-01-preview. In version 2.x of the Azure Native provider, it used API version 2021-12-01-preview.
 */
class ScopeAccessReviewHistoryDefinitionById extends pulumi.CustomResource {
    /**
     * Get an existing ScopeAccessReviewHistoryDefinitionById resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScopeAccessReviewHistoryDefinitionById(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScopeAccessReviewHistoryDefinitionById.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScopeAccessReviewHistoryDefinitionById.__pulumiType;
    }
    /**
     * Create a ScopeAccessReviewHistoryDefinitionById resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            resourceInputs["decisions"] = args ? args.decisions : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["historyDefinitionId"] = args ? args.historyDefinitionId : undefined;
            resourceInputs["instances"] = args ? args.instances : undefined;
            resourceInputs["interval"] = args ? args.interval : undefined;
            resourceInputs["range"] = args ? args.range : undefined;
            resourceInputs["scope"] = args ? args.scope : undefined;
            resourceInputs["scopes"] = args ? args.scopes : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdDateTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalId"] = undefined /*out*/;
            resourceInputs["principalName"] = undefined /*out*/;
            resourceInputs["principalType"] = undefined /*out*/;
            resourceInputs["reviewHistoryPeriodEndDateTime"] = undefined /*out*/;
            resourceInputs["reviewHistoryPeriodStartDateTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["userPrincipalName"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdDateTime"] = undefined /*out*/;
            resourceInputs["decisions"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["instances"] = undefined /*out*/;
            resourceInputs["interval"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["principalId"] = undefined /*out*/;
            resourceInputs["principalName"] = undefined /*out*/;
            resourceInputs["principalType"] = undefined /*out*/;
            resourceInputs["range"] = undefined /*out*/;
            resourceInputs["reviewHistoryPeriodEndDateTime"] = undefined /*out*/;
            resourceInputs["reviewHistoryPeriodStartDateTime"] = undefined /*out*/;
            resourceInputs["scopes"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userPrincipalName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:authorization/v20211201preview:ScopeAccessReviewHistoryDefinitionById" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ScopeAccessReviewHistoryDefinitionById.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ScopeAccessReviewHistoryDefinitionById = ScopeAccessReviewHistoryDefinitionById;
/** @internal */
ScopeAccessReviewHistoryDefinitionById.__pulumiType = 'azure-native:authorization:ScopeAccessReviewHistoryDefinitionById';
