// @pulumi/azure-native/netapp/account.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * NetApp account resource
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2022-11-01.
 *
 * Other available API versions: 2022-11-01, 2022-11-01-preview, 2023-05-01, 2023-05-01-preview, 2023-07-01, 2023-07-01-preview, 2023-11-01, 2023-11-01-preview, 2024-01-01, 2024-03-01, 2024-03-01-preview, 2024-05-01, 2024-05-01-preview, 2024-07-01, 2024-07-01-preview, 2024-09-01-preview, 2025-01-01, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native netapp [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Account extends pulumi.CustomResource {
    /**
     * Get an existing Account resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Account(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Account.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Account.__pulumiType;
    }
    /**
     * Create a Account resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["activeDirectories"] = args ? args.activeDirectories : undefined;
            resourceInputs["encryption"] = args ? (args.encryption ? pulumi.output(args.encryption).apply(inputs.netapp.accountEncryptionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["disableShowmount"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["activeDirectories"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["disableShowmount"] = undefined /*out*/;
            resourceInputs["encryption"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20170815:Account" }, { type: "azure-native:netapp/v20190501:Account" }, { type: "azure-native:netapp/v20190601:Account" }, { type: "azure-native:netapp/v20190701:Account" }, { type: "azure-native:netapp/v20190801:Account" }, { type: "azure-native:netapp/v20191001:Account" }, { type: "azure-native:netapp/v20191101:Account" }, { type: "azure-native:netapp/v20200201:Account" }, { type: "azure-native:netapp/v20200301:Account" }, { type: "azure-native:netapp/v20200501:Account" }, { type: "azure-native:netapp/v20200601:Account" }, { type: "azure-native:netapp/v20200701:Account" }, { type: "azure-native:netapp/v20200801:Account" }, { type: "azure-native:netapp/v20200901:Account" }, { type: "azure-native:netapp/v20201101:Account" }, { type: "azure-native:netapp/v20201201:Account" }, { type: "azure-native:netapp/v20210201:Account" }, { type: "azure-native:netapp/v20210401:Account" }, { type: "azure-native:netapp/v20210401preview:Account" }, { type: "azure-native:netapp/v20210601:Account" }, { type: "azure-native:netapp/v20210801:Account" }, { type: "azure-native:netapp/v20211001:Account" }, { type: "azure-native:netapp/v20220101:Account" }, { type: "azure-native:netapp/v20220301:Account" }, { type: "azure-native:netapp/v20220501:Account" }, { type: "azure-native:netapp/v20220901:Account" }, { type: "azure-native:netapp/v20221101:Account" }, { type: "azure-native:netapp/v20221101preview:Account" }, { type: "azure-native:netapp/v20230501:Account" }, { type: "azure-native:netapp/v20230501preview:Account" }, { type: "azure-native:netapp/v20230701:Account" }, { type: "azure-native:netapp/v20230701preview:Account" }, { type: "azure-native:netapp/v20231101:Account" }, { type: "azure-native:netapp/v20231101preview:Account" }, { type: "azure-native:netapp/v20240101:Account" }, { type: "azure-native:netapp/v20240301:Account" }, { type: "azure-native:netapp/v20240301preview:Account" }, { type: "azure-native:netapp/v20240501:Account" }, { type: "azure-native:netapp/v20240501preview:Account" }, { type: "azure-native:netapp/v20240701:Account" }, { type: "azure-native:netapp/v20240701preview:Account" }, { type: "azure-native:netapp/v20240901:Account" }, { type: "azure-native:netapp/v20240901preview:Account" }, { type: "azure-native:netapp/v20250101:Account" }, { type: "azure-native:netapp/v20250101preview:Account" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Account.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Account = Account;
/** @internal */
Account.__pulumiType = 'azure-native:netapp:Account';
