// @pulumi/azure-native/netapp/capacityPoolVolumeQuotaRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityPoolVolumeQuotaRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Quota Rule of a Volume
 *
 * Uses Azure REST API version 2024-09-01.
 *
 * Other available API versions: 2022-11-01, 2022-11-01-preview, 2023-05-01, 2023-05-01-preview, 2023-07-01, 2023-07-01-preview, 2023-11-01, 2023-11-01-preview, 2024-01-01, 2024-03-01, 2024-03-01-preview, 2024-05-01, 2024-05-01-preview, 2024-07-01, 2024-07-01-preview, 2024-09-01-preview, 2025-01-01, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native netapp [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class CapacityPoolVolumeQuotaRule extends pulumi.CustomResource {
    /**
     * Get an existing CapacityPoolVolumeQuotaRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CapacityPoolVolumeQuotaRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CapacityPoolVolumeQuotaRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityPoolVolumeQuotaRule.__pulumiType;
    }
    /**
     * Create a CapacityPoolVolumeQuotaRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.poolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.volumeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'volumeName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["quotaSizeInKiBs"] = args ? args.quotaSizeInKiBs : undefined;
            resourceInputs["quotaTarget"] = args ? args.quotaTarget : undefined;
            resourceInputs["quotaType"] = args ? args.quotaType : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["volumeName"] = args ? args.volumeName : undefined;
            resourceInputs["volumeQuotaRuleName"] = args ? args.volumeQuotaRuleName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["quotaSizeInKiBs"] = undefined /*out*/;
            resourceInputs["quotaTarget"] = undefined /*out*/;
            resourceInputs["quotaType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20220101:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20220301:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20220501:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20220901:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20221101:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20221101:VolumeQuotaRule" }, { type: "azure-native:netapp/v20221101preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20221101preview:VolumeQuotaRule" }, { type: "azure-native:netapp/v20230501:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20230501:VolumeQuotaRule" }, { type: "azure-native:netapp/v20230501preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20230501preview:VolumeQuotaRule" }, { type: "azure-native:netapp/v20230701:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20230701:VolumeQuotaRule" }, { type: "azure-native:netapp/v20230701preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20230701preview:VolumeQuotaRule" }, { type: "azure-native:netapp/v20231101:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20231101:VolumeQuotaRule" }, { type: "azure-native:netapp/v20231101preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20231101preview:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240101:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240101:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240301:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240301:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240301preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240301preview:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240501:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240501:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240501preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240501preview:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240701:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240701:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240701preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240701preview:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240901:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20240901:VolumeQuotaRule" }, { type: "azure-native:netapp/v20240901preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20250101:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp/v20250101preview:CapacityPoolVolumeQuotaRule" }, { type: "azure-native:netapp:VolumeQuotaRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CapacityPoolVolumeQuotaRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CapacityPoolVolumeQuotaRule = CapacityPoolVolumeQuotaRule;
/** @internal */
CapacityPoolVolumeQuotaRule.__pulumiType = 'azure-native:netapp:CapacityPoolVolumeQuotaRule';
