// @pulumi/azure-native/netapp/capacityPoolBackup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityPoolBackup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Backup of a Volume
 *
 * Uses Azure REST API version 2022-11-01.
 */
class CapacityPoolBackup extends pulumi.CustomResource {
    /**
     * Get an existing CapacityPoolBackup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CapacityPoolBackup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CapacityPoolBackup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityPoolBackup.__pulumiType;
    }
    /**
     * Create a CapacityPoolBackup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.poolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.volumeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'volumeName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["backupName"] = args ? args.backupName : undefined;
            resourceInputs["label"] = args ? args.label : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["useExistingSnapshot"] = (_a = (args ? args.useExistingSnapshot : undefined)) !== null && _a !== void 0 ? _a : false;
            resourceInputs["volumeName"] = args ? args.volumeName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backupId"] = undefined /*out*/;
            resourceInputs["backupType"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backupId"] = undefined /*out*/;
            resourceInputs["backupType"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["label"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useExistingSnapshot"] = undefined /*out*/;
            resourceInputs["volumeName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20200501:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200601:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200701:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200801:CapacityPoolBackup" }, { type: "azure-native:netapp/v20200901:CapacityPoolBackup" }, { type: "azure-native:netapp/v20201101:CapacityPoolBackup" }, { type: "azure-native:netapp/v20201201:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210201:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210401:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210401preview:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210601:CapacityPoolBackup" }, { type: "azure-native:netapp/v20210801:CapacityPoolBackup" }, { type: "azure-native:netapp/v20211001:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220101:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220301:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220501:CapacityPoolBackup" }, { type: "azure-native:netapp/v20220901:CapacityPoolBackup" }, { type: "azure-native:netapp/v20221101:Backup" }, { type: "azure-native:netapp/v20221101:CapacityPoolBackup" }, { type: "azure-native:netapp:Backup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CapacityPoolBackup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CapacityPoolBackup = CapacityPoolBackup;
/** @internal */
CapacityPoolBackup.__pulumiType = 'azure-native:netapp:CapacityPoolBackup';
