// @pulumi/azure-native/netapp/capacityPool.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Capacity pool resource
 *
 * Uses Azure REST API version 2024-09-01.
 *
 * Other available API versions: 2022-11-01, 2022-11-01-preview, 2023-05-01, 2023-05-01-preview, 2023-07-01, 2023-07-01-preview, 2023-11-01, 2023-11-01-preview, 2024-01-01, 2024-03-01, 2024-03-01-preview, 2024-05-01, 2024-05-01-preview, 2024-07-01, 2024-07-01-preview, 2024-09-01-preview, 2025-01-01, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native netapp [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class CapacityPool extends pulumi.CustomResource {
    /**
     * Get an existing CapacityPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CapacityPool(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CapacityPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityPool.__pulumiType;
    }
    /**
     * Create a CapacityPool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceLevel === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceLevel'");
            }
            if ((!args || args.size === undefined) && !opts.urn) {
                throw new Error("Missing required property 'size'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["coolAccess"] = (_a = (args ? args.coolAccess : undefined)) !== null && _a !== void 0 ? _a : false;
            resourceInputs["encryptionType"] = (_b = (args ? args.encryptionType : undefined)) !== null && _b !== void 0 ? _b : "Single";
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["qosType"] = (_c = (args ? args.qosType : undefined)) !== null && _c !== void 0 ? _c : "Auto";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceLevel"] = args ? args.serviceLevel : undefined;
            resourceInputs["size"] = (_d = (args ? args.size : undefined)) !== null && _d !== void 0 ? _d : 4398046511104;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["poolId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["totalThroughputMibps"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["utilizedThroughputMibps"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["coolAccess"] = undefined /*out*/;
            resourceInputs["encryptionType"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["poolId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["qosType"] = undefined /*out*/;
            resourceInputs["serviceLevel"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["totalThroughputMibps"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["utilizedThroughputMibps"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20170815:CapacityPool" }, { type: "azure-native:netapp/v20190501:CapacityPool" }, { type: "azure-native:netapp/v20190601:CapacityPool" }, { type: "azure-native:netapp/v20190701:CapacityPool" }, { type: "azure-native:netapp/v20190801:CapacityPool" }, { type: "azure-native:netapp/v20191001:CapacityPool" }, { type: "azure-native:netapp/v20191101:CapacityPool" }, { type: "azure-native:netapp/v20200201:CapacityPool" }, { type: "azure-native:netapp/v20200301:CapacityPool" }, { type: "azure-native:netapp/v20200501:CapacityPool" }, { type: "azure-native:netapp/v20200601:CapacityPool" }, { type: "azure-native:netapp/v20200701:CapacityPool" }, { type: "azure-native:netapp/v20200801:CapacityPool" }, { type: "azure-native:netapp/v20200901:CapacityPool" }, { type: "azure-native:netapp/v20201101:CapacityPool" }, { type: "azure-native:netapp/v20201201:CapacityPool" }, { type: "azure-native:netapp/v20210201:CapacityPool" }, { type: "azure-native:netapp/v20210401:CapacityPool" }, { type: "azure-native:netapp/v20210401preview:CapacityPool" }, { type: "azure-native:netapp/v20210601:CapacityPool" }, { type: "azure-native:netapp/v20210801:CapacityPool" }, { type: "azure-native:netapp/v20211001:CapacityPool" }, { type: "azure-native:netapp/v20220101:CapacityPool" }, { type: "azure-native:netapp/v20220301:CapacityPool" }, { type: "azure-native:netapp/v20220501:CapacityPool" }, { type: "azure-native:netapp/v20220901:CapacityPool" }, { type: "azure-native:netapp/v20221101:CapacityPool" }, { type: "azure-native:netapp/v20221101:Pool" }, { type: "azure-native:netapp/v20221101preview:CapacityPool" }, { type: "azure-native:netapp/v20221101preview:Pool" }, { type: "azure-native:netapp/v20230501:CapacityPool" }, { type: "azure-native:netapp/v20230501:Pool" }, { type: "azure-native:netapp/v20230501preview:CapacityPool" }, { type: "azure-native:netapp/v20230501preview:Pool" }, { type: "azure-native:netapp/v20230701:CapacityPool" }, { type: "azure-native:netapp/v20230701:Pool" }, { type: "azure-native:netapp/v20230701preview:CapacityPool" }, { type: "azure-native:netapp/v20230701preview:Pool" }, { type: "azure-native:netapp/v20231101:CapacityPool" }, { type: "azure-native:netapp/v20231101:Pool" }, { type: "azure-native:netapp/v20231101preview:CapacityPool" }, { type: "azure-native:netapp/v20231101preview:Pool" }, { type: "azure-native:netapp/v20240101:CapacityPool" }, { type: "azure-native:netapp/v20240101:Pool" }, { type: "azure-native:netapp/v20240301:CapacityPool" }, { type: "azure-native:netapp/v20240301:Pool" }, { type: "azure-native:netapp/v20240301preview:CapacityPool" }, { type: "azure-native:netapp/v20240301preview:Pool" }, { type: "azure-native:netapp/v20240501:CapacityPool" }, { type: "azure-native:netapp/v20240501:Pool" }, { type: "azure-native:netapp/v20240501preview:CapacityPool" }, { type: "azure-native:netapp/v20240501preview:Pool" }, { type: "azure-native:netapp/v20240701:CapacityPool" }, { type: "azure-native:netapp/v20240701:Pool" }, { type: "azure-native:netapp/v20240701preview:CapacityPool" }, { type: "azure-native:netapp/v20240701preview:Pool" }, { type: "azure-native:netapp/v20240901:CapacityPool" }, { type: "azure-native:netapp/v20240901:Pool" }, { type: "azure-native:netapp/v20240901preview:CapacityPool" }, { type: "azure-native:netapp/v20250101:CapacityPool" }, { type: "azure-native:netapp/v20250101preview:CapacityPool" }, { type: "azure-native:netapp:Pool" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CapacityPool.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CapacityPool = CapacityPool;
/** @internal */
CapacityPool.__pulumiType = 'azure-native:netapp:CapacityPool';
