// @pulumi/azure-native/netapp/capacityPoolVolume.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityPoolVolume = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Volume resource
 *
 * Uses Azure REST API version 2024-09-01.
 *
 * Other available API versions: 2022-11-01, 2022-11-01-preview, 2023-05-01, 2023-05-01-preview, 2023-07-01, 2023-07-01-preview, 2023-11-01, 2023-11-01-preview, 2024-01-01, 2024-03-01, 2024-03-01-preview, 2024-05-01, 2024-05-01-preview, 2024-07-01, 2024-07-01-preview, 2024-09-01-preview, 2025-01-01, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native netapp [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class CapacityPoolVolume extends pulumi.CustomResource {
    /**
     * Get an existing CapacityPoolVolume resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CapacityPoolVolume(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CapacityPoolVolume.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityPoolVolume.__pulumiType;
    }
    /**
     * Create a CapacityPoolVolume resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.creationToken === undefined) && !opts.urn) {
                throw new Error("Missing required property 'creationToken'");
            }
            if ((!args || args.poolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.subnetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetId'");
            }
            if ((!args || args.usageThreshold === undefined) && !opts.urn) {
                throw new Error("Missing required property 'usageThreshold'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["avsDataStore"] = (_a = (args ? args.avsDataStore : undefined)) !== null && _a !== void 0 ? _a : "Disabled";
            resourceInputs["backupId"] = args ? args.backupId : undefined;
            resourceInputs["capacityPoolResourceId"] = args ? args.capacityPoolResourceId : undefined;
            resourceInputs["coolAccess"] = (_b = (args ? args.coolAccess : undefined)) !== null && _b !== void 0 ? _b : false;
            resourceInputs["coolAccessRetrievalPolicy"] = args ? args.coolAccessRetrievalPolicy : undefined;
            resourceInputs["coolAccessTieringPolicy"] = args ? args.coolAccessTieringPolicy : undefined;
            resourceInputs["coolnessPeriod"] = args ? args.coolnessPeriod : undefined;
            resourceInputs["creationToken"] = args ? args.creationToken : undefined;
            resourceInputs["dataProtection"] = args ? args.dataProtection : undefined;
            resourceInputs["defaultGroupQuotaInKiBs"] = (_c = (args ? args.defaultGroupQuotaInKiBs : undefined)) !== null && _c !== void 0 ? _c : 0;
            resourceInputs["defaultUserQuotaInKiBs"] = (_d = (args ? args.defaultUserQuotaInKiBs : undefined)) !== null && _d !== void 0 ? _d : 0;
            resourceInputs["deleteBaseSnapshot"] = args ? args.deleteBaseSnapshot : undefined;
            resourceInputs["enableSubvolumes"] = (_e = (args ? args.enableSubvolumes : undefined)) !== null && _e !== void 0 ? _e : "Disabled";
            resourceInputs["encryptionKeySource"] = (_f = (args ? args.encryptionKeySource : undefined)) !== null && _f !== void 0 ? _f : "Microsoft.NetApp";
            resourceInputs["exportPolicy"] = args ? args.exportPolicy : undefined;
            resourceInputs["isDefaultQuotaEnabled"] = (_g = (args ? args.isDefaultQuotaEnabled : undefined)) !== null && _g !== void 0 ? _g : false;
            resourceInputs["isLargeVolume"] = (_h = (args ? args.isLargeVolume : undefined)) !== null && _h !== void 0 ? _h : false;
            resourceInputs["isRestoring"] = args ? args.isRestoring : undefined;
            resourceInputs["kerberosEnabled"] = (_j = (args ? args.kerberosEnabled : undefined)) !== null && _j !== void 0 ? _j : false;
            resourceInputs["keyVaultPrivateEndpointResourceId"] = args ? args.keyVaultPrivateEndpointResourceId : undefined;
            resourceInputs["ldapEnabled"] = (_k = (args ? args.ldapEnabled : undefined)) !== null && _k !== void 0 ? _k : false;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["networkFeatures"] = args ? args.networkFeatures : undefined;
            resourceInputs["placementRules"] = args ? args.placementRules : undefined;
            resourceInputs["poolName"] = args ? args.poolName : undefined;
            resourceInputs["protocolTypes"] = args ? args.protocolTypes : undefined;
            resourceInputs["proximityPlacementGroup"] = args ? args.proximityPlacementGroup : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["securityStyle"] = (_l = (args ? args.securityStyle : undefined)) !== null && _l !== void 0 ? _l : "unix";
            resourceInputs["serviceLevel"] = args ? args.serviceLevel : undefined;
            resourceInputs["smbAccessBasedEnumeration"] = args ? args.smbAccessBasedEnumeration : undefined;
            resourceInputs["smbContinuouslyAvailable"] = (_m = (args ? args.smbContinuouslyAvailable : undefined)) !== null && _m !== void 0 ? _m : false;
            resourceInputs["smbEncryption"] = (_o = (args ? args.smbEncryption : undefined)) !== null && _o !== void 0 ? _o : false;
            resourceInputs["smbNonBrowsable"] = args ? args.smbNonBrowsable : undefined;
            resourceInputs["snapshotDirectoryVisible"] = (_p = (args ? args.snapshotDirectoryVisible : undefined)) !== null && _p !== void 0 ? _p : true;
            resourceInputs["snapshotId"] = args ? args.snapshotId : undefined;
            resourceInputs["subnetId"] = args ? args.subnetId : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["throughputMibps"] = args ? args.throughputMibps : undefined;
            resourceInputs["unixPermissions"] = args ? args.unixPermissions : undefined;
            resourceInputs["usageThreshold"] = (_q = (args ? args.usageThreshold : undefined)) !== null && _q !== void 0 ? _q : 107374182400;
            resourceInputs["volumeName"] = args ? args.volumeName : undefined;
            resourceInputs["volumeSpecName"] = args ? args.volumeSpecName : undefined;
            resourceInputs["volumeType"] = args ? args.volumeType : undefined;
            resourceInputs["zones"] = args ? args.zones : undefined;
            resourceInputs["actualThroughputMibps"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["baremetalTenantId"] = undefined /*out*/;
            resourceInputs["cloneProgress"] = undefined /*out*/;
            resourceInputs["dataStoreResourceId"] = undefined /*out*/;
            resourceInputs["effectiveNetworkFeatures"] = undefined /*out*/;
            resourceInputs["encrypted"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fileAccessLogs"] = undefined /*out*/;
            resourceInputs["fileSystemId"] = undefined /*out*/;
            resourceInputs["maximumNumberOfFiles"] = undefined /*out*/;
            resourceInputs["mountTargets"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkSiblingSetId"] = undefined /*out*/;
            resourceInputs["originatingResourceId"] = undefined /*out*/;
            resourceInputs["provisionedAvailabilityZone"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["storageToNetworkProximity"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["t2Network"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["volumeGroupName"] = undefined /*out*/;
        }
        else {
            resourceInputs["actualThroughputMibps"] = undefined /*out*/;
            resourceInputs["avsDataStore"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["backupId"] = undefined /*out*/;
            resourceInputs["baremetalTenantId"] = undefined /*out*/;
            resourceInputs["capacityPoolResourceId"] = undefined /*out*/;
            resourceInputs["cloneProgress"] = undefined /*out*/;
            resourceInputs["coolAccess"] = undefined /*out*/;
            resourceInputs["coolAccessRetrievalPolicy"] = undefined /*out*/;
            resourceInputs["coolAccessTieringPolicy"] = undefined /*out*/;
            resourceInputs["coolnessPeriod"] = undefined /*out*/;
            resourceInputs["creationToken"] = undefined /*out*/;
            resourceInputs["dataProtection"] = undefined /*out*/;
            resourceInputs["dataStoreResourceId"] = undefined /*out*/;
            resourceInputs["defaultGroupQuotaInKiBs"] = undefined /*out*/;
            resourceInputs["defaultUserQuotaInKiBs"] = undefined /*out*/;
            resourceInputs["deleteBaseSnapshot"] = undefined /*out*/;
            resourceInputs["effectiveNetworkFeatures"] = undefined /*out*/;
            resourceInputs["enableSubvolumes"] = undefined /*out*/;
            resourceInputs["encrypted"] = undefined /*out*/;
            resourceInputs["encryptionKeySource"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["exportPolicy"] = undefined /*out*/;
            resourceInputs["fileAccessLogs"] = undefined /*out*/;
            resourceInputs["fileSystemId"] = undefined /*out*/;
            resourceInputs["isDefaultQuotaEnabled"] = undefined /*out*/;
            resourceInputs["isLargeVolume"] = undefined /*out*/;
            resourceInputs["isRestoring"] = undefined /*out*/;
            resourceInputs["kerberosEnabled"] = undefined /*out*/;
            resourceInputs["keyVaultPrivateEndpointResourceId"] = undefined /*out*/;
            resourceInputs["ldapEnabled"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maximumNumberOfFiles"] = undefined /*out*/;
            resourceInputs["mountTargets"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkFeatures"] = undefined /*out*/;
            resourceInputs["networkSiblingSetId"] = undefined /*out*/;
            resourceInputs["originatingResourceId"] = undefined /*out*/;
            resourceInputs["placementRules"] = undefined /*out*/;
            resourceInputs["protocolTypes"] = undefined /*out*/;
            resourceInputs["provisionedAvailabilityZone"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["proximityPlacementGroup"] = undefined /*out*/;
            resourceInputs["securityStyle"] = undefined /*out*/;
            resourceInputs["serviceLevel"] = undefined /*out*/;
            resourceInputs["smbAccessBasedEnumeration"] = undefined /*out*/;
            resourceInputs["smbContinuouslyAvailable"] = undefined /*out*/;
            resourceInputs["smbEncryption"] = undefined /*out*/;
            resourceInputs["smbNonBrowsable"] = undefined /*out*/;
            resourceInputs["snapshotDirectoryVisible"] = undefined /*out*/;
            resourceInputs["snapshotId"] = undefined /*out*/;
            resourceInputs["storageToNetworkProximity"] = undefined /*out*/;
            resourceInputs["subnetId"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["t2Network"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["throughputMibps"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["unixPermissions"] = undefined /*out*/;
            resourceInputs["usageThreshold"] = undefined /*out*/;
            resourceInputs["volumeGroupName"] = undefined /*out*/;
            resourceInputs["volumeSpecName"] = undefined /*out*/;
            resourceInputs["volumeType"] = undefined /*out*/;
            resourceInputs["zones"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:netapp/v20170815:CapacityPoolVolume" }, { type: "azure-native:netapp/v20190501:CapacityPoolVolume" }, { type: "azure-native:netapp/v20190601:CapacityPoolVolume" }, { type: "azure-native:netapp/v20190701:CapacityPoolVolume" }, { type: "azure-native:netapp/v20190801:CapacityPoolVolume" }, { type: "azure-native:netapp/v20191001:CapacityPoolVolume" }, { type: "azure-native:netapp/v20191101:CapacityPoolVolume" }, { type: "azure-native:netapp/v20200201:CapacityPoolVolume" }, { type: "azure-native:netapp/v20200301:CapacityPoolVolume" }, { type: "azure-native:netapp/v20200501:CapacityPoolVolume" }, { type: "azure-native:netapp/v20200601:CapacityPoolVolume" }, { type: "azure-native:netapp/v20200701:CapacityPoolVolume" }, { type: "azure-native:netapp/v20200801:CapacityPoolVolume" }, { type: "azure-native:netapp/v20200901:CapacityPoolVolume" }, { type: "azure-native:netapp/v20201101:CapacityPoolVolume" }, { type: "azure-native:netapp/v20201201:CapacityPoolVolume" }, { type: "azure-native:netapp/v20210201:CapacityPoolVolume" }, { type: "azure-native:netapp/v20210401:CapacityPoolVolume" }, { type: "azure-native:netapp/v20210401preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20210601:CapacityPoolVolume" }, { type: "azure-native:netapp/v20210801:CapacityPoolVolume" }, { type: "azure-native:netapp/v20211001:CapacityPoolVolume" }, { type: "azure-native:netapp/v20211001:Volume" }, { type: "azure-native:netapp/v20220101:CapacityPoolVolume" }, { type: "azure-native:netapp/v20220301:CapacityPoolVolume" }, { type: "azure-native:netapp/v20220501:CapacityPoolVolume" }, { type: "azure-native:netapp/v20220901:CapacityPoolVolume" }, { type: "azure-native:netapp/v20221101:CapacityPoolVolume" }, { type: "azure-native:netapp/v20221101:Volume" }, { type: "azure-native:netapp/v20221101preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20221101preview:Volume" }, { type: "azure-native:netapp/v20230501:CapacityPoolVolume" }, { type: "azure-native:netapp/v20230501:Volume" }, { type: "azure-native:netapp/v20230501preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20230501preview:Volume" }, { type: "azure-native:netapp/v20230701:CapacityPoolVolume" }, { type: "azure-native:netapp/v20230701:Volume" }, { type: "azure-native:netapp/v20230701preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20230701preview:Volume" }, { type: "azure-native:netapp/v20231101:CapacityPoolVolume" }, { type: "azure-native:netapp/v20231101:Volume" }, { type: "azure-native:netapp/v20231101preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20231101preview:Volume" }, { type: "azure-native:netapp/v20240101:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240101:Volume" }, { type: "azure-native:netapp/v20240301:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240301:Volume" }, { type: "azure-native:netapp/v20240301preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240301preview:Volume" }, { type: "azure-native:netapp/v20240501:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240501:Volume" }, { type: "azure-native:netapp/v20240501preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240501preview:Volume" }, { type: "azure-native:netapp/v20240701:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240701:Volume" }, { type: "azure-native:netapp/v20240701preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240701preview:Volume" }, { type: "azure-native:netapp/v20240901:CapacityPoolVolume" }, { type: "azure-native:netapp/v20240901:Volume" }, { type: "azure-native:netapp/v20240901preview:CapacityPoolVolume" }, { type: "azure-native:netapp/v20250101:CapacityPoolVolume" }, { type: "azure-native:netapp/v20250101preview:CapacityPoolVolume" }, { type: "azure-native:netapp:Volume" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CapacityPoolVolume.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CapacityPoolVolume = CapacityPoolVolume;
/** @internal */
CapacityPoolVolume.__pulumiType = 'azure-native:netapp:CapacityPoolVolume';
