// @pulumi/azure-native/netapp/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SnapshotPolicy = exports.listCapacityPoolVolumeReplicationsOutput = exports.listCapacityPoolVolumeReplications = exports.listCapacityPoolVolumeQuotaReportOutput = exports.listCapacityPoolVolumeQuotaReport = exports.getVolumeGroupOutput = exports.getVolumeGroup = exports.getSnapshotPolicyOutput = exports.getSnapshotPolicy = exports.getCapacityPoolVolumeQuotaRuleOutput = exports.getCapacityPoolVolumeQuotaRule = exports.getCapacityPoolVolumeGroupIdForLdapUserOutput = exports.getCapacityPoolVolumeGroupIdForLdapUser = exports.getCapacityPoolVolumeOutput = exports.getCapacityPoolVolume = exports.getCapacityPoolSubvolumeMetadataOutput = exports.getCapacityPoolSubvolumeMetadata = exports.getCapacityPoolSubvolumeOutput = exports.getCapacityPoolSubvolume = exports.getCapacityPoolSnapshotOutput = exports.getCapacityPoolSnapshot = exports.getCapacityPoolBucketOutput = exports.getCapacityPoolBucket = exports.getCapacityPoolBackupOutput = exports.getCapacityPoolBackup = exports.getCapacityPoolOutput = exports.getCapacityPool = exports.getBackupVaultOutput = exports.getBackupVault = exports.getBackupPolicyOutput = exports.getBackupPolicy = exports.getBackupOutput = exports.getBackup = exports.getAccountKeyVaultStatusOutput = exports.getAccountKeyVaultStatus = exports.getAccountChangeKeyVaultInformationOutput = exports.getAccountChangeKeyVaultInformation = exports.getAccountOutput = exports.getAccount = exports.CapacityPoolVolumeQuotaRule = exports.CapacityPoolVolume = exports.CapacityPoolSubvolume = exports.CapacityPoolSnapshot = exports.CapacityPoolBucket = exports.CapacityPoolBackup = exports.CapacityPool = exports.BackupVault = exports.BackupPolicy = exports.Backup = exports.Account = void 0;
exports.VolumeGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.Backup = null;
utilities.lazyLoad(exports, ["Backup"], () => require("./backup"));
exports.BackupPolicy = null;
utilities.lazyLoad(exports, ["BackupPolicy"], () => require("./backupPolicy"));
exports.BackupVault = null;
utilities.lazyLoad(exports, ["BackupVault"], () => require("./backupVault"));
exports.CapacityPool = null;
utilities.lazyLoad(exports, ["CapacityPool"], () => require("./capacityPool"));
exports.CapacityPoolBackup = null;
utilities.lazyLoad(exports, ["CapacityPoolBackup"], () => require("./capacityPoolBackup"));
exports.CapacityPoolBucket = null;
utilities.lazyLoad(exports, ["CapacityPoolBucket"], () => require("./capacityPoolBucket"));
exports.CapacityPoolSnapshot = null;
utilities.lazyLoad(exports, ["CapacityPoolSnapshot"], () => require("./capacityPoolSnapshot"));
exports.CapacityPoolSubvolume = null;
utilities.lazyLoad(exports, ["CapacityPoolSubvolume"], () => require("./capacityPoolSubvolume"));
exports.CapacityPoolVolume = null;
utilities.lazyLoad(exports, ["CapacityPoolVolume"], () => require("./capacityPoolVolume"));
exports.CapacityPoolVolumeQuotaRule = null;
utilities.lazyLoad(exports, ["CapacityPoolVolumeQuotaRule"], () => require("./capacityPoolVolumeQuotaRule"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getAccountChangeKeyVaultInformation = null;
exports.getAccountChangeKeyVaultInformationOutput = null;
utilities.lazyLoad(exports, ["getAccountChangeKeyVaultInformation", "getAccountChangeKeyVaultInformationOutput"], () => require("./getAccountChangeKeyVaultInformation"));
exports.getAccountKeyVaultStatus = null;
exports.getAccountKeyVaultStatusOutput = null;
utilities.lazyLoad(exports, ["getAccountKeyVaultStatus", "getAccountKeyVaultStatusOutput"], () => require("./getAccountKeyVaultStatus"));
exports.getBackup = null;
exports.getBackupOutput = null;
utilities.lazyLoad(exports, ["getBackup", "getBackupOutput"], () => require("./getBackup"));
exports.getBackupPolicy = null;
exports.getBackupPolicyOutput = null;
utilities.lazyLoad(exports, ["getBackupPolicy", "getBackupPolicyOutput"], () => require("./getBackupPolicy"));
exports.getBackupVault = null;
exports.getBackupVaultOutput = null;
utilities.lazyLoad(exports, ["getBackupVault", "getBackupVaultOutput"], () => require("./getBackupVault"));
exports.getCapacityPool = null;
exports.getCapacityPoolOutput = null;
utilities.lazyLoad(exports, ["getCapacityPool", "getCapacityPoolOutput"], () => require("./getCapacityPool"));
exports.getCapacityPoolBackup = null;
exports.getCapacityPoolBackupOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolBackup", "getCapacityPoolBackupOutput"], () => require("./getCapacityPoolBackup"));
exports.getCapacityPoolBucket = null;
exports.getCapacityPoolBucketOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolBucket", "getCapacityPoolBucketOutput"], () => require("./getCapacityPoolBucket"));
exports.getCapacityPoolSnapshot = null;
exports.getCapacityPoolSnapshotOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolSnapshot", "getCapacityPoolSnapshotOutput"], () => require("./getCapacityPoolSnapshot"));
exports.getCapacityPoolSubvolume = null;
exports.getCapacityPoolSubvolumeOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolSubvolume", "getCapacityPoolSubvolumeOutput"], () => require("./getCapacityPoolSubvolume"));
exports.getCapacityPoolSubvolumeMetadata = null;
exports.getCapacityPoolSubvolumeMetadataOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolSubvolumeMetadata", "getCapacityPoolSubvolumeMetadataOutput"], () => require("./getCapacityPoolSubvolumeMetadata"));
exports.getCapacityPoolVolume = null;
exports.getCapacityPoolVolumeOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolVolume", "getCapacityPoolVolumeOutput"], () => require("./getCapacityPoolVolume"));
exports.getCapacityPoolVolumeGroupIdForLdapUser = null;
exports.getCapacityPoolVolumeGroupIdForLdapUserOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolVolumeGroupIdForLdapUser", "getCapacityPoolVolumeGroupIdForLdapUserOutput"], () => require("./getCapacityPoolVolumeGroupIdForLdapUser"));
exports.getCapacityPoolVolumeQuotaRule = null;
exports.getCapacityPoolVolumeQuotaRuleOutput = null;
utilities.lazyLoad(exports, ["getCapacityPoolVolumeQuotaRule", "getCapacityPoolVolumeQuotaRuleOutput"], () => require("./getCapacityPoolVolumeQuotaRule"));
exports.getSnapshotPolicy = null;
exports.getSnapshotPolicyOutput = null;
utilities.lazyLoad(exports, ["getSnapshotPolicy", "getSnapshotPolicyOutput"], () => require("./getSnapshotPolicy"));
exports.getVolumeGroup = null;
exports.getVolumeGroupOutput = null;
utilities.lazyLoad(exports, ["getVolumeGroup", "getVolumeGroupOutput"], () => require("./getVolumeGroup"));
exports.listCapacityPoolVolumeQuotaReport = null;
exports.listCapacityPoolVolumeQuotaReportOutput = null;
utilities.lazyLoad(exports, ["listCapacityPoolVolumeQuotaReport", "listCapacityPoolVolumeQuotaReportOutput"], () => require("./listCapacityPoolVolumeQuotaReport"));
exports.listCapacityPoolVolumeReplications = null;
exports.listCapacityPoolVolumeReplicationsOutput = null;
utilities.lazyLoad(exports, ["listCapacityPoolVolumeReplications", "listCapacityPoolVolumeReplicationsOutput"], () => require("./listCapacityPoolVolumeReplications"));
exports.SnapshotPolicy = null;
utilities.lazyLoad(exports, ["SnapshotPolicy"], () => require("./snapshotPolicy"));
exports.VolumeGroup = null;
utilities.lazyLoad(exports, ["VolumeGroup"], () => require("./volumeGroup"));
// Export enums:
__exportStar(require("../types/enums/netapp"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:netapp:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:netapp:Backup":
                return new exports.Backup(name, undefined, { urn });
            case "azure-native:netapp:BackupPolicy":
                return new exports.BackupPolicy(name, undefined, { urn });
            case "azure-native:netapp:BackupVault":
                return new exports.BackupVault(name, undefined, { urn });
            case "azure-native:netapp:CapacityPool":
                return new exports.CapacityPool(name, undefined, { urn });
            case "azure-native:netapp:CapacityPoolBackup":
                return new exports.CapacityPoolBackup(name, undefined, { urn });
            case "azure-native:netapp:CapacityPoolBucket":
                return new exports.CapacityPoolBucket(name, undefined, { urn });
            case "azure-native:netapp:CapacityPoolSnapshot":
                return new exports.CapacityPoolSnapshot(name, undefined, { urn });
            case "azure-native:netapp:CapacityPoolSubvolume":
                return new exports.CapacityPoolSubvolume(name, undefined, { urn });
            case "azure-native:netapp:CapacityPoolVolume":
                return new exports.CapacityPoolVolume(name, undefined, { urn });
            case "azure-native:netapp:CapacityPoolVolumeQuotaRule":
                return new exports.CapacityPoolVolumeQuotaRule(name, undefined, { urn });
            case "azure-native:netapp:SnapshotPolicy":
                return new exports.SnapshotPolicy(name, undefined, { urn });
            case "azure-native:netapp:VolumeGroup":
                return new exports.VolumeGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "netapp", _module);
