// @pulumi/azure-native/testbase/draftPackage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DraftPackage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Test Base Draft Package resource.
 *
 * Uses Azure REST API version 2023-11-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-11-01-preview.
 */
class DraftPackage extends pulumi.CustomResource {
    /**
     * Get an existing DraftPackage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DraftPackage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DraftPackage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DraftPackage.__pulumiType;
    }
    /**
     * Create a DraftPackage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.testBaseAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'testBaseAccountName'");
            }
            resourceInputs["appFileName"] = args ? args.appFileName : undefined;
            resourceInputs["applicationName"] = args ? args.applicationName : undefined;
            resourceInputs["comments"] = args ? args.comments : undefined;
            resourceInputs["draftPackageName"] = args ? args.draftPackageName : undefined;
            resourceInputs["editPackage"] = (_a = (args ? args.editPackage : undefined)) !== null && _a !== void 0 ? _a : false;
            resourceInputs["executableLaunchCommand"] = args ? args.executableLaunchCommand : undefined;
            resourceInputs["firstPartyApps"] = args ? args.firstPartyApps : undefined;
            resourceInputs["flightingRing"] = args ? args.flightingRing : undefined;
            resourceInputs["galleryApps"] = args ? args.galleryApps : undefined;
            resourceInputs["highlightedFiles"] = args ? args.highlightedFiles : undefined;
            resourceInputs["inplaceUpgradeOSPair"] = args ? args.inplaceUpgradeOSPair : undefined;
            resourceInputs["intuneEnrollmentMetadata"] = args ? args.intuneEnrollmentMetadata : undefined;
            resourceInputs["intuneMetadata"] = args ? args.intuneMetadata : undefined;
            resourceInputs["packageId"] = args ? args.packageId : undefined;
            resourceInputs["packageTags"] = args ? args.packageTags : undefined;
            resourceInputs["processName"] = args ? args.processName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sourceType"] = (_b = (args ? args.sourceType : undefined)) !== null && _b !== void 0 ? _b : "Native";
            resourceInputs["tabState"] = args ? args.tabState : undefined;
            resourceInputs["targetOSList"] = args ? args.targetOSList : undefined;
            resourceInputs["testBaseAccountName"] = args ? args.testBaseAccountName : undefined;
            resourceInputs["testTypes"] = args ? args.testTypes : undefined;
            resourceInputs["tests"] = args ? args.tests : undefined;
            resourceInputs["useAutofill"] = args ? args.useAutofill : undefined;
            resourceInputs["useSample"] = args ? args.useSample : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["draftPackagePath"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["workingPath"] = undefined /*out*/;
        }
        else {
            resourceInputs["appFileName"] = undefined /*out*/;
            resourceInputs["applicationName"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["comments"] = undefined /*out*/;
            resourceInputs["draftPackagePath"] = undefined /*out*/;
            resourceInputs["editPackage"] = undefined /*out*/;
            resourceInputs["executableLaunchCommand"] = undefined /*out*/;
            resourceInputs["firstPartyApps"] = undefined /*out*/;
            resourceInputs["flightingRing"] = undefined /*out*/;
            resourceInputs["galleryApps"] = undefined /*out*/;
            resourceInputs["highlightedFiles"] = undefined /*out*/;
            resourceInputs["inplaceUpgradeOSPair"] = undefined /*out*/;
            resourceInputs["intuneEnrollmentMetadata"] = undefined /*out*/;
            resourceInputs["intuneMetadata"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["packageId"] = undefined /*out*/;
            resourceInputs["packageTags"] = undefined /*out*/;
            resourceInputs["processName"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sourceType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tabState"] = undefined /*out*/;
            resourceInputs["targetOSList"] = undefined /*out*/;
            resourceInputs["testTypes"] = undefined /*out*/;
            resourceInputs["tests"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["useAutofill"] = undefined /*out*/;
            resourceInputs["useSample"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
            resourceInputs["workingPath"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:testbase/v20231101preview:DraftPackage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DraftPackage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DraftPackage = DraftPackage;
/** @internal */
DraftPackage.__pulumiType = 'azure-native:testbase:DraftPackage';
