// @pulumi/azure-native/azurestackhci/deploymentSetting.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Edge device resource
 *
 * Uses Azure REST API version 2024-04-01. In version 2.x of the Azure Native provider, it used API version 2023-08-01-preview.
 *
 * Other available API versions: 2023-08-01-preview, 2023-11-01-preview, 2024-01-01, 2024-02-15-preview, 2024-09-01-preview, 2024-12-01-preview, 2025-02-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native azurestackhci [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class DeploymentSetting extends pulumi.CustomResource {
    /**
     * Get an existing DeploymentSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeploymentSetting(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeploymentSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeploymentSetting.__pulumiType;
    }
    /**
     * Create a DeploymentSetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.arcNodeResourceIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'arcNodeResourceIds'");
            }
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.deploymentConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deploymentConfiguration'");
            }
            if ((!args || args.deploymentMode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deploymentMode'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["arcNodeResourceIds"] = args ? args.arcNodeResourceIds : undefined;
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["deploymentConfiguration"] = args ? args.deploymentConfiguration : undefined;
            resourceInputs["deploymentMode"] = args ? args.deploymentMode : undefined;
            resourceInputs["deploymentSettingsName"] = args ? args.deploymentSettingsName : undefined;
            resourceInputs["operationType"] = (_a = (args ? args.operationType : undefined)) !== null && _a !== void 0 ? _a : "ClusterProvisioning";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["reportedProperties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["arcNodeResourceIds"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["deploymentConfiguration"] = undefined /*out*/;
            resourceInputs["deploymentMode"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["operationType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["reportedProperties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20230801preview:DeploymentSetting" }, { type: "azure-native:azurestackhci/v20231101preview:DeploymentSetting" }, { type: "azure-native:azurestackhci/v20240101:DeploymentSetting" }, { type: "azure-native:azurestackhci/v20240215preview:DeploymentSetting" }, { type: "azure-native:azurestackhci/v20240401:DeploymentSetting" }, { type: "azure-native:azurestackhci/v20240901preview:DeploymentSetting" }, { type: "azure-native:azurestackhci/v20241201preview:DeploymentSetting" }, { type: "azure-native:azurestackhci/v20250201preview:DeploymentSetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DeploymentSetting.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DeploymentSetting = DeploymentSetting;
/** @internal */
DeploymentSetting.__pulumiType = 'azure-native:azurestackhci:DeploymentSetting';
