// @pulumi/azure-native/azurestackhci/marketplaceGalleryImage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MarketplaceGalleryImage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The marketplace gallery image resource definition.
 *
 * Uses Azure REST API version 2025-02-01-preview. In version 2.x of the Azure Native provider, it used API version 2022-12-15-preview.
 *
 * Other available API versions: 2022-12-15-preview, 2023-07-01-preview, 2023-09-01-preview, 2024-01-01, 2024-02-01-preview, 2024-05-01-preview, 2024-07-15-preview, 2024-08-01-preview, 2024-10-01-preview, 2025-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native azurestackhci [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class MarketplaceGalleryImage extends pulumi.CustomResource {
    /**
     * Get an existing MarketplaceGalleryImage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MarketplaceGalleryImage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MarketplaceGalleryImage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MarketplaceGalleryImage.__pulumiType;
    }
    /**
     * Create a MarketplaceGalleryImage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.osType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'osType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["cloudInitDataSource"] = args ? args.cloudInitDataSource : undefined;
            resourceInputs["containerId"] = args ? args.containerId : undefined;
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["hyperVGeneration"] = args ? args.hyperVGeneration : undefined;
            resourceInputs["identifier"] = args ? args.identifier : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["marketplaceGalleryImageName"] = args ? args.marketplaceGalleryImageName : undefined;
            resourceInputs["osType"] = args ? args.osType : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["cloudInitDataSource"] = undefined /*out*/;
            resourceInputs["containerId"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["hyperVGeneration"] = undefined /*out*/;
            resourceInputs["identifier"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestackhci/v20210901preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20210901preview:Marketplacegalleryimage" }, { type: "azure-native:azurestackhci/v20221215preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20230701preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20230901preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20240101:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20240201preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20240501preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20240715preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20240801preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20241001preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20250201preview:MarketplaceGalleryImage" }, { type: "azure-native:azurestackhci/v20250401preview:MarketplaceGalleryImage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MarketplaceGalleryImage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MarketplaceGalleryImage = MarketplaceGalleryImage;
/** @internal */
MarketplaceGalleryImage.__pulumiType = 'azure-native:azurestackhci:MarketplaceGalleryImage';
