// @pulumi/azure-native/machinelearningservices/componentContainer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComponentContainer = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2022-02-01-preview, 2022-05-01, 2022-06-01-preview, 2022-10-01, 2022-10-01-preview, 2022-12-01-preview, 2023-02-01-preview, 2023-04-01, 2023-04-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-07-01-preview, 2024-10-01-preview, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native machinelearningservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ComponentContainer extends pulumi.CustomResource {
    /**
     * Get an existing ComponentContainer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ComponentContainer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ComponentContainer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ComponentContainer.__pulumiType;
    }
    /**
     * Create a ComponentContainer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.componentContainerProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'componentContainerProperties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["componentContainerProperties"] = args ? (args.componentContainerProperties ? pulumi.output(args.componentContainerProperties).apply(inputs.machinelearningservices.componentContainerArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["componentContainerProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20220201preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20220501:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20220601preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20221001:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20221001preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20221201preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20230201preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20230401:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20230401preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20230601preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20230801preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20231001:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20240101preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20240401:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20240401preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20240701preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20241001:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20241001preview:ComponentContainer" }, { type: "azure-native:machinelearningservices/v20250101preview:ComponentContainer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ComponentContainer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ComponentContainer = ComponentContainer;
/** @internal */
ComponentContainer.__pulumiType = 'azure-native:machinelearningservices:ComponentContainer';
