// @pulumi/azure-native/machinelearningservices/onlineEndpoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnlineEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2021-03-01-preview, 2022-02-01-preview, 2022-05-01, 2022-06-01-preview, 2022-10-01, 2022-10-01-preview, 2022-12-01-preview, 2023-02-01-preview, 2023-04-01, 2023-04-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-07-01-preview, 2024-10-01-preview, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native machinelearningservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class OnlineEndpoint extends pulumi.CustomResource {
    /**
     * Get an existing OnlineEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OnlineEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OnlineEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OnlineEndpoint.__pulumiType;
    }
    /**
     * Create a OnlineEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.onlineEndpointProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'onlineEndpointProperties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["endpointName"] = args ? args.endpointName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["onlineEndpointProperties"] = args ? (args.onlineEndpointProperties ? pulumi.output(args.onlineEndpointProperties).apply(inputs.machinelearningservices.onlineEndpointArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["onlineEndpointProperties"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20210301preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20220201preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20220501:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20220601preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20221001:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20221001preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20221201preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20230201preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20230401:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20230401preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20230601preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20230801preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20231001:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20240101preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20240401:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20240401preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20240701preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20241001:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20241001preview:OnlineEndpoint" }, { type: "azure-native:machinelearningservices/v20250101preview:OnlineEndpoint" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(OnlineEndpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OnlineEndpoint = OnlineEndpoint;
/** @internal */
OnlineEndpoint.__pulumiType = 'azure-native:machinelearningservices:OnlineEndpoint';
