// @pulumi/azure-native/machinelearningservices/registryCodeVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistryCodeVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2022-10-01-preview, 2022-12-01-preview, 2023-02-01-preview, 2023-04-01, 2023-04-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-07-01-preview, 2024-10-01-preview, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native machinelearningservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class RegistryCodeVersion extends pulumi.CustomResource {
    /**
     * Get an existing RegistryCodeVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RegistryCodeVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegistryCodeVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegistryCodeVersion.__pulumiType;
    }
    /**
     * Create a RegistryCodeVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.codeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'codeName'");
            }
            if ((!args || args.codeVersionProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'codeVersionProperties'");
            }
            if ((!args || args.registryName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'registryName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["codeName"] = args ? args.codeName : undefined;
            resourceInputs["codeVersionProperties"] = args ? (args.codeVersionProperties ? pulumi.output(args.codeVersionProperties).apply(inputs.machinelearningservices.codeVersionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["registryName"] = args ? args.registryName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["codeVersionProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20221001preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20221201preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20230201preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20230401:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20230401preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20230601preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20230801preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20231001:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20240101preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20240401:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20240401preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20240701preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20241001:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20241001preview:RegistryCodeVersion" }, { type: "azure-native:machinelearningservices/v20250101preview:RegistryCodeVersion" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RegistryCodeVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegistryCodeVersion = RegistryCodeVersion;
/** @internal */
RegistryCodeVersion.__pulumiType = 'azure-native:machinelearningservices:RegistryCodeVersion';
