// @pulumi/azure-native/machinelearningservices/registryModelContainer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistryModelContainer = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * Azure Resource Manager resource envelope.
 *
 * Uses Azure REST API version 2024-10-01. In version 2.x of the Azure Native provider, it used API version 2023-04-01.
 *
 * Other available API versions: 2022-10-01-preview, 2022-12-01-preview, 2023-02-01-preview, 2023-04-01, 2023-04-01-preview, 2023-06-01-preview, 2023-08-01-preview, 2023-10-01, 2024-01-01-preview, 2024-04-01, 2024-07-01-preview, 2024-10-01-preview, 2025-01-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native machinelearningservices [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class RegistryModelContainer extends pulumi.CustomResource {
    /**
     * Get an existing RegistryModelContainer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RegistryModelContainer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegistryModelContainer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegistryModelContainer.__pulumiType;
    }
    /**
     * Create a RegistryModelContainer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.modelContainerProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelContainerProperties'");
            }
            if ((!args || args.registryName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'registryName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["modelContainerProperties"] = args ? (args.modelContainerProperties ? pulumi.output(args.modelContainerProperties).apply(inputs.machinelearningservices.modelContainerArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["modelName"] = args ? args.modelName : undefined;
            resourceInputs["registryName"] = args ? args.registryName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["modelContainerProperties"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices/v20221001preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20221201preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20230201preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20230401:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20230401preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20230601preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20230801preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20231001:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20240101preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20240401:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20240401preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20240701preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20241001:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20241001preview:RegistryModelContainer" }, { type: "azure-native:machinelearningservices/v20250101preview:RegistryModelContainer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RegistryModelContainer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegistryModelContainer = RegistryModelContainer;
/** @internal */
RegistryModelContainer.__pulumiType = 'azure-native:machinelearningservices:RegistryModelContainer';
