// @pulumi/azure-native/network/expressRouteCircuitConnection.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressRouteCircuitConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Express Route Circuit Connection in an ExpressRouteCircuitPeering resource.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2018-06-01, 2018-07-01, 2018-08-01, 2018-10-01, 2018-11-01, 2018-12-01, 2019-02-01, 2019-04-01, 2019-06-01, 2019-07-01, 2019-08-01, 2019-09-01, 2019-11-01, 2019-12-01, 2020-03-01, 2020-04-01, 2020-05-01, 2020-06-01, 2020-07-01, 2020-08-01, 2020-11-01, 2021-02-01, 2021-03-01, 2021-05-01, 2021-08-01, 2022-01-01, 2022-05-01, 2022-07-01, 2022-09-01, 2022-11-01, 2023-02-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-07-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native network [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ExpressRouteCircuitConnection extends pulumi.CustomResource {
    /**
     * Get an existing ExpressRouteCircuitConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExpressRouteCircuitConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExpressRouteCircuitConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExpressRouteCircuitConnection.__pulumiType;
    }
    /**
     * Create a ExpressRouteCircuitConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.circuitName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'circuitName'");
            }
            if ((!args || args.peeringName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peeringName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["addressPrefix"] = args ? args.addressPrefix : undefined;
            resourceInputs["authorizationKey"] = args ? args.authorizationKey : undefined;
            resourceInputs["circuitName"] = args ? args.circuitName : undefined;
            resourceInputs["connectionName"] = args ? args.connectionName : undefined;
            resourceInputs["expressRouteCircuitPeering"] = args ? args.expressRouteCircuitPeering : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["ipv6CircuitConnectionConfig"] = args ? args.ipv6CircuitConnectionConfig : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["peerExpressRouteCircuitPeering"] = args ? args.peerExpressRouteCircuitPeering : undefined;
            resourceInputs["peeringName"] = args ? args.peeringName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["circuitConnectionStatus"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["addressPrefix"] = undefined /*out*/;
            resourceInputs["authorizationKey"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["circuitConnectionStatus"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["expressRouteCircuitPeering"] = undefined /*out*/;
            resourceInputs["ipv6CircuitConnectionConfig"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["peerExpressRouteCircuitPeering"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20180201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180401:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180601:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180701:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180801:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20181001:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20181101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20181201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190401:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190601:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190701:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190801:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190901:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20191101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20191201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200301:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200401:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200501:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200601:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200701:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200801:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20201101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20210201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20210301:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20210501:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20210801:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20220101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20220501:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20220701:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20220901:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20221101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20230201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20230401:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20230501:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20230601:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20230901:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20231101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20240101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20240301:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20240501:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20240701:ExpressRouteCircuitConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExpressRouteCircuitConnection.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExpressRouteCircuitConnection = ExpressRouteCircuitConnection;
/** @internal */
ExpressRouteCircuitConnection.__pulumiType = 'azure-native:network:ExpressRouteCircuitConnection';
