// @pulumi/azure-native/network/securityAdminConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityAdminConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Defines the security admin configuration
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2021-02-01-preview, 2022-01-01, 2022-02-01-preview, 2022-04-01-preview, 2022-05-01, 2022-07-01, 2022-09-01, 2022-11-01, 2023-02-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-01-01-preview, 2024-03-01, 2024-07-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native network [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SecurityAdminConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing SecurityAdminConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SecurityAdminConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecurityAdminConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityAdminConfiguration.__pulumiType;
    }
    /**
     * Create a SecurityAdminConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.networkManagerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkManagerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["applyOnNetworkIntentPolicyBasedServices"] = args ? args.applyOnNetworkIntentPolicyBasedServices : undefined;
            resourceInputs["configurationName"] = args ? args.configurationName : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["networkGroupAddressSpaceAggregationOption"] = args ? args.networkGroupAddressSpaceAggregationOption : undefined;
            resourceInputs["networkManagerName"] = args ? args.networkManagerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["applyOnNetworkIntentPolicyBasedServices"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkGroupAddressSpaceAggregationOption"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20210201preview:SecurityAdminConfiguration" }, { type: "azure-native:network/v20210501preview:SecurityAdminConfiguration" }, { type: "azure-native:network/v20220101:SecurityAdminConfiguration" }, { type: "azure-native:network/v20220201preview:SecurityAdminConfiguration" }, { type: "azure-native:network/v20220401preview:SecurityAdminConfiguration" }, { type: "azure-native:network/v20220501:SecurityAdminConfiguration" }, { type: "azure-native:network/v20220701:SecurityAdminConfiguration" }, { type: "azure-native:network/v20220901:SecurityAdminConfiguration" }, { type: "azure-native:network/v20221101:SecurityAdminConfiguration" }, { type: "azure-native:network/v20230201:SecurityAdminConfiguration" }, { type: "azure-native:network/v20230401:SecurityAdminConfiguration" }, { type: "azure-native:network/v20230501:SecurityAdminConfiguration" }, { type: "azure-native:network/v20230601:SecurityAdminConfiguration" }, { type: "azure-native:network/v20230901:SecurityAdminConfiguration" }, { type: "azure-native:network/v20231101:SecurityAdminConfiguration" }, { type: "azure-native:network/v20240101:SecurityAdminConfiguration" }, { type: "azure-native:network/v20240101preview:SecurityAdminConfiguration" }, { type: "azure-native:network/v20240301:SecurityAdminConfiguration" }, { type: "azure-native:network/v20240501:SecurityAdminConfiguration" }, { type: "azure-native:network/v20240701:SecurityAdminConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SecurityAdminConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecurityAdminConfiguration = SecurityAdminConfiguration;
/** @internal */
SecurityAdminConfiguration.__pulumiType = 'azure-native:network:SecurityAdminConfiguration';
