// @pulumi/azure-native/network/virtualHubRouteTableV2.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualHubRouteTableV2 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * VirtualHubRouteTableV2 Resource.
 *
 * Uses Azure REST API version 2024-05-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2019-09-01, 2019-11-01, 2019-12-01, 2020-03-01, 2020-04-01, 2020-05-01, 2020-06-01, 2020-07-01, 2020-08-01, 2020-11-01, 2021-02-01, 2021-03-01, 2021-05-01, 2021-08-01, 2022-01-01, 2022-05-01, 2022-07-01, 2022-09-01, 2022-11-01, 2023-02-01, 2023-04-01, 2023-05-01, 2023-06-01, 2023-09-01, 2023-11-01, 2024-01-01, 2024-03-01, 2024-07-01. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native network [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class VirtualHubRouteTableV2 extends pulumi.CustomResource {
    /**
     * Get an existing VirtualHubRouteTableV2 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualHubRouteTableV2(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualHubRouteTableV2.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualHubRouteTableV2.__pulumiType;
    }
    /**
     * Create a VirtualHubRouteTableV2 resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.virtualHubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'virtualHubName'");
            }
            resourceInputs["attachedConnections"] = args ? args.attachedConnections : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["routeTableName"] = args ? args.routeTableName : undefined;
            resourceInputs["routes"] = args ? args.routes : undefined;
            resourceInputs["virtualHubName"] = args ? args.virtualHubName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
        }
        else {
            resourceInputs["attachedConnections"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["routes"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network/v20190901:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20191101:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20191201:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20200301:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20200401:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20200501:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20200601:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20200701:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20200801:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20201101:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20210201:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20210301:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20210501:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20210801:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20220101:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20220501:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20220701:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20220901:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20221101:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20230201:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20230401:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20230501:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20230601:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20230901:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20231101:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20240101:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20240301:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20240501:VirtualHubRouteTableV2" }, { type: "azure-native:network/v20240701:VirtualHubRouteTableV2" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualHubRouteTableV2.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VirtualHubRouteTableV2 = VirtualHubRouteTableV2;
/** @internal */
VirtualHubRouteTableV2.__pulumiType = 'azure-native:network:VirtualHubRouteTableV2';
