// @pulumi/azure-native/applicationinsights/componentLinkedStorageAccount.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComponentLinkedStorageAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Application Insights component linked storage accounts
 *
 * Uses Azure REST API version 2020-03-01-preview.
 */
class ComponentLinkedStorageAccount extends pulumi.CustomResource {
    /**
     * Get an existing ComponentLinkedStorageAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ComponentLinkedStorageAccount(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ComponentLinkedStorageAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ComponentLinkedStorageAccount.__pulumiType;
    }
    /**
     * Create a ComponentLinkedStorageAccount resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["linkedStorageAccount"] = args ? args.linkedStorageAccount : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["storageType"] = args ? args.storageType : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["linkedStorageAccount"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:applicationinsights/v20200301preview:ComponentLinkedStorageAccount" }, { type: "azure-native:insights/v20200301preview:ComponentLinkedStorageAccount" }, { type: "azure-native:insights:ComponentLinkedStorageAccount" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ComponentLinkedStorageAccount.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ComponentLinkedStorageAccount = ComponentLinkedStorageAccount;
/** @internal */
ComponentLinkedStorageAccount.__pulumiType = 'azure-native:applicationinsights:ComponentLinkedStorageAccount';
