// @pulumi/azure-native/applicationinsights/exportConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Properties that define a Continuous Export configuration.
 *
 * Uses Azure REST API version 2015-05-01.
 */
class ExportConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing ExportConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExportConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExportConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExportConfiguration.__pulumiType;
    }
    /**
     * Create a ExportConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["destinationAccountId"] = args ? args.destinationAccountId : undefined;
            resourceInputs["destinationAddress"] = args ? args.destinationAddress : undefined;
            resourceInputs["destinationStorageLocationId"] = args ? args.destinationStorageLocationId : undefined;
            resourceInputs["destinationStorageSubscriptionId"] = args ? args.destinationStorageSubscriptionId : undefined;
            resourceInputs["destinationType"] = args ? args.destinationType : undefined;
            resourceInputs["exportId"] = args ? args.exportId : undefined;
            resourceInputs["isEnabled"] = args ? args.isEnabled : undefined;
            resourceInputs["notificationQueueEnabled"] = args ? args.notificationQueueEnabled : undefined;
            resourceInputs["notificationQueueUri"] = args ? args.notificationQueueUri : undefined;
            resourceInputs["recordTypes"] = args ? args.recordTypes : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["applicationName"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["containerName"] = undefined /*out*/;
            resourceInputs["exportStatus"] = undefined /*out*/;
            resourceInputs["instrumentationKey"] = undefined /*out*/;
            resourceInputs["isUserEnabled"] = undefined /*out*/;
            resourceInputs["lastGapTime"] = undefined /*out*/;
            resourceInputs["lastSuccessTime"] = undefined /*out*/;
            resourceInputs["lastUserUpdate"] = undefined /*out*/;
            resourceInputs["permanentErrorReason"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["storageName"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationName"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["containerName"] = undefined /*out*/;
            resourceInputs["destinationAccountId"] = undefined /*out*/;
            resourceInputs["destinationStorageLocationId"] = undefined /*out*/;
            resourceInputs["destinationStorageSubscriptionId"] = undefined /*out*/;
            resourceInputs["destinationType"] = undefined /*out*/;
            resourceInputs["exportId"] = undefined /*out*/;
            resourceInputs["exportStatus"] = undefined /*out*/;
            resourceInputs["instrumentationKey"] = undefined /*out*/;
            resourceInputs["isUserEnabled"] = undefined /*out*/;
            resourceInputs["lastGapTime"] = undefined /*out*/;
            resourceInputs["lastSuccessTime"] = undefined /*out*/;
            resourceInputs["lastUserUpdate"] = undefined /*out*/;
            resourceInputs["notificationQueueEnabled"] = undefined /*out*/;
            resourceInputs["permanentErrorReason"] = undefined /*out*/;
            resourceInputs["recordTypes"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["storageName"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:applicationinsights/v20150501:ExportConfiguration" }, { type: "azure-native:insights/v20150501:ExportConfiguration" }, { type: "azure-native:insights:ExportConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExportConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ExportConfiguration = ExportConfiguration;
/** @internal */
ExportConfiguration.__pulumiType = 'azure-native:applicationinsights:ExportConfiguration';
