// @pulumi/azure-native/applicationinsights/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkbookTemplate = exports.Workbook = exports.WebTest = exports.ProactiveDetectionConfiguration = exports.MyWorkbook = exports.listEASubscriptionListMigrationDatePostOutput = exports.listEASubscriptionListMigrationDatePost = exports.getWorkbookTemplateOutput = exports.getWorkbookTemplate = exports.getWorkbookOutput = exports.getWorkbook = exports.getWebTestOutput = exports.getWebTest = exports.getTestResultFileOutput = exports.getTestResultFile = exports.getProactiveDetectionConfigurationOutput = exports.getProactiveDetectionConfiguration = exports.getMyWorkbookOutput = exports.getMyWorkbook = exports.getLiveTokenOutput = exports.getLiveToken = exports.getFavoriteOutput = exports.getFavorite = exports.getExportConfigurationOutput = exports.getExportConfiguration = exports.getComponentLinkedStorageAccountOutput = exports.getComponentLinkedStorageAccount = exports.getComponentCurrentBillingFeatureOutput = exports.getComponentCurrentBillingFeature = exports.getComponentOutput = exports.getComponent = exports.getAnalyticsItemOutput = exports.getAnalyticsItem = exports.Favorite = exports.ExportConfiguration = exports.ComponentLinkedStorageAccount = exports.ComponentCurrentBillingFeature = exports.Component = exports.AnalyticsItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AnalyticsItem = null;
utilities.lazyLoad(exports, ["AnalyticsItem"], () => require("./analyticsItem"));
exports.Component = null;
utilities.lazyLoad(exports, ["Component"], () => require("./component"));
exports.ComponentCurrentBillingFeature = null;
utilities.lazyLoad(exports, ["ComponentCurrentBillingFeature"], () => require("./componentCurrentBillingFeature"));
exports.ComponentLinkedStorageAccount = null;
utilities.lazyLoad(exports, ["ComponentLinkedStorageAccount"], () => require("./componentLinkedStorageAccount"));
exports.ExportConfiguration = null;
utilities.lazyLoad(exports, ["ExportConfiguration"], () => require("./exportConfiguration"));
exports.Favorite = null;
utilities.lazyLoad(exports, ["Favorite"], () => require("./favorite"));
exports.getAnalyticsItem = null;
exports.getAnalyticsItemOutput = null;
utilities.lazyLoad(exports, ["getAnalyticsItem", "getAnalyticsItemOutput"], () => require("./getAnalyticsItem"));
exports.getComponent = null;
exports.getComponentOutput = null;
utilities.lazyLoad(exports, ["getComponent", "getComponentOutput"], () => require("./getComponent"));
exports.getComponentCurrentBillingFeature = null;
exports.getComponentCurrentBillingFeatureOutput = null;
utilities.lazyLoad(exports, ["getComponentCurrentBillingFeature", "getComponentCurrentBillingFeatureOutput"], () => require("./getComponentCurrentBillingFeature"));
exports.getComponentLinkedStorageAccount = null;
exports.getComponentLinkedStorageAccountOutput = null;
utilities.lazyLoad(exports, ["getComponentLinkedStorageAccount", "getComponentLinkedStorageAccountOutput"], () => require("./getComponentLinkedStorageAccount"));
exports.getExportConfiguration = null;
exports.getExportConfigurationOutput = null;
utilities.lazyLoad(exports, ["getExportConfiguration", "getExportConfigurationOutput"], () => require("./getExportConfiguration"));
exports.getFavorite = null;
exports.getFavoriteOutput = null;
utilities.lazyLoad(exports, ["getFavorite", "getFavoriteOutput"], () => require("./getFavorite"));
exports.getLiveToken = null;
exports.getLiveTokenOutput = null;
utilities.lazyLoad(exports, ["getLiveToken", "getLiveTokenOutput"], () => require("./getLiveToken"));
exports.getMyWorkbook = null;
exports.getMyWorkbookOutput = null;
utilities.lazyLoad(exports, ["getMyWorkbook", "getMyWorkbookOutput"], () => require("./getMyWorkbook"));
exports.getProactiveDetectionConfiguration = null;
exports.getProactiveDetectionConfigurationOutput = null;
utilities.lazyLoad(exports, ["getProactiveDetectionConfiguration", "getProactiveDetectionConfigurationOutput"], () => require("./getProactiveDetectionConfiguration"));
exports.getTestResultFile = null;
exports.getTestResultFileOutput = null;
utilities.lazyLoad(exports, ["getTestResultFile", "getTestResultFileOutput"], () => require("./getTestResultFile"));
exports.getWebTest = null;
exports.getWebTestOutput = null;
utilities.lazyLoad(exports, ["getWebTest", "getWebTestOutput"], () => require("./getWebTest"));
exports.getWorkbook = null;
exports.getWorkbookOutput = null;
utilities.lazyLoad(exports, ["getWorkbook", "getWorkbookOutput"], () => require("./getWorkbook"));
exports.getWorkbookTemplate = null;
exports.getWorkbookTemplateOutput = null;
utilities.lazyLoad(exports, ["getWorkbookTemplate", "getWorkbookTemplateOutput"], () => require("./getWorkbookTemplate"));
exports.listEASubscriptionListMigrationDatePost = null;
exports.listEASubscriptionListMigrationDatePostOutput = null;
utilities.lazyLoad(exports, ["listEASubscriptionListMigrationDatePost", "listEASubscriptionListMigrationDatePostOutput"], () => require("./listEASubscriptionListMigrationDatePost"));
exports.MyWorkbook = null;
utilities.lazyLoad(exports, ["MyWorkbook"], () => require("./myWorkbook"));
exports.ProactiveDetectionConfiguration = null;
utilities.lazyLoad(exports, ["ProactiveDetectionConfiguration"], () => require("./proactiveDetectionConfiguration"));
exports.WebTest = null;
utilities.lazyLoad(exports, ["WebTest"], () => require("./webTest"));
exports.Workbook = null;
utilities.lazyLoad(exports, ["Workbook"], () => require("./workbook"));
exports.WorkbookTemplate = null;
utilities.lazyLoad(exports, ["WorkbookTemplate"], () => require("./workbookTemplate"));
// Export enums:
__exportStar(require("../types/enums/applicationinsights"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:applicationinsights:AnalyticsItem":
                return new exports.AnalyticsItem(name, undefined, { urn });
            case "azure-native:applicationinsights:Component":
                return new exports.Component(name, undefined, { urn });
            case "azure-native:applicationinsights:ComponentCurrentBillingFeature":
                return new exports.ComponentCurrentBillingFeature(name, undefined, { urn });
            case "azure-native:applicationinsights:ComponentLinkedStorageAccount":
                return new exports.ComponentLinkedStorageAccount(name, undefined, { urn });
            case "azure-native:applicationinsights:ExportConfiguration":
                return new exports.ExportConfiguration(name, undefined, { urn });
            case "azure-native:applicationinsights:Favorite":
                return new exports.Favorite(name, undefined, { urn });
            case "azure-native:applicationinsights:MyWorkbook":
                return new exports.MyWorkbook(name, undefined, { urn });
            case "azure-native:applicationinsights:ProactiveDetectionConfiguration":
                return new exports.ProactiveDetectionConfiguration(name, undefined, { urn });
            case "azure-native:applicationinsights:WebTest":
                return new exports.WebTest(name, undefined, { urn });
            case "azure-native:applicationinsights:Workbook":
                return new exports.Workbook(name, undefined, { urn });
            case "azure-native:applicationinsights:WorkbookTemplate":
                return new exports.WorkbookTemplate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "applicationinsights", _module);
