// @pulumi/azure-native/securityinsights/anomalies.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Anomalies = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Settings with single toggle.
 *
 * Uses Azure REST API version 2025-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-06-01-preview.
 */
class Anomalies extends pulumi.CustomResource {
    /**
     * Get an existing Anomalies resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Anomalies(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Anomalies.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Anomalies.__pulumiType;
    }
    /**
     * Create a Anomalies resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["kind"] = "Anomalies";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["settingsName"] = args ? args.settingsName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["isEnabled"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["isEnabled"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:Anomalies" }, { type: "azure-native:securityinsights/v20190101preview:IPSyncer" }, { type: "azure-native:securityinsights/v20210301preview:Anomalies" }, { type: "azure-native:securityinsights/v20210301preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20210301preview:EyesOn" }, { type: "azure-native:securityinsights/v20210301preview:Ueba" }, { type: "azure-native:securityinsights/v20210901preview:Anomalies" }, { type: "azure-native:securityinsights/v20211001preview:Anomalies" }, { type: "azure-native:securityinsights/v20220101preview:Anomalies" }, { type: "azure-native:securityinsights/v20220101preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20220401preview:Anomalies" }, { type: "azure-native:securityinsights/v20220501preview:Anomalies" }, { type: "azure-native:securityinsights/v20220601preview:Anomalies" }, { type: "azure-native:securityinsights/v20220701preview:Anomalies" }, { type: "azure-native:securityinsights/v20220801preview:Anomalies" }, { type: "azure-native:securityinsights/v20220901preview:Anomalies" }, { type: "azure-native:securityinsights/v20221001preview:Anomalies" }, { type: "azure-native:securityinsights/v20221101preview:Anomalies" }, { type: "azure-native:securityinsights/v20221201preview:Anomalies" }, { type: "azure-native:securityinsights/v20230201preview:Anomalies" }, { type: "azure-native:securityinsights/v20230301preview:Anomalies" }, { type: "azure-native:securityinsights/v20230401preview:Anomalies" }, { type: "azure-native:securityinsights/v20230501preview:Anomalies" }, { type: "azure-native:securityinsights/v20230601preview:Anomalies" }, { type: "azure-native:securityinsights/v20230601preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20230601preview:EyesOn" }, { type: "azure-native:securityinsights/v20230601preview:Ueba" }, { type: "azure-native:securityinsights/v20230701preview:Anomalies" }, { type: "azure-native:securityinsights/v20230701preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20230701preview:EyesOn" }, { type: "azure-native:securityinsights/v20230701preview:Ueba" }, { type: "azure-native:securityinsights/v20230801preview:Anomalies" }, { type: "azure-native:securityinsights/v20230801preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20230801preview:EyesOn" }, { type: "azure-native:securityinsights/v20230801preview:Ueba" }, { type: "azure-native:securityinsights/v20230901preview:Anomalies" }, { type: "azure-native:securityinsights/v20230901preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20230901preview:EyesOn" }, { type: "azure-native:securityinsights/v20230901preview:Ueba" }, { type: "azure-native:securityinsights/v20231001preview:Anomalies" }, { type: "azure-native:securityinsights/v20231001preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20231001preview:EyesOn" }, { type: "azure-native:securityinsights/v20231001preview:Ueba" }, { type: "azure-native:securityinsights/v20231201preview:Anomalies" }, { type: "azure-native:securityinsights/v20231201preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20231201preview:EyesOn" }, { type: "azure-native:securityinsights/v20231201preview:Ueba" }, { type: "azure-native:securityinsights/v20240101preview:Anomalies" }, { type: "azure-native:securityinsights/v20240101preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20240101preview:EyesOn" }, { type: "azure-native:securityinsights/v20240101preview:Ueba" }, { type: "azure-native:securityinsights/v20240401preview:Anomalies" }, { type: "azure-native:securityinsights/v20240401preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20240401preview:EyesOn" }, { type: "azure-native:securityinsights/v20240401preview:Ueba" }, { type: "azure-native:securityinsights/v20241001preview:Anomalies" }, { type: "azure-native:securityinsights/v20241001preview:EntityAnalytics" }, { type: "azure-native:securityinsights/v20241001preview:EyesOn" }, { type: "azure-native:securityinsights/v20241001preview:Ueba" }, { type: "azure-native:securityinsights/v20250101preview:Anomalies" }, { type: "azure-native:securityinsights/v20250401preview:Anomalies" }, { type: "azure-native:securityinsights:EntityAnalytics" }, { type: "azure-native:securityinsights:EyesOn" }, { type: "azure-native:securityinsights:Ueba" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Anomalies.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Anomalies = Anomalies;
/** @internal */
Anomalies.__pulumiType = 'azure-native:securityinsights:Anomalies';
