// @pulumi/azure-native/securityinsights/contentPackage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentPackage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a Package in Azure Security Insights.
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-06-01-preview.
 *
 * Other available API versions: 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-11-01, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class ContentPackage extends pulumi.CustomResource {
    /**
     * Get an existing ContentPackage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ContentPackage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ContentPackage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ContentPackage.__pulumiType;
    }
    /**
     * Create a ContentPackage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.contentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contentId'");
            }
            if ((!args || args.contentKind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contentKind'");
            }
            if ((!args || args.contentProductId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contentProductId'");
            }
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.version === undefined) && !opts.urn) {
                throw new Error("Missing required property 'version'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["author"] = args ? args.author : undefined;
            resourceInputs["categories"] = args ? args.categories : undefined;
            resourceInputs["contentId"] = args ? args.contentId : undefined;
            resourceInputs["contentKind"] = args ? args.contentKind : undefined;
            resourceInputs["contentProductId"] = args ? args.contentProductId : undefined;
            resourceInputs["contentSchemaVersion"] = args ? args.contentSchemaVersion : undefined;
            resourceInputs["dependencies"] = args ? args.dependencies : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["firstPublishDate"] = args ? args.firstPublishDate : undefined;
            resourceInputs["icon"] = args ? args.icon : undefined;
            resourceInputs["isDeprecated"] = args ? args.isDeprecated : undefined;
            resourceInputs["isFeatured"] = args ? args.isFeatured : undefined;
            resourceInputs["isNew"] = args ? args.isNew : undefined;
            resourceInputs["isPreview"] = args ? args.isPreview : undefined;
            resourceInputs["lastPublishDate"] = args ? args.lastPublishDate : undefined;
            resourceInputs["packageId"] = args ? args.packageId : undefined;
            resourceInputs["providers"] = args ? args.providers : undefined;
            resourceInputs["publisherDisplayName"] = args ? args.publisherDisplayName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["source"] = args ? args.source : undefined;
            resourceInputs["support"] = args ? args.support : undefined;
            resourceInputs["threatAnalysisTactics"] = args ? args.threatAnalysisTactics : undefined;
            resourceInputs["threatAnalysisTechniques"] = args ? args.threatAnalysisTechniques : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["author"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["categories"] = undefined /*out*/;
            resourceInputs["contentId"] = undefined /*out*/;
            resourceInputs["contentKind"] = undefined /*out*/;
            resourceInputs["contentProductId"] = undefined /*out*/;
            resourceInputs["contentSchemaVersion"] = undefined /*out*/;
            resourceInputs["dependencies"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["firstPublishDate"] = undefined /*out*/;
            resourceInputs["icon"] = undefined /*out*/;
            resourceInputs["isDeprecated"] = undefined /*out*/;
            resourceInputs["isFeatured"] = undefined /*out*/;
            resourceInputs["isNew"] = undefined /*out*/;
            resourceInputs["isPreview"] = undefined /*out*/;
            resourceInputs["lastPublishDate"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["providers"] = undefined /*out*/;
            resourceInputs["publisherDisplayName"] = undefined /*out*/;
            resourceInputs["source"] = undefined /*out*/;
            resourceInputs["support"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["threatAnalysisTactics"] = undefined /*out*/;
            resourceInputs["threatAnalysisTechniques"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20230401preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230501preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230601preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230701preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230801preview:ContentPackage" }, { type: "azure-native:securityinsights/v20230901preview:ContentPackage" }, { type: "azure-native:securityinsights/v20231001preview:ContentPackage" }, { type: "azure-native:securityinsights/v20231101:ContentPackage" }, { type: "azure-native:securityinsights/v20231201preview:ContentPackage" }, { type: "azure-native:securityinsights/v20240101preview:ContentPackage" }, { type: "azure-native:securityinsights/v20240301:ContentPackage" }, { type: "azure-native:securityinsights/v20240401preview:ContentPackage" }, { type: "azure-native:securityinsights/v20240901:ContentPackage" }, { type: "azure-native:securityinsights/v20241001preview:ContentPackage" }, { type: "azure-native:securityinsights/v20250101preview:ContentPackage" }, { type: "azure-native:securityinsights/v20250301:ContentPackage" }, { type: "azure-native:securityinsights/v20250401preview:ContentPackage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ContentPackage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ContentPackage = ContentPackage;
/** @internal */
ContentPackage.__pulumiType = 'azure-native:securityinsights:ContentPackage';
