// @pulumi/azure-native/securityinsights/customizableConnectorDefinition.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomizableConnectorDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Connector definition for kind 'Customizable'.
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-07-01-preview.
 */
class CustomizableConnectorDefinition extends pulumi.CustomResource {
    /**
     * Get an existing CustomizableConnectorDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CustomizableConnectorDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomizableConnectorDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomizableConnectorDefinition.__pulumiType;
    }
    /**
     * Create a CustomizableConnectorDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.connectorUiConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectorUiConfig'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["connectionsConfig"] = args ? args.connectionsConfig : undefined;
            resourceInputs["connectorUiConfig"] = args ? args.connectorUiConfig : undefined;
            resourceInputs["createdTimeUtc"] = args ? args.createdTimeUtc : undefined;
            resourceInputs["dataConnectorDefinitionName"] = args ? args.dataConnectorDefinitionName : undefined;
            resourceInputs["kind"] = "Customizable";
            resourceInputs["lastModifiedUtc"] = args ? args.lastModifiedUtc : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["connectionsConfig"] = undefined /*out*/;
            resourceInputs["connectorUiConfig"] = undefined /*out*/;
            resourceInputs["createdTimeUtc"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastModifiedUtc"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20230701preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20230801preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20230901preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20231001preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20231201preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20240101preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20240401preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20240901:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20241001preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20250101preview:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20250301:CustomizableConnectorDefinition" }, { type: "azure-native:securityinsights/v20250401preview:CustomizableConnectorDefinition" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CustomizableConnectorDefinition.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CustomizableConnectorDefinition = CustomizableConnectorDefinition;
/** @internal */
CustomizableConnectorDefinition.__pulumiType = 'azure-native:securityinsights:CustomizableConnectorDefinition';
