// @pulumi/azure-native/securityinsights/huntComment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HuntComment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a Hunt Comment in Azure Security Insights
 *
 * Uses Azure REST API version 2025-01-01-preview. In version 2.x of the Azure Native provider, it used API version 2023-06-01-preview.
 *
 * Other available API versions: 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-12-01-preview, 2024-01-01-preview, 2024-04-01-preview, 2024-10-01-preview, 2025-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class HuntComment extends pulumi.CustomResource {
    /**
     * Get an existing HuntComment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new HuntComment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HuntComment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HuntComment.__pulumiType;
    }
    /**
     * Create a HuntComment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.huntId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'huntId'");
            }
            if ((!args || args.message === undefined) && !opts.urn) {
                throw new Error("Missing required property 'message'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["huntCommentId"] = args ? args.huntCommentId : undefined;
            resourceInputs["huntId"] = args ? args.huntId : undefined;
            resourceInputs["message"] = args ? args.message : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["message"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20230401preview:HuntComment" }, { type: "azure-native:securityinsights/v20230501preview:HuntComment" }, { type: "azure-native:securityinsights/v20230601preview:HuntComment" }, { type: "azure-native:securityinsights/v20230701preview:HuntComment" }, { type: "azure-native:securityinsights/v20230801preview:HuntComment" }, { type: "azure-native:securityinsights/v20230901preview:HuntComment" }, { type: "azure-native:securityinsights/v20231001preview:HuntComment" }, { type: "azure-native:securityinsights/v20231201preview:HuntComment" }, { type: "azure-native:securityinsights/v20240101preview:HuntComment" }, { type: "azure-native:securityinsights/v20240401preview:HuntComment" }, { type: "azure-native:securityinsights/v20241001preview:HuntComment" }, { type: "azure-native:securityinsights/v20250101preview:HuntComment" }, { type: "azure-native:securityinsights/v20250401preview:HuntComment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(HuntComment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HuntComment = HuntComment;
/** @internal */
HuntComment.__pulumiType = 'azure-native:securityinsights:HuntComment';
