// @pulumi/azure-native/securityinsights/metadata.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Metadata = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Metadata resource definition.
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-11-01, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Metadata extends pulumi.CustomResource {
    /**
     * Get an existing Metadata resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Metadata(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Metadata.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Metadata.__pulumiType;
    }
    /**
     * Create a Metadata resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.parentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'parentId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["author"] = args ? args.author : undefined;
            resourceInputs["categories"] = args ? args.categories : undefined;
            resourceInputs["contentId"] = args ? args.contentId : undefined;
            resourceInputs["contentSchemaVersion"] = args ? args.contentSchemaVersion : undefined;
            resourceInputs["customVersion"] = args ? args.customVersion : undefined;
            resourceInputs["dependencies"] = args ? args.dependencies : undefined;
            resourceInputs["firstPublishDate"] = args ? args.firstPublishDate : undefined;
            resourceInputs["icon"] = args ? args.icon : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["lastPublishDate"] = args ? args.lastPublishDate : undefined;
            resourceInputs["metadataName"] = args ? args.metadataName : undefined;
            resourceInputs["parentId"] = args ? args.parentId : undefined;
            resourceInputs["previewImages"] = args ? args.previewImages : undefined;
            resourceInputs["previewImagesDark"] = args ? args.previewImagesDark : undefined;
            resourceInputs["providers"] = args ? args.providers : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["source"] = args ? args.source : undefined;
            resourceInputs["support"] = args ? args.support : undefined;
            resourceInputs["threatAnalysisTactics"] = args ? args.threatAnalysisTactics : undefined;
            resourceInputs["threatAnalysisTechniques"] = args ? args.threatAnalysisTechniques : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["author"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["categories"] = undefined /*out*/;
            resourceInputs["contentId"] = undefined /*out*/;
            resourceInputs["contentSchemaVersion"] = undefined /*out*/;
            resourceInputs["customVersion"] = undefined /*out*/;
            resourceInputs["dependencies"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["firstPublishDate"] = undefined /*out*/;
            resourceInputs["icon"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastPublishDate"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["parentId"] = undefined /*out*/;
            resourceInputs["previewImages"] = undefined /*out*/;
            resourceInputs["previewImagesDark"] = undefined /*out*/;
            resourceInputs["providers"] = undefined /*out*/;
            resourceInputs["source"] = undefined /*out*/;
            resourceInputs["support"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["threatAnalysisTactics"] = undefined /*out*/;
            resourceInputs["threatAnalysisTechniques"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20210301preview:Metadata" }, { type: "azure-native:securityinsights/v20210901preview:Metadata" }, { type: "azure-native:securityinsights/v20211001preview:Metadata" }, { type: "azure-native:securityinsights/v20220101preview:Metadata" }, { type: "azure-native:securityinsights/v20220401preview:Metadata" }, { type: "azure-native:securityinsights/v20220501preview:Metadata" }, { type: "azure-native:securityinsights/v20220601preview:Metadata" }, { type: "azure-native:securityinsights/v20220701preview:Metadata" }, { type: "azure-native:securityinsights/v20220801preview:Metadata" }, { type: "azure-native:securityinsights/v20220901preview:Metadata" }, { type: "azure-native:securityinsights/v20221001preview:Metadata" }, { type: "azure-native:securityinsights/v20221101preview:Metadata" }, { type: "azure-native:securityinsights/v20221201preview:Metadata" }, { type: "azure-native:securityinsights/v20230201:Metadata" }, { type: "azure-native:securityinsights/v20230201preview:Metadata" }, { type: "azure-native:securityinsights/v20230301preview:Metadata" }, { type: "azure-native:securityinsights/v20230401preview:Metadata" }, { type: "azure-native:securityinsights/v20230501preview:Metadata" }, { type: "azure-native:securityinsights/v20230601preview:Metadata" }, { type: "azure-native:securityinsights/v20230701preview:Metadata" }, { type: "azure-native:securityinsights/v20230801preview:Metadata" }, { type: "azure-native:securityinsights/v20230901preview:Metadata" }, { type: "azure-native:securityinsights/v20231001preview:Metadata" }, { type: "azure-native:securityinsights/v20231101:Metadata" }, { type: "azure-native:securityinsights/v20231201preview:Metadata" }, { type: "azure-native:securityinsights/v20240101preview:Metadata" }, { type: "azure-native:securityinsights/v20240301:Metadata" }, { type: "azure-native:securityinsights/v20240401preview:Metadata" }, { type: "azure-native:securityinsights/v20240901:Metadata" }, { type: "azure-native:securityinsights/v20241001preview:Metadata" }, { type: "azure-native:securityinsights/v20250101preview:Metadata" }, { type: "azure-native:securityinsights/v20250301:Metadata" }, { type: "azure-native:securityinsights/v20250401preview:Metadata" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Metadata.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Metadata = Metadata;
/** @internal */
Metadata.__pulumiType = 'azure-native:securityinsights:Metadata';
