// @pulumi/azure-native/securityinsights/sentinelOnboardingState.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SentinelOnboardingState = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Sentinel onboarding state
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-11-01, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class SentinelOnboardingState extends pulumi.CustomResource {
    /**
     * Get an existing SentinelOnboardingState resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SentinelOnboardingState(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SentinelOnboardingState.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SentinelOnboardingState.__pulumiType;
    }
    /**
     * Create a SentinelOnboardingState resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["customerManagedKey"] = args ? args.customerManagedKey : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sentinelOnboardingStateName"] = args ? args.sentinelOnboardingStateName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["customerManagedKey"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20210301preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20210901preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20211001:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20211001preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220101preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220401preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220501preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220601preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220701preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220801:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220801preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20220901preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20221001preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20221101:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20221101preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20221201preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230201:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230201preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230301preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230401preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230501preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230601preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230701preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230801preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20230901preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20231001preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20231101:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20231201preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20240101preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20240301:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20240401preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20240901:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20241001preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20250101preview:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20250301:SentinelOnboardingState" }, { type: "azure-native:securityinsights/v20250401preview:SentinelOnboardingState" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SentinelOnboardingState.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SentinelOnboardingState = SentinelOnboardingState;
/** @internal */
SentinelOnboardingState.__pulumiType = 'azure-native:securityinsights:SentinelOnboardingState';
