// @pulumi/azure-native/securityinsights/watchlist.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a Watchlist in Azure Security Insights.
 *
 * Uses Azure REST API version 2024-09-01. In version 2.x of the Azure Native provider, it used API version 2023-02-01.
 *
 * Other available API versions: 2023-02-01, 2023-03-01-preview, 2023-04-01-preview, 2023-05-01-preview, 2023-06-01-preview, 2023-07-01-preview, 2023-08-01-preview, 2023-09-01-preview, 2023-10-01-preview, 2023-11-01, 2023-12-01-preview, 2024-01-01-preview, 2024-03-01, 2024-04-01-preview, 2024-10-01-preview, 2025-01-01-preview, 2025-03-01, 2025-04-01-preview. These can be accessed by generating a local SDK package using the CLI command `pulumi package add azure-native securityinsights [ApiVersion]`. See the [version guide](../../../version-guide/#accessing-any-api-version-via-local-packages) for details.
 */
class Watchlist extends pulumi.CustomResource {
    /**
     * Get an existing Watchlist resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Watchlist(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Watchlist.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Watchlist.__pulumiType;
    }
    /**
     * Create a Watchlist resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.itemsSearchKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'itemsSearchKey'");
            }
            if ((!args || args.provider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'provider'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["contentType"] = args ? args.contentType : undefined;
            resourceInputs["created"] = args ? args.created : undefined;
            resourceInputs["createdBy"] = args ? args.createdBy : undefined;
            resourceInputs["defaultDuration"] = args ? args.defaultDuration : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["isDeleted"] = args ? args.isDeleted : undefined;
            resourceInputs["itemsSearchKey"] = args ? args.itemsSearchKey : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["numberOfLinesToSkip"] = args ? args.numberOfLinesToSkip : undefined;
            resourceInputs["provider"] = args ? args.provider : undefined;
            resourceInputs["rawContent"] = args ? args.rawContent : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["source"] = args ? args.source : undefined;
            resourceInputs["sourceType"] = args ? args.sourceType : undefined;
            resourceInputs["tenantId"] = args ? args.tenantId : undefined;
            resourceInputs["updated"] = args ? args.updated : undefined;
            resourceInputs["updatedBy"] = args ? args.updatedBy : undefined;
            resourceInputs["uploadStatus"] = args ? args.uploadStatus : undefined;
            resourceInputs["watchlistAlias"] = args ? args.watchlistAlias : undefined;
            resourceInputs["watchlistId"] = args ? args.watchlistId : undefined;
            resourceInputs["watchlistType"] = args ? args.watchlistType : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["contentType"] = undefined /*out*/;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["defaultDuration"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["isDeleted"] = undefined /*out*/;
            resourceInputs["itemsSearchKey"] = undefined /*out*/;
            resourceInputs["labels"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfLinesToSkip"] = undefined /*out*/;
            resourceInputs["provider"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["rawContent"] = undefined /*out*/;
            resourceInputs["source"] = undefined /*out*/;
            resourceInputs["sourceType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updated"] = undefined /*out*/;
            resourceInputs["updatedBy"] = undefined /*out*/;
            resourceInputs["uploadStatus"] = undefined /*out*/;
            resourceInputs["watchlistAlias"] = undefined /*out*/;
            resourceInputs["watchlistId"] = undefined /*out*/;
            resourceInputs["watchlistType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:Watchlist" }, { type: "azure-native:securityinsights/v20210301preview:Watchlist" }, { type: "azure-native:securityinsights/v20210401:Watchlist" }, { type: "azure-native:securityinsights/v20210901preview:Watchlist" }, { type: "azure-native:securityinsights/v20211001:Watchlist" }, { type: "azure-native:securityinsights/v20211001preview:Watchlist" }, { type: "azure-native:securityinsights/v20220101preview:Watchlist" }, { type: "azure-native:securityinsights/v20220401preview:Watchlist" }, { type: "azure-native:securityinsights/v20220501preview:Watchlist" }, { type: "azure-native:securityinsights/v20220601preview:Watchlist" }, { type: "azure-native:securityinsights/v20220701preview:Watchlist" }, { type: "azure-native:securityinsights/v20220801:Watchlist" }, { type: "azure-native:securityinsights/v20220801preview:Watchlist" }, { type: "azure-native:securityinsights/v20220901preview:Watchlist" }, { type: "azure-native:securityinsights/v20221001preview:Watchlist" }, { type: "azure-native:securityinsights/v20221101:Watchlist" }, { type: "azure-native:securityinsights/v20221101preview:Watchlist" }, { type: "azure-native:securityinsights/v20221201preview:Watchlist" }, { type: "azure-native:securityinsights/v20230201:Watchlist" }, { type: "azure-native:securityinsights/v20230201preview:Watchlist" }, { type: "azure-native:securityinsights/v20230301preview:Watchlist" }, { type: "azure-native:securityinsights/v20230401preview:Watchlist" }, { type: "azure-native:securityinsights/v20230501preview:Watchlist" }, { type: "azure-native:securityinsights/v20230601preview:Watchlist" }, { type: "azure-native:securityinsights/v20230701preview:Watchlist" }, { type: "azure-native:securityinsights/v20230801preview:Watchlist" }, { type: "azure-native:securityinsights/v20230901preview:Watchlist" }, { type: "azure-native:securityinsights/v20231001preview:Watchlist" }, { type: "azure-native:securityinsights/v20231101:Watchlist" }, { type: "azure-native:securityinsights/v20231201preview:Watchlist" }, { type: "azure-native:securityinsights/v20240101preview:Watchlist" }, { type: "azure-native:securityinsights/v20240301:Watchlist" }, { type: "azure-native:securityinsights/v20240401preview:Watchlist" }, { type: "azure-native:securityinsights/v20240901:Watchlist" }, { type: "azure-native:securityinsights/v20241001preview:Watchlist" }, { type: "azure-native:securityinsights/v20250101preview:Watchlist" }, { type: "azure-native:securityinsights/v20250301:Watchlist" }, { type: "azure-native:securityinsights/v20250401preview:Watchlist" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Watchlist.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Watchlist = Watchlist;
/** @internal */
Watchlist.__pulumiType = 'azure-native:securityinsights:Watchlist';
