// @pulumi/azure-native/types/enums/web/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WsdlImportMethod = exports.UpgradePreference = exports.UnauthenticatedClientActionV2 = exports.UnauthenticatedClientAction = exports.TlsCipherSuites = exports.SupportedTlsVersions = exports.StorageType = exports.StagingEnvironmentPolicy = exports.SslState = exports.SiteLoadBalancing = exports.ScmType = exports.RuntimeName = exports.RouteType = exports.RedundancyMode = exports.PublicCertificateLocation = exports.ManagedServiceIdentityType = exports.ManagedPipelineMode = exports.LogLevel = exports.LoadBalancingMode = exports.IpFilterTag = exports.IPMode = exports.HostType = exports.HostNameType = exports.FunctionsDeploymentStorageType = exports.FtpsState = exports.FrontEndServiceType = exports.FrequencyUnit = exports.ForwardProxyConvention = exports.EnterpriseGradeCdnStatus = exports.DefaultAction = exports.DatabaseType = exports.DaprLogLevel = exports.CustomHostNameDnsRecordType = exports.CookieExpirationConvention = exports.ConnectionStringType = exports.ConnectionParameterType = exports.ClientCredentialMethod = exports.ClientCertMode = exports.BuiltInAuthenticationProvider = exports.AzureStorageType = exports.AzureStorageProtocol = exports.AzureResourceType = exports.AutoHealActionType = exports.AutoGeneratedDomainNameLabelScope = exports.AuthenticationType = exports.AuthType = exports.ApiType = void 0;
exports.ApiType = {
    NotSpecified: "NotSpecified",
    Rest: "Rest",
    Soap: "Soap",
};
exports.AuthType = {
    Anonymous: "Anonymous",
    UserCredentials: "UserCredentials",
    SystemIdentity: "SystemIdentity",
    UserAssigned: "UserAssigned",
};
exports.AuthenticationType = {
    SystemAssignedIdentity: "SystemAssignedIdentity",
    UserAssignedIdentity: "UserAssignedIdentity",
    StorageAccountConnectionString: "StorageAccountConnectionString",
};
exports.AutoGeneratedDomainNameLabelScope = {
    TenantReuse: "TenantReuse",
    SubscriptionReuse: "SubscriptionReuse",
    ResourceGroupReuse: "ResourceGroupReuse",
    NoReuse: "NoReuse",
};
exports.AutoHealActionType = {
    Recycle: "Recycle",
    LogEvent: "LogEvent",
    CustomAction: "CustomAction",
};
exports.AzureResourceType = {
    Website: "Website",
    TrafficManager: "TrafficManager",
};
exports.AzureStorageProtocol = {
    Smb: "Smb",
    Http: "Http",
    Nfs: "Nfs",
};
exports.AzureStorageType = {
    AzureFiles: "AzureFiles",
    AzureBlob: "AzureBlob",
};
exports.BuiltInAuthenticationProvider = {
    AzureActiveDirectory: "AzureActiveDirectory",
    Facebook: "Facebook",
    Google: "Google",
    MicrosoftAccount: "MicrosoftAccount",
    Twitter: "Twitter",
    Github: "Github",
};
exports.ClientCertMode = {
    Required: "Required",
    Optional: "Optional",
    OptionalInteractiveUser: "OptionalInteractiveUser",
};
exports.ClientCredentialMethod = {
    ClientSecretPost: "ClientSecretPost",
};
exports.ConnectionParameterType = {
    String: "string",
    Securestring: "securestring",
    Secureobject: "secureobject",
    Int: "int",
    Bool: "bool",
    Object: "object",
    Array: "array",
    OauthSetting: "oauthSetting",
    Connection: "connection",
};
exports.ConnectionStringType = {
    MySql: "MySql",
    SQLServer: "SQLServer",
    SQLAzure: "SQLAzure",
    Custom: "Custom",
    NotificationHub: "NotificationHub",
    ServiceBus: "ServiceBus",
    EventHub: "EventHub",
    ApiHub: "ApiHub",
    DocDb: "DocDb",
    RedisCache: "RedisCache",
    PostgreSQL: "PostgreSQL",
};
exports.CookieExpirationConvention = {
    FixedTime: "FixedTime",
    IdentityProviderDerived: "IdentityProviderDerived",
};
exports.CustomHostNameDnsRecordType = {
    CName: "CName",
    A: "A",
};
exports.DaprLogLevel = {
    Info: "info",
    Debug: "debug",
    Warn: "warn",
    Error: "error",
};
exports.DatabaseType = {
    SqlAzure: "SqlAzure",
    MySql: "MySql",
    LocalMySql: "LocalMySql",
    PostgreSql: "PostgreSql",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EnterpriseGradeCdnStatus = {
    Enabled: "Enabled",
    Enabling: "Enabling",
    Disabled: "Disabled",
    Disabling: "Disabling",
};
exports.ForwardProxyConvention = {
    NoProxy: "NoProxy",
    Standard: "Standard",
    Custom: "Custom",
};
exports.FrequencyUnit = {
    Day: "Day",
    Hour: "Hour",
};
exports.FrontEndServiceType = {
    NodePort: "NodePort",
    LoadBalancer: "LoadBalancer",
};
exports.FtpsState = {
    AllAllowed: "AllAllowed",
    FtpsOnly: "FtpsOnly",
    Disabled: "Disabled",
};
exports.FunctionsDeploymentStorageType = {
    BlobContainer: "blobContainer",
};
exports.HostNameType = {
    Verified: "Verified",
    Managed: "Managed",
};
exports.HostType = {
    Standard: "Standard",
    Repository: "Repository",
};
exports.IPMode = {
    IPv4: "IPv4",
    IPv6: "IPv6",
    IPv4AndIPv6: "IPv4AndIPv6",
};
exports.IpFilterTag = {
    Default: "Default",
    XffProxy: "XffProxy",
    ServiceTag: "ServiceTag",
};
exports.LoadBalancingMode = {
    None: "None",
    Web: "Web",
    Publishing: "Publishing",
    Web_Publishing: "Web, Publishing",
};
exports.LogLevel = {
    Off: "Off",
    Verbose: "Verbose",
    Information: "Information",
    Warning: "Warning",
    Error: "Error",
};
exports.ManagedPipelineMode = {
    Integrated: "Integrated",
    Classic: "Classic",
};
exports.ManagedServiceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.PublicCertificateLocation = {
    CurrentUserMy: "CurrentUserMy",
    LocalMachineMy: "LocalMachineMy",
    Unknown: "Unknown",
};
exports.RedundancyMode = {
    None: "None",
    Manual: "Manual",
    Failover: "Failover",
    ActiveActive: "ActiveActive",
    GeoRedundant: "GeoRedundant",
};
exports.RouteType = {
    DEFAULT: "DEFAULT",
    INHERITED: "INHERITED",
    STATIC: "STATIC",
};
exports.RuntimeName = {
    Dotnet_isolated: "dotnet-isolated",
    Node: "node",
    Java: "java",
    Powershell: "powershell",
    Python: "python",
    Custom: "custom",
};
exports.ScmType = {
    None: "None",
    Dropbox: "Dropbox",
    Tfs: "Tfs",
    LocalGit: "LocalGit",
    GitHub: "GitHub",
    CodePlexGit: "CodePlexGit",
    CodePlexHg: "CodePlexHg",
    BitbucketGit: "BitbucketGit",
    BitbucketHg: "BitbucketHg",
    ExternalGit: "ExternalGit",
    ExternalHg: "ExternalHg",
    OneDrive: "OneDrive",
    VSO: "VSO",
    VSTSRM: "VSTSRM",
};
exports.SiteLoadBalancing = {
    WeightedRoundRobin: "WeightedRoundRobin",
    LeastRequests: "LeastRequests",
    LeastResponseTime: "LeastResponseTime",
    WeightedTotalTraffic: "WeightedTotalTraffic",
    RequestHash: "RequestHash",
    PerSiteRoundRobin: "PerSiteRoundRobin",
    LeastRequestsWithTieBreaker: "LeastRequestsWithTieBreaker",
};
exports.SslState = {
    Disabled: "Disabled",
    SniEnabled: "SniEnabled",
    IpBasedEnabled: "IpBasedEnabled",
};
exports.StagingEnvironmentPolicy = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageType = {
    LocalNode: "LocalNode",
    NetworkFileSystem: "NetworkFileSystem",
};
exports.SupportedTlsVersions = {
    SupportedTlsVersions_1_0: "1.0",
    SupportedTlsVersions_1_1: "1.1",
    SupportedTlsVersions_1_2: "1.2",
    SupportedTlsVersions_1_3: "1.3",
};
exports.TlsCipherSuites = {
    TLS_AES_256_GCM_SHA384: "TLS_AES_256_GCM_SHA384",
    TLS_AES_128_GCM_SHA256: "TLS_AES_128_GCM_SHA256",
    TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384: "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
    TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256: "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
    TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256: "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
    TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
    TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256: "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
    TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384: "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384",
    TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256: "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
    TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA: "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
    TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA: "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
    TLS_RSA_WITH_AES_256_GCM_SHA384: "TLS_RSA_WITH_AES_256_GCM_SHA384",
    TLS_RSA_WITH_AES_128_GCM_SHA256: "TLS_RSA_WITH_AES_128_GCM_SHA256",
    TLS_RSA_WITH_AES_256_CBC_SHA256: "TLS_RSA_WITH_AES_256_CBC_SHA256",
    TLS_RSA_WITH_AES_128_CBC_SHA256: "TLS_RSA_WITH_AES_128_CBC_SHA256",
    TLS_RSA_WITH_AES_256_CBC_SHA: "TLS_RSA_WITH_AES_256_CBC_SHA",
    TLS_RSA_WITH_AES_128_CBC_SHA: "TLS_RSA_WITH_AES_128_CBC_SHA",
};
exports.UnauthenticatedClientAction = {
    RedirectToLoginPage: "RedirectToLoginPage",
    AllowAnonymous: "AllowAnonymous",
};
exports.UnauthenticatedClientActionV2 = {
    RedirectToLoginPage: "RedirectToLoginPage",
    AllowAnonymous: "AllowAnonymous",
    Return401: "Return401",
    Return403: "Return403",
};
exports.UpgradePreference = {
    /**
     * No preference on when this App Service Environment will be upgraded
     */
    None: "None",
    /**
     * This App Service Environment will be upgraded before others in the same region that have Upgrade Preference 'Late'
     */
    Early: "Early",
    /**
     * This App Service Environment will be upgraded after others in the same region that have Upgrade Preference 'Early'
     */
    Late: "Late",
    /**
     * ASEv3 only. Once an upgrade is available, this App Service Environment will wait 10 days for the upgrade to be manually initiated. After 10 days the upgrade will begin automatically
     */
    Manual: "Manual",
};
exports.WsdlImportMethod = {
    NotSpecified: "NotSpecified",
    SoapToRest: "SoapToRest",
    SoapPassThrough: "SoapPassThrough",
};
