// @pulumi/azure-native/secretsynccontroller/secretSync.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecretSync = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The SecretSync resource.
 *
 * Uses Azure REST API version 2024-08-21-preview. In version 2.x of the Azure Native provider, it used API version 2024-08-21-preview.
 */
class SecretSync extends pulumi.CustomResource {
    /**
     * Get an existing SecretSync resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SecretSync(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecretSync.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecretSync.__pulumiType;
    }
    /**
     * Create a SecretSync resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kubernetesSecretType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kubernetesSecretType'");
            }
            if ((!args || args.objectSecretMapping === undefined) && !opts.urn) {
                throw new Error("Missing required property 'objectSecretMapping'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.secretProviderClassName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'secretProviderClassName'");
            }
            if ((!args || args.serviceAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceAccountName'");
            }
            resourceInputs["extendedLocation"] = args ? args.extendedLocation : undefined;
            resourceInputs["forceSynchronization"] = args ? args.forceSynchronization : undefined;
            resourceInputs["kubernetesSecretType"] = args ? args.kubernetesSecretType : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["objectSecretMapping"] = args ? args.objectSecretMapping : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["secretProviderClassName"] = args ? args.secretProviderClassName : undefined;
            resourceInputs["secretSyncName"] = args ? args.secretSyncName : undefined;
            resourceInputs["serviceAccountName"] = args ? args.serviceAccountName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["extendedLocation"] = undefined /*out*/;
            resourceInputs["forceSynchronization"] = undefined /*out*/;
            resourceInputs["kubernetesSecretType"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["objectSecretMapping"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["secretProviderClassName"] = undefined /*out*/;
            resourceInputs["serviceAccountName"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:secretsynccontroller/v20240821preview:SecretSync" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SecretSync.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SecretSync = SecretSync;
/** @internal */
SecretSync.__pulumiType = 'azure-native:secretsynccontroller:SecretSync';
