// @pulumi/azure-native/dnsresolver/dnsSecurityRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DnsSecurityRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes a DNS security rule.
 *
 * Uses Azure REST API version 2023-07-01-preview.
 */
class DnsSecurityRule extends pulumi.CustomResource {
    /**
     * Get an existing DnsSecurityRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DnsSecurityRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DnsSecurityRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DnsSecurityRule.__pulumiType;
    }
    /**
     * Create a DnsSecurityRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.action === undefined) && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if ((!args || args.dnsResolverDomainLists === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dnsResolverDomainLists'");
            }
            if ((!args || args.dnsResolverPolicyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dnsResolverPolicyName'");
            }
            if ((!args || args.priority === undefined) && !opts.urn) {
                throw new Error("Missing required property 'priority'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["action"] = args ? args.action : undefined;
            resourceInputs["dnsResolverDomainLists"] = args ? args.dnsResolverDomainLists : undefined;
            resourceInputs["dnsResolverPolicyName"] = args ? args.dnsResolverPolicyName : undefined;
            resourceInputs["dnsSecurityRuleName"] = args ? args.dnsSecurityRuleName : undefined;
            resourceInputs["dnsSecurityRuleState"] = args ? args.dnsSecurityRuleState : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["priority"] = args ? args.priority : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["action"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["dnsResolverDomainLists"] = undefined /*out*/;
            resourceInputs["dnsSecurityRuleState"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:dnsresolver/v20230701preview:DnsSecurityRule" }, { type: "azure-native:network/v20230701preview:DnsSecurityRule" }, { type: "azure-native:network:DnsSecurityRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DnsSecurityRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DnsSecurityRule = DnsSecurityRule;
/** @internal */
DnsSecurityRule.__pulumiType = 'azure-native:dnsresolver:DnsSecurityRule';
