// @pulumi/azure-native/devtestlab/lab.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Lab = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A lab.
 *
 * Uses Azure REST API version 2018-09-15. In version 2.x of the Azure Native provider, it used API version 2018-09-15.
 */
class Lab extends pulumi.CustomResource {
    /**
     * Get an existing Lab resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Lab(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Lab.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Lab.__pulumiType;
    }
    /**
     * Create a Lab resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["announcement"] = args ? args.announcement : undefined;
            resourceInputs["environmentPermission"] = args ? args.environmentPermission : undefined;
            resourceInputs["extendedProperties"] = args ? args.extendedProperties : undefined;
            resourceInputs["labStorageType"] = (_a = (args ? args.labStorageType : undefined)) !== null && _a !== void 0 ? _a : "Premium";
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["mandatoryArtifactsResourceIdsLinux"] = args ? args.mandatoryArtifactsResourceIdsLinux : undefined;
            resourceInputs["mandatoryArtifactsResourceIdsWindows"] = args ? args.mandatoryArtifactsResourceIdsWindows : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["premiumDataDisks"] = args ? args.premiumDataDisks : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["support"] = args ? args.support : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["artifactsStorageAccount"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["defaultPremiumStorageAccount"] = undefined /*out*/;
            resourceInputs["defaultStorageAccount"] = undefined /*out*/;
            resourceInputs["loadBalancerId"] = undefined /*out*/;
            resourceInputs["networkSecurityGroupId"] = undefined /*out*/;
            resourceInputs["premiumDataDiskStorageAccount"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicIpId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueIdentifier"] = undefined /*out*/;
            resourceInputs["vaultName"] = undefined /*out*/;
            resourceInputs["vmCreationResourceGroup"] = undefined /*out*/;
        }
        else {
            resourceInputs["announcement"] = undefined /*out*/;
            resourceInputs["artifactsStorageAccount"] = undefined /*out*/;
            resourceInputs["azureApiVersion"] = undefined /*out*/;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["defaultPremiumStorageAccount"] = undefined /*out*/;
            resourceInputs["defaultStorageAccount"] = undefined /*out*/;
            resourceInputs["environmentPermission"] = undefined /*out*/;
            resourceInputs["extendedProperties"] = undefined /*out*/;
            resourceInputs["labStorageType"] = undefined /*out*/;
            resourceInputs["loadBalancerId"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["mandatoryArtifactsResourceIdsLinux"] = undefined /*out*/;
            resourceInputs["mandatoryArtifactsResourceIdsWindows"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkSecurityGroupId"] = undefined /*out*/;
            resourceInputs["premiumDataDiskStorageAccount"] = undefined /*out*/;
            resourceInputs["premiumDataDisks"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicIpId"] = undefined /*out*/;
            resourceInputs["support"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueIdentifier"] = undefined /*out*/;
            resourceInputs["vaultName"] = undefined /*out*/;
            resourceInputs["vmCreationResourceGroup"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devtestlab/v20150521preview:Lab" }, { type: "azure-native:devtestlab/v20160515:Lab" }, { type: "azure-native:devtestlab/v20180915:Lab" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Lab.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Lab = Lab;
/** @internal */
Lab.__pulumiType = 'azure-native:devtestlab:Lab';
