// @pulumi/azure-native/types/enums/kusto/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VnetState = exports.ScriptLevel = exports.PublicNetworkAccess = exports.PublicIPType = exports.PrincipalType = exports.PrincipalPermissionsAction = exports.OutboundAccess = exports.LanguageExtensionName = exports.LanguageExtensionImageName = exports.Language = exports.Kind = exports.IotHubDataFormat = exports.IdentityType = exports.EventHubDataFormat = exports.EventGridDataFormat = exports.EngineType = exports.DefaultPrincipalsModificationKind = exports.DatabaseRouting = exports.DatabasePrincipalRole = exports.DataFormat = exports.DataConnectionKind = exports.Compression = exports.ClusterPrincipalRole = exports.ClusterNetworkAccessFlag = exports.CalloutType = exports.BlobStorageEventType = exports.AzureSkuTier = exports.AzureSkuName = void 0;
exports.AzureSkuName = {
    Dev_No_SLA_Standard_D11_v2: "Dev(No SLA)_Standard_D11_v2",
    Dev_No_SLA_Standard_E2a_v4: "Dev(No SLA)_Standard_E2a_v4",
    Standard_D11_v2: "Standard_D11_v2",
    Standard_D12_v2: "Standard_D12_v2",
    Standard_D13_v2: "Standard_D13_v2",
    Standard_D14_v2: "Standard_D14_v2",
    Standard_D32d_v4: "Standard_D32d_v4",
    Standard_D16d_v5: "Standard_D16d_v5",
    Standard_D32d_v5: "Standard_D32d_v5",
    Standard_DS13_v2_1TB_PS: "Standard_DS13_v2+1TB_PS",
    Standard_DS13_v2_2TB_PS: "Standard_DS13_v2+2TB_PS",
    Standard_DS14_v2_3TB_PS: "Standard_DS14_v2+3TB_PS",
    Standard_DS14_v2_4TB_PS: "Standard_DS14_v2+4TB_PS",
    Standard_L4s: "Standard_L4s",
    Standard_L8s: "Standard_L8s",
    Standard_L16s: "Standard_L16s",
    Standard_L8s_v2: "Standard_L8s_v2",
    Standard_L16s_v2: "Standard_L16s_v2",
    Standard_L8s_v3: "Standard_L8s_v3",
    Standard_L16s_v3: "Standard_L16s_v3",
    Standard_L32s_v3: "Standard_L32s_v3",
    Standard_L8as_v3: "Standard_L8as_v3",
    Standard_L16as_v3: "Standard_L16as_v3",
    Standard_L32as_v3: "Standard_L32as_v3",
    Standard_E64i_v3: "Standard_E64i_v3",
    Standard_E80ids_v4: "Standard_E80ids_v4",
    Standard_E2a_v4: "Standard_E2a_v4",
    Standard_E4a_v4: "Standard_E4a_v4",
    Standard_E8a_v4: "Standard_E8a_v4",
    Standard_E16a_v4: "Standard_E16a_v4",
    Standard_E8as_v4_1TB_PS: "Standard_E8as_v4+1TB_PS",
    Standard_E8as_v4_2TB_PS: "Standard_E8as_v4+2TB_PS",
    Standard_E16as_v4_3TB_PS: "Standard_E16as_v4+3TB_PS",
    Standard_E16as_v4_4TB_PS: "Standard_E16as_v4+4TB_PS",
    Standard_E8as_v5_1TB_PS: "Standard_E8as_v5+1TB_PS",
    Standard_E8as_v5_2TB_PS: "Standard_E8as_v5+2TB_PS",
    Standard_E16as_v5_3TB_PS: "Standard_E16as_v5+3TB_PS",
    Standard_E16as_v5_4TB_PS: "Standard_E16as_v5+4TB_PS",
    Standard_E2ads_v5: "Standard_E2ads_v5",
    Standard_E4ads_v5: "Standard_E4ads_v5",
    Standard_E8ads_v5: "Standard_E8ads_v5",
    Standard_E16ads_v5: "Standard_E16ads_v5",
    Standard_EC8as_v5_1TB_PS: "Standard_EC8as_v5+1TB_PS",
    Standard_EC8as_v5_2TB_PS: "Standard_EC8as_v5+2TB_PS",
    Standard_EC16as_v5_3TB_PS: "Standard_EC16as_v5+3TB_PS",
    Standard_EC16as_v5_4TB_PS: "Standard_EC16as_v5+4TB_PS",
    Standard_EC8ads_v5: "Standard_EC8ads_v5",
    Standard_EC16ads_v5: "Standard_EC16ads_v5",
    Standard_E8s_v4_1TB_PS: "Standard_E8s_v4+1TB_PS",
    Standard_E8s_v4_2TB_PS: "Standard_E8s_v4+2TB_PS",
    Standard_E16s_v4_3TB_PS: "Standard_E16s_v4+3TB_PS",
    Standard_E16s_v4_4TB_PS: "Standard_E16s_v4+4TB_PS",
    Standard_E8s_v5_1TB_PS: "Standard_E8s_v5+1TB_PS",
    Standard_E8s_v5_2TB_PS: "Standard_E8s_v5+2TB_PS",
    Standard_E16s_v5_3TB_PS: "Standard_E16s_v5+3TB_PS",
    Standard_E16s_v5_4TB_PS: "Standard_E16s_v5+4TB_PS",
    Standard_E2d_v4: "Standard_E2d_v4",
    Standard_E4d_v4: "Standard_E4d_v4",
    Standard_E8d_v4: "Standard_E8d_v4",
    Standard_E16d_v4: "Standard_E16d_v4",
    Standard_E2d_v5: "Standard_E2d_v5",
    Standard_E4d_v5: "Standard_E4d_v5",
    Standard_E8d_v5: "Standard_E8d_v5",
    Standard_E16d_v5: "Standard_E16d_v5",
};
exports.AzureSkuTier = {
    Basic: "Basic",
    Standard: "Standard",
};
exports.BlobStorageEventType = {
    Microsoft_Storage_BlobCreated: "Microsoft.Storage.BlobCreated",
    Microsoft_Storage_BlobRenamed: "Microsoft.Storage.BlobRenamed",
};
exports.CalloutType = {
    Kusto: "kusto",
    Sql: "sql",
    Cosmosdb: "cosmosdb",
    External_data: "external_data",
    Azure_digital_twins: "azure_digital_twins",
    Sandbox_artifacts: "sandbox_artifacts",
    Webapi: "webapi",
    Mysql: "mysql",
    Postgresql: "postgresql",
    Genevametrics: "genevametrics",
    Azure_openai: "azure_openai",
};
exports.ClusterNetworkAccessFlag = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ClusterPrincipalRole = {
    AllDatabasesAdmin: "AllDatabasesAdmin",
    AllDatabasesViewer: "AllDatabasesViewer",
    AllDatabasesMonitor: "AllDatabasesMonitor",
};
exports.Compression = {
    None: "None",
    GZip: "GZip",
};
exports.DataConnectionKind = {
    EventHub: "EventHub",
    EventGrid: "EventGrid",
    IotHub: "IotHub",
    CosmosDb: "CosmosDb",
};
exports.DataFormat = {
    MULTIJSON: "MULTIJSON",
    JSON: "JSON",
    CSV: "CSV",
};
exports.DatabasePrincipalRole = {
    Admin: "Admin",
    Ingestor: "Ingestor",
    Monitor: "Monitor",
    User: "User",
    UnrestrictedViewer: "UnrestrictedViewer",
    Viewer: "Viewer",
};
exports.DatabaseRouting = {
    Single: "Single",
    Multi: "Multi",
};
exports.DefaultPrincipalsModificationKind = {
    Union: "Union",
    Replace: "Replace",
    None: "None",
};
exports.EngineType = {
    V2: "V2",
    V3: "V3",
};
exports.EventGridDataFormat = {
    MULTIJSON: "MULTIJSON",
    JSON: "JSON",
    CSV: "CSV",
    TSV: "TSV",
    SCSV: "SCSV",
    SOHSV: "SOHSV",
    PSV: "PSV",
    TXT: "TXT",
    RAW: "RAW",
    SINGLEJSON: "SINGLEJSON",
    AVRO: "AVRO",
    TSVE: "TSVE",
    PARQUET: "PARQUET",
    ORC: "ORC",
    APACHEAVRO: "APACHEAVRO",
    W3CLOGFILE: "W3CLOGFILE",
};
exports.EventHubDataFormat = {
    MULTIJSON: "MULTIJSON",
    JSON: "JSON",
    CSV: "CSV",
    TSV: "TSV",
    SCSV: "SCSV",
    SOHSV: "SOHSV",
    PSV: "PSV",
    TXT: "TXT",
    RAW: "RAW",
    SINGLEJSON: "SINGLEJSON",
    AVRO: "AVRO",
    TSVE: "TSVE",
    PARQUET: "PARQUET",
    ORC: "ORC",
    APACHEAVRO: "APACHEAVRO",
    W3CLOGFILE: "W3CLOGFILE",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.IotHubDataFormat = {
    MULTIJSON: "MULTIJSON",
    JSON: "JSON",
    CSV: "CSV",
    TSV: "TSV",
    SCSV: "SCSV",
    SOHSV: "SOHSV",
    PSV: "PSV",
    TXT: "TXT",
    RAW: "RAW",
    SINGLEJSON: "SINGLEJSON",
    AVRO: "AVRO",
    TSVE: "TSVE",
    PARQUET: "PARQUET",
    ORC: "ORC",
    APACHEAVRO: "APACHEAVRO",
    W3CLOGFILE: "W3CLOGFILE",
};
exports.Kind = {
    ReadWrite: "ReadWrite",
    ReadOnlyFollowing: "ReadOnlyFollowing",
};
exports.Language = {
    Python: "Python",
};
exports.LanguageExtensionImageName = {
    R: "R",
    Python3_6_5: "Python3_6_5",
    Python3_10_8: "Python3_10_8",
    Python3_10_8_DL: "Python3_10_8_DL",
    PythonCustomImage: "PythonCustomImage",
    Python3_11_7: "Python3_11_7",
    Python3_11_7_DL: "Python3_11_7_DL",
};
exports.LanguageExtensionName = {
    PYTHON: "PYTHON",
    R: "R",
};
exports.OutboundAccess = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.PrincipalPermissionsAction = {
    RetainPermissionOnScriptCompletion: "RetainPermissionOnScriptCompletion",
    RemovePermissionOnScriptCompletion: "RemovePermissionOnScriptCompletion",
};
exports.PrincipalType = {
    App: "App",
    Group: "Group",
    User: "User",
};
exports.PublicIPType = {
    IPv4: "IPv4",
    DualStack: "DualStack",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ScriptLevel = {
    Database: "Database",
    Cluster: "Cluster",
};
exports.VnetState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
