// @pulumi/azure/eventgrid/partnerConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Event Grid Partner Configuration.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example-resources",
 *     location: "West Europe",
 * });
 * const examplePartnerConfiguration = new azure.eventgrid.PartnerConfiguration("example", {
 *     resourceGroupName: example.name,
 *     defaultMaximumExpirationTimeInDays: 14,
 *     partnerAuthorizations: [{
 *         partnerRegistrationId: "804a11ca-ce9b-4158-8e94-3c8dc7a072ec",
 *         partnerName: "Auth0",
 *         authorizationExpirationTimeInUtc: "2025-02-05T00:00:00Z",
 *     }],
 *     tags: {
 *         environment: "Production",
 *     },
 * });
 * ```
 *
 * ## API Providers
 *
 * <!-- This section is generated, changes will be overwritten -->
 * This resource uses the following Azure API Providers:
 *
 * * `Microsoft.EventGrid`: 2022-06-15
 *
 * ## Import
 *
 * Event Grid Partner Configurations can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:eventgrid/partnerConfiguration:PartnerConfiguration example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1
 * ```
 */
class PartnerConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing PartnerConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PartnerConfiguration(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PartnerConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PartnerConfiguration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["defaultMaximumExpirationTimeInDays"] = state ? state.defaultMaximumExpirationTimeInDays : undefined;
            resourceInputs["partnerAuthorizations"] = state ? state.partnerAuthorizations : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["defaultMaximumExpirationTimeInDays"] = args ? args.defaultMaximumExpirationTimeInDays : undefined;
            resourceInputs["partnerAuthorizations"] = args ? args.partnerAuthorizations : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PartnerConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PartnerConfiguration = PartnerConfiguration;
/** @internal */
PartnerConfiguration.__pulumiType = 'azure:eventgrid/partnerConfiguration:PartnerConfiguration';
