// @pulumi/azure/appplatform/springCloudConfigurationService.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpringCloudConfigurationService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Spring Cloud Configuration Service.
 *
 * > **Note:** This resource is applicable only for Spring Cloud Service with enterprise tier.
 *
 * !> **Note:** Azure Spring Apps is now deprecated and will be retired on 2028-05-31 - as such the `azure.appplatform.SpringCloudConfigurationService` resource is deprecated and will be removed in a future major version of the AzureRM Provider. See https://aka.ms/asaretirement for more information.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {
 *     name: "example",
 *     location: "West Europe",
 * });
 * const exampleSpringCloudService = new azure.appplatform.SpringCloudService("example", {
 *     name: "example",
 *     location: example.location,
 *     resourceGroupName: example.name,
 *     skuName: "E0",
 * });
 * const exampleSpringCloudConfigurationService = new azure.appplatform.SpringCloudConfigurationService("example", {
 *     name: "default",
 *     springCloudServiceId: exampleSpringCloudService.id,
 *     repositories: [{
 *         name: "fake",
 *         label: "master",
 *         patterns: ["app/dev"],
 *         uri: "https://github.com/Azure-Samples/piggymetrics",
 *         searchPaths: [
 *             "dir1",
 *             "dir2",
 *         ],
 *         strictHostKeyChecking: false,
 *         username: "adminuser",
 *         password: "H@Sh1CoR3!",
 *     }],
 * });
 * ```
 *
 * ## API Providers
 *
 * <!-- This section is generated, changes will be overwritten -->
 * This resource uses the following Azure API Providers:
 *
 * * `Microsoft.AppPlatform`: 2024-01-01-preview
 *
 * ## Import
 *
 * Spring Cloud Configuration Services can be imported using the `resource id`, e.g.
 *
 * ```sh
 * $ pulumi import azure:appplatform/springCloudConfigurationService:SpringCloudConfigurationService example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resourceGroup1/providers/Microsoft.AppPlatform/spring/service1/configurationServices/configurationService1
 * ```
 */
class SpringCloudConfigurationService extends pulumi.CustomResource {
    /**
     * Get an existing SpringCloudConfigurationService resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SpringCloudConfigurationService(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SpringCloudConfigurationService.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SpringCloudConfigurationService.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["generation"] = state ? state.generation : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["refreshIntervalInSeconds"] = state ? state.refreshIntervalInSeconds : undefined;
            resourceInputs["repositories"] = state ? state.repositories : undefined;
            resourceInputs["springCloudServiceId"] = state ? state.springCloudServiceId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.springCloudServiceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'springCloudServiceId'");
            }
            resourceInputs["generation"] = args ? args.generation : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["refreshIntervalInSeconds"] = args ? args.refreshIntervalInSeconds : undefined;
            resourceInputs["repositories"] = args ? args.repositories : undefined;
            resourceInputs["springCloudServiceId"] = args ? args.springCloudServiceId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SpringCloudConfigurationService.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SpringCloudConfigurationService = SpringCloudConfigurationService;
/** @internal */
SpringCloudConfigurationService.__pulumiType = 'azure:appplatform/springCloudConfigurationService:SpringCloudConfigurationService';
